<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>5mC data analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">JumpSeq-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/JumpSeq-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">5mC data analysis</h1>
<h4 class="author"><em>Shengtong Han</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-07-31</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 028a217</p>
<!-- Add your analysis here -->
<p>Samples are <a href="https://docs.google.com/spreadsheets/d/1FHkVSgTCJDY19VcFg2goat4rXJfxofCn3CFRV-gCbfo/edit#gid=271045416">here</a>: and original samples are <a href="https://docs.google.com/spreadsheets/d/1WesFCQTrFplnXRKCHmhqtm1529DHc5Sc2tX7GkRa_-U/edit#gid=1664366512">here</a></p>
<div id="ng-5mc-replicates" class="section level2">
<h2>48ng 5mC replicates</h2>
<p>There are two replicates.</p>
<div id="reads-distribution" class="section level3">
<h3>reads distribution</h3>
<table>
<caption>number of reads in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>4,270,565</td>
<td>4,246,992</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>5,707,602</td>
<td>5,686,410</td>
</tr>
</tbody>
</table>
<p>Peak windows with reads greater than 10 are put together into category “&gt;10”.</p>
<pre class="r"><code>reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), &quot;&gt;10&quot;); reads[,1]=c(3338973, 312411, 45971, 8205, 1741, 435, 202, 83, 52, 45, 393)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of peak windows (rep 1 minus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/5mCAnalysis.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), &quot;&gt;10&quot;); reads[,1]=c(4066783, 506148, 99300, 22528, 5818, 1551, 563, 215, 130, 62, 639)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of peak windows (rep 2 plus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/5mCAnalysis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<table>
<caption>number of peak windows at FDR 0.05 (~ pvalue <span class="math inline">\(1.2\times 10^{-5}\)</span>) in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>33,312</td>
<td>34,307</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>94,687</td>
<td>94,320</td>
</tr>
</tbody>
</table>
<table>
<caption>number of peak windows at FDR 0.1 (~ pvalue <span class="math inline">\(3.5\times 10^{-5}\)</span>) in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>56,390</td>
<td>57,010</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>134,018</td>
<td>134,260</td>
</tr>
</tbody>
</table>
<table>
<caption>number of peak windows at FDR 0.5 (~ pvalue <span class="math inline">\(1.5\times 10^{-3}\)</span>) in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>181,825</td>
<td>188,726</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>188,726</td>
<td>431,088</td>
</tr>
</tbody>
</table>
<div id="use-p-value-as-cutoffs" class="section level4">
<h4>use p value as cutoffs</h4>
<table>
<caption>number of peak windows with pvalue less than <span class="math inline">\(10^{-3}\)</span> in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>237,674</td>
<td>238,205</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>356,244</td>
<td>358,641</td>
</tr>
</tbody>
</table>
<table>
<caption>number of peak windows with pvalue less than <span class="math inline">\(10^{-6}\)</span> in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>13,341</td>
<td>13,559</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>29,038</td>
<td>28,901</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="overlap-with-other-bisulfite-peaks" class="section level3">
<h3>Overlap with other bisulfite peaks</h3>
<p>download bisulfite peaks at this <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48229">site</a>. Use the data <em>GSM1173118_E14-d0.CPG.methylPattern.txt</em> (denoted as E14-d0). It has a total of 3605387 CG sites, with varying methylation levels between 0 and 1. Use 0.9 as threshold to choose CG sites as peaks, which takes 64.55%.</p>
<table>
<caption>Proportion of 5mC data (FDR0.05) overlapping with E14_d0 and the background rate is 0.0157.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.0250</td>
<td>0.0254</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.0216</td>
<td>0.0221</td>
</tr>
</tbody>
</table>
<p>Got methylated cytosines in Tab-seq paper from Gary Hon. It has 35,452,294 sites, including plus strand and minus strand. Extending the site by one base upstream and one base downstream to form 3 bp windows. All 2,057,636 strong Tab-seq sites are included in 35,452,294 methylated cytosines.</p>
<table>
<caption>Proportion of 5mC data (FDR0.05) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.8470</td>
<td>0.8520</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.8597</td>
<td>0.8618</td>
</tr>
</tbody>
</table>
<table>
<caption>Proportion of 5mC data (p value less than <span class="math inline">\(10^{-3}\)</span>) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.7890</td>
<td>0.7882</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.8023</td>
<td>0.8034</td>
</tr>
</tbody>
</table>
<table>
<caption>Proportion of 5mC data (p value less than <span class="math inline">\(10^{-6}\)</span>) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.8374</td>
<td>0.8403</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.8598</td>
<td>0.8584</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ng-5mc-replicates-1" class="section level2">
<h2>24ng 5mC replicates</h2>
<div id="reads-distribution-1" class="section level3">
<h3>reads distribution</h3>
<table>
<caption>number of reads in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>4,479,619</td>
<td>4,456,960</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>1,848,313</td>
<td>1,833,769</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-5mc-peaks" class="section level3">
<h3>overlap with 5mC peaks</h3>
<table>
<caption>Proportion of 5mC data (p value less than <span class="math inline">\(10^{-6}\)</span>) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.8511</td>
<td>0.8482</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.8441</td>
<td>0.8651</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ng-5mc-replicates-2" class="section level2">
<h2>12ng 5mC replicates</h2>
<div id="reads-distribution-2" class="section level3">
<h3>reads distribution</h3>
<table>
<caption>number of reads in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>2,954,172</td>
<td>2,927,975</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>1,006,187</td>
<td>986,516</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-5mc-peaks-1" class="section level3">
<h3>overlap with 5mC peaks</h3>
<table>
<caption>Proportion of 5mC data (p value less than <span class="math inline">\(10^{-6}\)</span>) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.8051</td>
<td>0.8356</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.8160</td>
<td>0.8403</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ng-5mc-replicates-3" class="section level2">
<h2>6ng 5mC replicates</h2>
<div id="reads-distribution-3" class="section level3">
<h3>reads distribution</h3>
<table>
<caption>number of reads in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>2,049,655</td>
<td>2,025,183</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>330,093</td>
<td>845,322</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-5mc-peaks-2" class="section level3">
<h3>overlap with 5mC peaks</h3>
<table>
<caption>Proportion of 5mC data (p value less than <span class="math inline">\(10^{-6}\)</span>) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.8069</td>
<td>0.8473</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.7305</td>
<td>0.8704</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="point4ng-5mc-replicates" class="section level2">
<h2>2point4ng 5mC replicates</h2>
<div id="reads-distribution-4" class="section level3">
<h3>reads distribution</h3>
<table>
<caption>number of reads in every replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>562,794</td>
<td>550,530</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>231,751</td>
<td>224,961</td>
</tr>
</tbody>
</table>
<table>
<caption>number of peak windows at FDR 0.05 in every replicate.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>494</td>
<td>552</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>345</td>
<td>469</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-other-bisulfite-peaks-1" class="section level3">
<h3>Overlap with other bisulfite peaks</h3>
<div id="overlap-with-cpgs-from-gary-hon." class="section level4">
<h4>overlap with CpG’s from Gary Hon.</h4>
<table>
<caption>Proportion of 5mC data (FDR0.05) overlapping with methylated cytosines and the background rate is 0.1450.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>minus</th>
<th>plus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep1</td>
<td>0.7895</td>
<td>0.7826</td>
</tr>
<tr class="even">
<td>rep2</td>
<td>0.6174</td>
<td>0.6397</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<div id="enrichment" class="section level3">
<h3>Enrichment</h3>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.7895, 0.1450)
percent[,2]=c(0.8069, 0.1450)

percent[,3]=c(0.8051, 0.1450)
percent[,4]=c(0.8511, 0.1450)

percent[,5]=c(0.8374, 0.1450)
colnames(percent)=c(&quot;2.4ng&quot;, &quot;6ng&quot;, &quot;12ng&quot;, &quot;24ng&quot;, &quot;48ng&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,1.2),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;Enrichment with methylated cytosine from Gary Hon (Data:rep 1 minus)&quot;, xlab=&quot;&quot;, cex.names=0.7)</code></pre>
<p><img src="figure/5mCAnalysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/5mCAnalysis.Rmd/unnamed-chunk-4-1.png" alt="Enrichment of methylated cytosine from Gary Hon  at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. " width="672" />
<p class="caption">
Enrichment of methylated cytosine from Gary Hon at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng.
</p>
</div>
<ul>
<li>cutoff used for peak windows: 2.4ng (FDR0.05), others use p values less than <span class="math inline">\(10^{-6}\)</span>.</li>
</ul>
</div>
<div id="sensitivity" class="section level3">
<h3>Sensitivity</h3>
<div class="figure" style="text-align: center">
<img src="figure/5mCAnalysis.Rmd/unnamed-chunk-5-1.png" alt="Proportion of  methylated cytosine from Gary Hon  at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng" width="672" />
<p class="caption">
Proportion of methylated cytosine from Gary Hon at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng
</p>
</div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.5.0  backports_1.1.2 magrittr_1.5    rprojroot_1.3-2
 [5] tools_3.5.0     htmltools_0.3.6 yaml_2.1.19     Rcpp_0.12.16   
 [9] stringi_1.1.7   rmarkdown_1.9   highr_0.6       knitr_1.20     
[13] git2r_0.21.0    stringr_1.3.0   digest_0.6.15   evaluate_0.10.1</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
