<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong" />

<meta name="date" content="2018-07-25" />

<title>FinalAnalysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">JumpSeq-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/JumpSeq-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FinalAnalysis</h1>
<h4 class="author"><em>Shengtong</em></h4>
<h4 class="date"><em>July 25, 2018</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<div id="proof-of-concept" class="section level2">
<h2>Proof of concept</h2>
<p>`</p>
<div id="accuracy-of-jump-seq-assessed-with-synthetic-dna" class="section level3">
<h3>Accuracy of Jump-seq assessed with synthetic DNA</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-1-1.png" alt="Fig 1.1, Number of reads (in million on y axis) at different sites, 5hmC (red) and unmodified C (blue), whose numbers are in brackets." width="672" />
<p class="caption">
Fig 1.1, Number of reads (in million on y axis) at different sites, 5hmC (red) and unmodified C (blue), whose numbers are in brackets.
</p>
</div>
</div>
<div id="quantification-of-5hmc-linear-correlation-of-read-counts-vs.5hmc-amount" class="section level3">
<h3>Quantification of 5hmC: linear correlation of read counts vs.Â 5hmC amount</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-2-1.png" alt=" Fig 1.2: The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in" width="672" />
<p class="caption">
Fig 1.2: The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in
</p>
</div>
</div>
<div id="visualization" class="section level3">
<h3>Visualization</h3>
<p>Figure: IGV view, show peaks at different DNA levels, also show peaks from Tab-seq or Nano-seal. See Figure 1B of Nano-seal paper.</p>
</div>
<div id="spatial-distribution" class="section level3">
<h3>Spatial distribution</h3>
<p>The reference is GSM882244_FDR_0.0484. First got non-overlapping 5hmC regions and make sure each window has only one 5hmC site. For each window, count the reads depth at every site, then add up reads depth of these non-overlapping windows. Reads distribution is the accumulative reads depth at every base of all non-overlapping windows. Input Bam file is the replicate with largest number of reads.</p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
</div>
<div id="self-quantification" class="section level2">
<h2>Self-quantification</h2>
<div id="corrleation-among-replicates" class="section level3">
<h3>corrleation among replicates</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-5-1.png" alt="Reads correlations in 1kb windows between two 5hmC replicates. log2 transformation in calculation of correlations  of number of reads" width="672" />
<p class="caption">
Reads correlations in 1kb windows between two 5hmC replicates. log2 transformation in calculation of correlations of number of reads
</p>
</div>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<div class="figure">
<img src="figure/correltion-combined-5hmc.png" alt="Six samples of 12 samplers are combined as combined sample 1 and the rest six are combined sample 2. Reads in 20 bp windows are used to calculate correlations between these two combined samples. log2 transformation is used for reads on both x and y axis." />
<p class="caption">Six samples of 12 samplers are combined as combined sample 1 and the rest six are combined sample 2. Reads in 20 bp windows are used to calculate correlations between these two combined samples. log2 transformation is used for reads on both x and y axis.</p>
</div>
<div class="figure">
<img src="figure/VennDiagram_5hmC_0.6ng_tworeplicate.png" alt="Venn diagram of two 0.6ng 5hmC replicates. For each replicate, 1kb windows are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle." />
<p class="caption">Venn diagram of two 0.6ng 5hmC replicates. For each replicate, 1kb windows are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle.</p>
</div>
</div>
<div id="correlation-between-different-dna-levels" class="section level3">
<h3>correlation between different DNA levels</h3>
<div class="figure">
<img src="figure/VennDiagram_5hmC_0.6ng_48ng.png" alt="Venn diagram of 0.6ng 5hmC and 48ng 5hmC. For each sample, 1kb windows are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle." />
<p class="caption">Venn diagram of 0.6ng 5hmC and 48ng 5hmC. For each sample, 1kb windows are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle.</p>
</div>
</div>
</div>
<div id="comparison-with-other-methods" class="section level2">
<h2>Comparison with other methods</h2>
<div id="correlation-between-jump-seq-and-nano-seal" class="section level3">
<h3>correlation between Jump-seq and nano-seal</h3>
<p>The correlation between combined 48ng 5hmC sample and 50ng nano-seal sample is 0.08. The calculation is problematic as jump-seq is using the actual number of reads while nano-seal uses average number across cells.</p>
</div>
<div id="comparison-of-jump-vs-nano-seal-with-strong-tab-seq-as-gold-standard" class="section level3">
<h3>Comparison of Jump vs nano-seal with strong TAB-seq as gold standard</h3>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<ul>
<li>JumpSeq: 1kb windows are called at FDR 0.05. To make a fair comparison, we trim the jump-seq peak windows to be of the same length as nano_seq peaks.</li>
</ul>
</div>
<div id="enrichments" class="section level3">
<h3>Enrichments</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-15-1.png" alt="For jump-seq, we called 1kb windows at FDR0.05 as peaks and computed the proportion of peaks (in red) overlapping with Nano-seal peaks and Tab-seq peaks. Randomly pick same number of 1kb windows in the whole genome as jump-seq peaks at different DNA levels and calculate its proportion of windows (in blue) overlapping with Nano-seal peaks and Tab-seq peaks extended to 1kb. " width="672" />
<p class="caption">
For jump-seq, we called 1kb windows at FDR0.05 as peaks and computed the proportion of peaks (in red) overlapping with Nano-seal peaks and Tab-seq peaks. Randomly pick same number of 1kb windows in the whole genome as jump-seq peaks at different DNA levels and calculate its proportion of windows (in blue) overlapping with Nano-seal peaks and Tab-seq peaks extended to 1kb.
</p>
</div>
<ul>
<li>Jump-seq; 1kb window are called at FDR 0.05</li>
</ul>
</div>
<div id="sensitvity" class="section level3">
<h3>Sensitvity</h3>
<table>
<caption>
Number of Tab-seq peaks recovered by Jump_seq and nano-seal
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
0.6ng
</th>
<th style="text-align:right;">
2.4ng
</th>
<th style="text-align:right;">
6ng
</th>
<th style="text-align:right;">
12ng
</th>
<th style="text-align:right;">
24ng
</th>
<th style="text-align:right;">
48ng
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Jump_seq
</td>
<td style="text-align:right;">
832
</td>
<td style="text-align:right;">
1116
</td>
<td style="text-align:right;">
19686
</td>
<td style="text-align:right;">
219644
</td>
<td style="text-align:right;">
649783
</td>
<td style="text-align:right;">
715179
</td>
</tr>
<tr>
<td style="text-align:left;">
Nano_seal (50ng)
</td>
<td style="text-align:right;">
465501
</td>
<td style="text-align:right;">
465501
</td>
<td style="text-align:right;">
465501
</td>
<td style="text-align:right;">
465501
</td>
<td style="text-align:right;">
465501
</td>
<td style="text-align:right;">
465501
</td>
</tr>
</tbody>
</table>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>To be fair, 1kb Jump-sea peaks are trimed to be of the same length as nano-seal peaks.</p>
</div>
<div id="overlap-with-other-peaks-by-venn-diagram" class="section level3">
<h3>Overlap with other peaks by Venn diagram</h3>
<p>If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of which 686652 windows are overlapping with strong TAB-seq windows which are extended 10 bp downtream and 10 bp upstream. for strong TAB-seq, there are 2057636 20bp extended windows and 763937 are overlapping with Jump-seq windows.</p>
<div class="figure">
<img src="figure/VennDiagram.png" alt="Venn diagram of 5hmC and TAB-seq. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with TAB-seq and the proportion of peaks of TAB-seq overlapping with jump-seq are presented inside the circle." />
<p class="caption">Venn diagram of 5hmC and TAB-seq. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with TAB-seq and the proportion of peaks of TAB-seq overlapping with jump-seq are presented inside the circle.</p>
</div>
<p>If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of which 386331 (16.32%) windows are overlapping with hmC seal. For hmC seal, there are 64865 peaks and 58280 (89.85%) are overlapping with 5hmC peaks.</p>
<div class="figure">
<img src="figure/VennDiagram_hmc_hmcseal.png" alt="Venn diagram of 5hmC and hmC-seal. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with hmC-seal and the proportion of peaks of hmC-seal overlapping with jump-seq are presented inside the circle." />
<p class="caption">Venn diagram of 5hmC and hmC-seal. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with hmC-seal and the proportion of peaks of hmC-seal overlapping with jump-seq are presented inside the circle.</p>
</div>
<p>If we use single 48gn 5hmC sample, then 28095/97548=28.80% 1kb windows are overllaping with nano-seal peaks and 24920/64865=38.42% nano-seal peaks overlapping with jump-seq.</p>
</div>
</div>
<div id="bis-seq-5mc" class="section level2">
<h2>BiS-seq (5mC)</h2>
<div id="enrichment" class="section level3">
<h3>Enrichment</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-18-1.png" alt="Enrichment of methylated cytosine from Gary Hon  at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. Number of peak windows are shown over the bar." width="672" />
<p class="caption">
Enrichment of methylated cytosine from Gary Hon at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. Number of peak windows are shown over the bar.
</p>
</div>
<ul>
<li>cutoff used for peak windows: 2.4ng (FDR0.05), others use p values less than <span class="math inline">\(10^{-6}\)</span>.</li>
</ul>
</div>
<div id="sensitivity" class="section level3">
<h3>Sensitivity</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-19-1.png" alt="Normalized proportion of  methylated cytosine from Gary Hon  at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng" width="672" />
<p class="caption">
Normalized proportion of methylated cytosine from Gary Hon at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng
</p>
</div>
</div>
<div id="correlation-between-replicates" class="section level3">
<h3>Correlation between replicates</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-20-1.png" alt="Correlation of reads in 20 bp windows between two replicates of 2.4ng 5mC. log2 transformation is used for computing correlations" width="672" />
<p class="caption">
Correlation of reads in 20 bp windows between two replicates of 2.4ng 5mC. log2 transformation is used for computing correlations
</p>
</div>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-21-1.png" alt="Correlation of reads in 20 bp windows between two replicates of 6ng 5mC. log2 transformation is used for computing correlations" width="672" />
<p class="caption">
Correlation of reads in 20 bp windows between two replicates of 6ng 5mC. log2 transformation is used for computing correlations
</p>
</div>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-22-1.png" alt="Correlation of reads in 20 bp windows between two replicates of 12ng 5mC. log2 transformation is used for computing correlations" width="672" />
<p class="caption">
Correlation of reads in 20 bp windows between two replicates of 12ng 5mC. log2 transformation is used for computing correlations
</p>
</div>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-23-1.png" alt="Correlation of reads in 20 bp windows between two replicates of 24ng 5mC. log2 transformation is used for computing correlations" width="672" />
<p class="caption">
Correlation of reads in 20 bp windows between two replicates of 24ng 5mC. log2 transformation is used for computing correlations
</p>
</div>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-24-1.png" alt="Correlation of reads in 20 bp windows between two replicates of 48ng 5mC. log2 transformation is used for computing correlations" width="672" />
<p class="caption">
Correlation of reads in 20 bp windows between two replicates of 48ng 5mC. log2 transformation is used for computing correlations
</p>
</div>
</div>
</div>
<div id="enrichment-of-genomic-features" class="section level2">
<h2>Enrichment of genomic features</h2>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-25-1.png" alt=" Eleven 48 ng samples are combined together and 20 bp peak windows are called at FDR 0.05. Random windows are randomly selected  in the whole genome with the same amount of peak windows. The proportion of chosen peak windows (red) and random windows (blue) overlapping in  different genomic regions are calculated and the corresponding odds ratio are shown above the red bar" width="672" />
<p class="caption">
Eleven 48 ng samples are combined together and 20 bp peak windows are called at FDR 0.05. Random windows are randomly selected in the whole genome with the same amount of peak windows. The proportion of chosen peak windows (red) and random windows (blue) overlapping in different genomic regions are calculated and the corresponding odds ratio are shown above the red bar
</p>
</div>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-26-1.png" alt="For 0.6ng 5hmC sample, 20 bp windows with at least one reads are called peak windows. Random windows are chosen in the whole genome in the same amount of peak windows. The proportion of peak windows and random windows in different genomic regions (200bp bins) are computeed. " width="672" />
<p class="caption">
For 0.6ng 5hmC sample, 20 bp windows with at least one reads are called peak windows. Random windows are chosen in the whole genome in the same amount of peak windows. The proportion of peak windows and random windows in different genomic regions (200bp bins) are computeed.
</p>
</div>
<ul>
<li>cis-regulatory regions are extended 200bp windows.</li>
<li>jump peaks are 20 bp windows with at least one reads.</li>
</ul>
</div>
<div id="pn-analysis" class="section level2">
<h2>PN analysis</h2>
<p>For both male and female, at every stage, the nnumber of reads are counted in each region. Within every region, the reads are normalized around zero. To see the overall pattern, in the second figure, reads from all regions are put together.</p>
<div id="hmc" class="section level3">
<h3>5hmC</h3>
<p><strong>Use maximum number of reads as baseline</strong></p>
<div id="replicate-1" class="section level4">
<h4>Replicate 1</h4>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-29-1.png" width="672" /><img src="FinalAnalysis_files/figure-html/unnamed-chunk-29-2.png" width="672" /></p>
</div>
<div id="replicate-2" class="section level4">
<h4>Replicate 2</h4>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-30-1.png" width="672" /><img src="FinalAnalysis_files/figure-html/unnamed-chunk-30-2.png" width="672" /></p>
</div>
</div>
<div id="mc" class="section level3">
<h3>5mC</h3>
<p><strong>Use maximum number of reads as baseline</strong></p>
<div id="replicate-1-1" class="section level4">
<h4>Replicate 1</h4>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-31-1.png" width="672" /><img src="FinalAnalysis_files/figure-html/unnamed-chunk-31-2.png" width="672" /></p>
</div>
<div id="replicate-2-1" class="section level4">
<h4>Replicate 2</h4>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-32-1.png" width="672" /><img src="FinalAnalysis_files/figure-html/unnamed-chunk-32-2.png" width="672" /></p>
</div>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
