<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong" />

<meta name="date" content="2018-07-25" />

<title>FinalAnalysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">JumpSeq-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/JumpSeq-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FinalAnalysis</h1>
<h4 class="author"><em>Shengtong</em></h4>
<h4 class="date"><em>July 25, 2018</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<div id="accuracy-of-jump-seq-assessed-with-synthetic-dna-estimation-of-fp-rates." class="section level2">
<h2>Accuracy of Jump-seq assessed with synthetic DNA: estimation of FP rates.</h2>
<p>Figure: barplot, CPM (counts per million) of 5hmC vs. unmodified C. Show number of sites in the plot.</p>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-1-1.png" alt="Number of sites (5hmC or unmodified C ) are in brackets" width="672" />
<p class="caption">
Number of sites (5hmC or unmodified C ) are in brackets
</p>
</div>
</div>
<div id="quantification-of-5hmc-linear-correlation-of-read-counts-vs.5hmc-amount" class="section level2">
<h2>Quantification of 5hmC: linear correlation of read counts vs. 5hmC amount</h2>
<p>Figure: scatter plot.</p>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-2-1.png" alt="The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in" width="672" />
<p class="caption">
The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in
</p>
</div>
</div>
<div id="visualization" class="section level2">
<h2>Visualization</h2>
<p>Figure: IGV view, show peaks at different DNA levels, also show peaks from Tab-seq or Nano-seal. See Figure 1B of Nano-seal paper.</p>
</div>
<div id="comparison-across-replicates-and-different-dna-levels" class="section level2">
<h2>Comparison across replicates and different DNA levels</h2>
<p>Correlation: 0.6 ng - 48 ng. Both 5mC and 5hmC. Figure: scatter plot, choose appropriate window size. Quantification of peak strength by CPM, divided by number of C’s in a peak. See Figure 1C. of Nano-seal. Figure: heat-map to summarize the results. See Figure S1C. Estimate precision and sensitivity of 0.6 ng vs. 48 ng. Figure: barplots, show precision and sens. at different DNA levels (using sites at 48ng as true sites).</p>
<div class="figure">
<img src="figure/5hmC_replicate_correlation_1000bpwin_48ng.png" alt="Correlation among 48ng 5hmC with window size of 1kbp" />
<p class="caption">Correlation among 48ng 5hmC with window size of 1kbp</p>
</div>
<div class="figure">
<img src="figure/5hmC_replicate_correlation_100bpwin.png" alt="Correlation among 48ng 5hmC with window size of 100bp" />
<p class="caption">Correlation among 48ng 5hmC with window size of 100bp</p>
</div>
<div class="figure">
<img src="figure/5hmC_replicate_correlation_500bpwin.png" alt="Correlation among 48ng 5hmC with window size of 500bp" />
<p class="caption">Correlation among 48ng 5hmC with window size of 500bp</p>
</div>
<div class="figure">
<img src="figure/5hmC_replicate_correlation_1000bpwin.png" alt="Correlation among 48ng 5hmC with window size of 1kbp" />
<p class="caption">Correlation among 48ng 5hmC with window size of 1kbp</p>
</div>
<div class="figure">
<img src="figure/5hmC_replicate_correlation_1000bpwin_48ng0.6ng.png" alt="Correlation of 5hmC between 48ng and 0.6ng with window size of 1kbp" />
<p class="caption">Correlation of 5hmC between 48ng and 0.6ng with window size of 1kbp</p>
</div>
</div>
<div id="comparison-of-jump-vs-nano-seal-with-strong-tab-seq-as-gold-standard" class="section level2">
<h2>Comparison of Jump vs nano-seal with strong TAB-seq as gold standard</h2>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<ul>
<li>JumpSeq: 1kb windows are called at FDR 0.05</li>
</ul>
</div>
<div id="comparison-with-tab-seq" class="section level2">
<h2>Comparison with Tab-seq</h2>
<p>48ng hmC: combine the replicates. Assess FP rates, sensitivity. Figure: barplot, precision (percent of Tab-seq peaks) as function of DNA level, and number of sites/peaks found. Comparison at window level (e.g. 100bp). See Figure S1F of Nano-seal paper. Figure: spatial distribution of reads, centered on Tab-seq peaks. See Figure S1G and S1H of Nano-seal paper.</p>
<div id="section" class="section level3">
<h3></h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-5-1.png" alt=" From 0.6ng to  48ng 5hmC samples, 20bp peak windows are called at FDR 0.05 for jump-seq. Random windows are chosen in the whole genome in the same amount of peak windows. The overlap of peak windows by jump-seq and random windows with TAB-seq peaks are colored in red and blue respectively. Number of peak windows are shown over the bar. " width="672" />
<p class="caption">
From 0.6ng to 48ng 5hmC samples, 20bp peak windows are called at FDR 0.05 for jump-seq. Random windows are chosen in the whole genome in the same amount of peak windows. The overlap of peak windows by jump-seq and random windows with TAB-seq peaks are colored in red and blue respectively. Number of peak windows are shown over the bar.
</p>
</div>
<ul>
<li>FDR 0.05 is used for determine peak windows and sated clearly otherwise</li>
<li>To make fair comparison, in each case, the same amount of random windows as peak windows are picked in the whole genome and compute the its proportion windows overlpping with peaks from other methods.</li>
<li>In some case, sinlge sample was used and in some cases, may be combined samples are used, but the enrichments are similar in both cases.</li>
</ul>
</div>
<div id="sensitivity" class="section level3">
<h3>Sensitivity</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-6-1.png" alt="Sensitivity" width="672" />
<p class="caption">
Sensitivity
</p>
</div>
<ul>
<li>1kb windows are called at FDR 0.05</li>
</ul>
</div>
<div id="spatial-distribution" class="section level3">
<h3>Spatial distribution</h3>
<p>The reference is GSM882244_FDR_0.0484. First got non-overlapping 5hmC regions and make sure each window has only one 5hmC site. For each window, count the reads depth at every site, then add up reads depth of these non-overlapping windows. Reads distribution is the accumulative reads depth at every base of all non-overlapping windows. Input Bam file is the replicate with largest number of reads.</p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
</div>
<div id="comparison-with-nano-seal" class="section level2">
<h2>Comparison with Nano-seal</h2>
<p>0.6 - 48 ng hmC. Assess FP rates, sensitivity. Figure: similar to Tab-seq.</p>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-8-1.png" alt=" From 0.6ng to  48ng 5hmC samples, 20bp peak windows are called at FDR 0.05 for jump-seq. Random windows are chosen in the whole genome in the same amount of peak windows. The overlap of peak windows by jump-seq and random windows with Nano-seal peaks are colored in red and blue respectively.  Number of peak windows are shown over the bar." width="672" />
<p class="caption">
From 0.6ng to 48ng 5hmC samples, 20bp peak windows are called at FDR 0.05 for jump-seq. Random windows are chosen in the whole genome in the same amount of peak windows. The overlap of peak windows by jump-seq and random windows with Nano-seal peaks are colored in red and blue respectively. Number of peak windows are shown over the bar.
</p>
</div>
<ul>
<li>FDR 0.05 is used for determine peak windows and sated clearly otherwise</li>
<li>To make fair comparison, in each case, the same amount of random windows as peak windows are picked in the whole genome and compute the its proportion windows overlpping with peaks from other methods.</li>
<li>In some case, sinlge sample was used and in some cases, may be combined samples are used, but the enrichments are similar in both cases.</li>
</ul>
<div id="sensitivity-1" class="section level3">
<h3>Sensitivity</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-9-1.png" alt="Sensitivity" width="672" />
<p class="caption">
Sensitivity
</p>
</div>
<ul>
<li>1kb windows are called at FDR 0.05</li>
</ul>
</div>
</div>
<div id="comparison-with-bis-seq-5mc" class="section level2">
<h2>Comparison with BiS-seq (5mC)</h2>
<p>2.4 - 48 ng. Assess FP rates, sensitivity. Figure: similar to Tab-seq.</p>
<div id="enrichment" class="section level3">
<h3>Enrichment</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-10-1.png" alt="Enrichment of methylated cytosine from Gary Hon  at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. Number of peak windows are shown over the bar." width="672" />
<p class="caption">
Enrichment of methylated cytosine from Gary Hon at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. Number of peak windows are shown over the bar.
</p>
</div>
<ul>
<li>cutoff used for peak windows: 2.4ng (FDR0.05), others use p values less than <span class="math inline">\(10^{-6}\)</span>.</li>
</ul>
</div>
<div id="sensitivity-2" class="section level3">
<h3>Sensitivity</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-11-1.png" alt="Normalized proportion of  methylated cytosine from Gary Hon  at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng" width="672" />
<p class="caption">
Normalized proportion of methylated cytosine from Gary Hon at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng
</p>
</div>
</div>
</div>
<div id="enrichment-of-genomic-features-enhancer-ctcf-etc." class="section level2">
<h2>Enrichment of genomic features: enhancer, CTCF, etc.</h2>
<p>Figure: barplot, for each feature, two bars, observed enrichment (fold or proportion) vs. random. See Figure 3B. of Tab-seq paper. Remark: alt. figure, correlation of read density vs. strength of features. Box-plot: range of read density vs. strong, weak enhancers.</p>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-12-1.png" alt=" Eleven 48 ng samples are combined together and 20 bp peak windows are called at FDR 0.05. Random windows are randomly selected  in the whole genome with the same amount of peak windows. The proportion of chosen peak windows (red) and random windows (blue) overlapping in  different genomic regions are calculated and the corresponding odds ratio are shown above the red bar" width="672" />
<p class="caption">
Eleven 48 ng samples are combined together and 20 bp peak windows are called at FDR 0.05. Random windows are randomly selected in the whole genome with the same amount of peak windows. The proportion of chosen peak windows (red) and random windows (blue) overlapping in different genomic regions are calculated and the corresponding odds ratio are shown above the red bar
</p>
</div>
</div>
<div id="high-resolution-distribution-of-jump-seq-reads" class="section level2">
<h2>High resolution distribution of Jump-seq reads</h2>
<p>Figure: histogram of reads centered on Tab-seq sites, at a small scale 0-50bp.</p>
<p><img src="FinalAnalysis_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="ng-5hmc" class="section level2">
<h2>48ng 5hmC</h2>
<div id="overlap-by-venn-diagram" class="section level3">
<h3>Overlap by Venn diagram</h3>
<p>If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of which 686652 windows are overlapping with strong TAB-seq windows which are extended 10 bp downtream and 10 bp upstream. for strong TAB-seq, there are 2057636 20bp extended windows and 763937 are overlapping with Jump-seq windows.</p>
<div class="figure">
<img src="figure/VennDiagram.png" alt="Venn diagram of 5hmC and TAB-seq. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with TAB-seq and the proportion of peaks of TAB-seq overlapping with jump-seq are presented inside the circle." />
<p class="caption">Venn diagram of 5hmC and TAB-seq. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with TAB-seq and the proportion of peaks of TAB-seq overlapping with jump-seq are presented inside the circle.</p>
</div>
<p>If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of which 386331 (16.32%) windows are overlapping with hmC seal. For hmC seal, there are 64865 peaks and 58280 (89.85%) are overlapping with 5hmC peaks.</p>
<div class="figure">
<img src="figure/VennDiagram_hmc_hmcseal.png" alt="Venn diagram of 5hmC and hmC-seal. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with hmC-seal and the proportion of peaks of hmC-seal overlapping with jump-seq are presented inside the circle." />
<p class="caption">Venn diagram of 5hmC and hmC-seal. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with hmC-seal and the proportion of peaks of hmC-seal overlapping with jump-seq are presented inside the circle.</p>
</div>
</div>
</div>
<div id="ng-5hmc-1" class="section level2">
<h2>0.6ng 5hmC</h2>
<div id="enrichment-in-genomic-regions" class="section level3">
<h3>enrichment in genomic regions</h3>
<div class="figure">
<img src="FinalAnalysis_files/figure-html/unnamed-chunk-14-1.png" alt="For 0.6ng 5hmC sample, 20 bp windows with at least one reads are called peak windows. Random windows are chosen in the whole genome in the same amount of peak windows. The proportion of peak windows and random windows in different genomic regions (200bp bins) are computeed. " width="672" />
<p class="caption">
For 0.6ng 5hmC sample, 20 bp windows with at least one reads are called peak windows. Random windows are chosen in the whole genome in the same amount of peak windows. The proportion of peak windows and random windows in different genomic regions (200bp bins) are computeed.
</p>
</div>
<ul>
<li>cis-regulatory regions are extended 200bp windows.</li>
<li>jump peaks are 20 bp windows with at least one reads.</li>
</ul>
</div>
<div id="reads-overlap-with-single-48ng-sample" class="section level3">
<h3>Reads overlap with single 48ng sample</h3>
<p>0.6ng: 1085 20 bp windows of 21188</p>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
