<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>5hmC data analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">JumpSeq-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/JumpSeq-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">5hmC data analysis</h1>
<h4 class="author"><em>Shengtong Han</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-03-19</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4ba08f4</p>
<!-- Add your analysis here -->
<p>All samples are <a href="https://docs.google.com/spreadsheets/d/1FHkVSgTCJDY19VcFg2goat4rXJfxofCn3CFRV-gCbfo/edit#gid=400442004">here</a>: 48 ng (11); 24 ng (5); 12 ng (5); 6ng (5); 2.4 ng (5); 0.6ng (2).</p>
<p>Mouse genome mm9 has 2,654,911,517 bases, 507,439,500 (19.11324%) cytosines and 507,585,491 (19.11873%) Gumines.</p>
<div id="ng-5hmc-replicates" class="section level2">
<h2>48ng 5hmC replicates</h2>
<p>There are eleven 48ng 5hmC replicates and each one has two strands, plus+minus.</p>
<table style="width:81%;">
<caption>Number of reads in plus and minus strand for each 48 ng replicate.</caption>
<colgroup>
<col width="8%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="9%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th>rep</th>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
<th>6</th>
<th>7</th>
<th>8</th>
<th>9</th>
<th>10</th>
<th>11</th>
<th>combine</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>#reads(plus)</td>
<td>3,516,149</td>
<td>3,781,024</td>
<td>2,014,714</td>
<td>6,982,592</td>
<td>4,741,743</td>
<td>5,036,004</td>
<td>5,825,862</td>
<td>5,532,109</td>
<td>5,072,780</td>
<td>3,951,163</td>
<td>3,641,213</td>
<td>50,057,101</td>
</tr>
<tr class="even">
<td>#reads(minus)</td>
<td>3,524,627</td>
<td>3,793,291</td>
<td>2,021,689</td>
<td>6,995,038</td>
<td>4,754,910</td>
<td>5,048,479</td>
<td>5,834,324</td>
<td>5,546,279</td>
<td>5,059,828</td>
<td>3,935,679</td>
<td>3,631,379</td>
<td>50,183,793</td>
</tr>
</tbody>
</table>
<div id="downsample" class="section level3">
<h3>Downsample</h3>
<p>Combine all 48ng replicates together, downsample <span class="math inline">\(40\%, 50\%, \cdots, 90\%\)</span> reads, call peak windows, and see how #peak.windows (absolute peak windows or overlapping with other data) vary with number of reads.</p>
<pre class="bash"><code>samtools merge output.sorted.bam input1.sorted.bam input2.sorted.bam # merge together sorted bam files</code></pre>
<pre class="bash"><code>samtools view -s 0.25 -b input.all.bam &gt; input_25p.sam # downsample 25% reads </code></pre>
<table>
<caption>Actutrally number of reads from each downsample with ratios of <span class="math inline">\(40\%, 50\%, \cdots, 100\%\)</span>.</caption>
<thead>
<tr class="header">
<th>ratio</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>40%</td>
<td>20,027,749</td>
<td>20,070,771</td>
</tr>
<tr class="even">
<td>50%</td>
<td>25,029,777</td>
<td>25,091,279</td>
</tr>
<tr class="odd">
<td>60%</td>
<td>30,035,435</td>
<td>30,110,375</td>
</tr>
<tr class="even">
<td>70%</td>
<td>35,040,232</td>
<td>35,127,703</td>
</tr>
<tr class="odd">
<td>80%</td>
<td>40,044,961</td>
<td>40,149,113</td>
</tr>
<tr class="even">
<td>90%</td>
<td>45,048,466</td>
<td>45,168,687</td>
</tr>
<tr class="odd">
<td>100%</td>
<td>50,057,101</td>
<td>50,183,793</td>
</tr>
</tbody>
</table>
<div id="reads-distribution" class="section level4">
<h4>Reads distribution</h4>
<p>Windows with reads greater than 10 are put together into category 11.</p>
<pre class="r"><code>reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), &quot;&gt;10&quot;); reads.plus[,1]=c(9371352, 2190106, 772609, 334745, 162092, 84161, 45907, 25962, 15529, 9208, 16960)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of peak windows (40% reads plus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), &quot;&gt;10&quot;); reads.plus[,1]=c(14075383, 4568606, 2003109, 1078497, 642833, 411173, 274861, 188434, 133511, 95662, 300518)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of peak windows (100% reads plus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For each down sample, call base level coverage, call peak windows and count how many peak windows at specified FDR level.</p>
<pre class="r"><code>num.win.FDR5per=c(759517, 969627, 1299793, 1599795, 1891138, 2107755, 2368196)
num.reads=c(20070771, 25091279, 30110375, 35127703, 40149113, 45168687, 50183793)
plot(num.reads, num.win.FDR5per, xlab=&quot;Number of reads&quot;, ylab=&quot;# peak windowsof 20 bp at FDR 0.05&quot;, type=&quot;o&quot;, main=&quot;minus strands&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>num.win.FDR5per=c(760022, 968832, 1309038, 1597802, 1905666, 2124123, 2366463)
num.reads=c(20027749, 25029777, 30035435, 35040232, 40044961, 45048466, 50057101)
plot(num.reads, num.win.FDR5per, xlab=&quot;Number of reads&quot;, ylab=&quot;# peak windowsof 20 bp at FDR 0.05&quot;, type=&quot;o&quot;, main=&quot;plus strands&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Based on the current observations, the number of peak windows grows as the read coverage increases, it does not show significant plateau</strong>.</p>
</div>
<div id="comparison-with-dalis-peaks" class="section level4">
<h4>Comparison with Dali’s Peaks</h4>
<p>There are 64865 peak windows of varying size in Dali’s data. With each down sample, call peak windows of 20bp.</p>
<pre class="r"><code>jump.rate.minus=c(0.1853, 0.1842, 0.1779, 0.1722, 0.1678, 0.1648, 0.1633)
jump.rate.plus=c(0.1850, 0.1840, 0.1775, 0.1717, 0.1669, 0.1635, 0.1631)
dali.rate.minus=c(0.7455, 0.7967, 0.8420, 0.8654, 0.8816, 0.8898, 0.8988)
dali.rate.plus=c(0.7456, 0.7956, 0.8427, 0.8657, 0.8810, 0.8897, 0.8982)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of Jump-seq&quot;, col=1, main=&quot;20bp, FDR 0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of nano hmC_seal&quot;, col=3, main=&quot;20bp, FDR 0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(3,4), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)
mtext(&quot;overlap between Jump-seq &amp; nano hmC_seal&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>jump.rate.minus=c(0.1501, 0.1496, 0.1450, 0.1394, 0.1345, 0.1305, 0.1275)
jump.rate.plus=c(0.1495, 0.1488, 0.1447, 0.1392, 0.1344, 0.1304, 0.1274)
dali.rate.minus=c(0.8736, 0.8920, 0.9042, 0.9133, 0.9193, 0.9238, 0.9273)
dali.rate.plus=c(0.8732, 0.8913, 0.9036, 0.9132, 0.9187, 0.9242, 0.9271)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of Jump-seq&quot;, col=1, main=&quot;20bp, FDR 0.5&quot;, cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of nano hmC_seal&quot;, col=3, main=&quot;20bp, FDR 0.5&quot;, cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(3,4), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)
mtext(&quot;overlap between Jump-seq &amp; nano hmC_seal&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cmparison-with-strong-tab-seq-peaks" class="section level4">
<h4>Cmparison with strong TAB-seq peaks</h4>
<p>There are 2,057,636 strong base level peaks from TAB-seq. To compute overlapping rate, each base is extend by 10bp downstream, 10bp upstream, the same length as the called peak windows.</p>
<pre class="r"><code>jump.rate.minus=c(0.3387, 0.3346, 0.3199, 0.3082, 0.2984, 0.2926, 0.2898)
jump.rate.plus=c(0.3397, 0.3354, 0.3208, 0.3094, 0.2989, 0.2919, 0.2902)
TAB.rate.minus=c(0.1577, 0.1947, 0.2424, 0.2810, 0.3152, 0.3392, 0.3713)
TAB.rate.plus=c(0.1582, 0.1950, 0.2445, 0.2813, 0.3172, 0.3407, 0.3713)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of Jump-seq&quot;, col=1, main=&quot;20bp, FDR 0.5&quot;, cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)


plot(TAB.rate.minus, ylim=c(0,1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of strong TAB-seq&quot;, col=3, main=&quot;20bp, FDR 0.5&quot;, cex.main=0.8, cex.lab=0.8)
lines(TAB.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(3,4), lty=c(1,1))
axis.labels &lt;-c(&quot;40%&quot;, &quot;50%&quot;, &quot;60%&quot;, &quot;70%&quot;, &quot;80%&quot;, &quot;90%&quot;, &quot;100%&quot;)
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)
mtext(&quot;overlap between Jump-seq &amp; strong TAB-seq&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-with-100-reads" class="section level4">
<h4>Sensitivity with 100% reads</h4>
<p>Senstivibity is defined as the proportion of nano hmc seal or strong Tab-seq that are recovered by jump seq peak windows at FDR0.05 with 100% reads here.</p>
<pre class="r"><code>ratio=matrix(nrow=2, ncol=2)
ratio[,1]=c(0.9271, 0.3713)
ratio[,2]=c(0.9273, 0.3713)
rownames(ratio)=c(&quot;Nano_seal&quot;, &quot;Strong Tab-seq&quot;)
par(mfrow=c(1,2))
barplot(ratio[,1], beside=F, ylim=c(0,1.2), col=c(&quot;red&quot;, &quot;darkblue&quot;), xlab=c(&quot;plus strand&quot;), main=&quot;&quot;, ylab=&quot;Sensitivity&quot;)
text(0.6, 1.0, ratio[1,1])
text(2, 0.45, ratio[2,1])
barplot(ratio[,2], beside=F, ylim=c(0,1.2), col=c(&quot;red&quot;, &quot;darkblue&quot;), xlab=c(&quot;minus strand&quot;), main=&quot;&quot;, ylab=&quot;Sensitivity&quot;)
text(0.6, 1.0, ratio[1,2])
text(2, 0.45, ratio[2,2])</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-12-1.png" alt="This figure shows the sensitivity of the method defined as the proportion of peak windows by other methods (Tab-seq and Nano seal ) that are recovered by our method" width="672" />
<p class="caption">
This figure shows the sensitivity of the method defined as the proportion of peak windows by other methods (Tab-seq and Nano seal ) that are recovered by our method
</p>
</div>
<div id="enrichment-100-reads-minus" class="section level5">
<h5>Enrichment (100% reads minus )</h5>
<p>To do enrichments analysis, peak windows are called with varying p values. Use FDR 0.05 as threshold to select enriched windows. Randomly pick the same number of windows as enriched windows from the whole genome as background. Perform enrichment analysis for enriched windows versus background windows when overlapping with nano hmC_seal and strong TAB-seq. FDR 0.05 is used throughout and stated clearly otherwise.</p>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(519216, 73689)
count[,2]=3789453-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;hmC_Seal&quot;, main=&quot;48ng(combine)  minus strand (FDR 0.05)&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(921479, 98580)
count[,2]=3789453-count[,1]
pval=fisher.test(count, alternative = &quot;greater&quot;)$p.value
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;strong TAB-seq&quot;, main=&quot;48ng(combine) minus strand (FDR 0.05)&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="jump-distribution" class="section level3">
<h3>Jump distribution</h3>
<p>The reference is GSM882244_FDR_0.0484. First got non-overlapping 5hmC regions and make sure each window has only one 5hmC site. For each window, count the reads depth at every site, then add up reads depth of these non-overlapping windows. Reads distribution is the accumulative reads depth at every base of all non-overlapping windows. Input Bam file is the replicate with largest number of reads.</p>
<pre class="r"><code>minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\20161115_5hmC_Jump_Seq_2hmC_spike_in.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_minus_with_1_coverage.ave.perbase&quot;, header=F)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)-1,minus.48ng[,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)

plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
mtext(&quot;20161115_5hmC_Jump_Seq_2hmC_spike_in.minus&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\20161115_5hmC_Jump_Seq_2hmC_spike_in.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_plus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)+1,plus.48ng[,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
mtext(&quot;20161115_5hmC_Jump_Seq_2hmC_spike_in.plus&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Look into jump distribution of one base nearby CpG in the center. Bam file are the combined 11 replicates.</p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_aa_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_aa_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***aC/Ga***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>aC/Ga means the center three nucleotides are aCa (plus strand) or aGa (minus strand).</li>
</ul>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ac_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ac_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***aC/Gc***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ag_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ag_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***aC/Gg***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_at_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_at_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***aC/Gt***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ca_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ca_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***cC/Ga***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_cc_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_cc_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***cC/Gc***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_cg_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_cg_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***cC/Gg***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ct_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ct_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***cC/Gt***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ga_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ga_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***gC/Ga***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gc_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gc_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***gC/Gc***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gg_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gg_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***gC/Gg***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gt_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_gt_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***gC/Gt***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ta_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_ta_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***tC/Ga***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tc_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tc_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***tC/Gc***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tg_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tg_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***tC/Gg***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tt_plus_with_1_coverage.ave.perbase&quot;, header=F)

minus.48ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\center_three_jump\\48ng_11rep_merge.sorted_GSM882244_FDR_0.0484_tt_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;plus&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;minus&quot;)
abline(v=0, col=2)
mtext(&quot;***tC/Gt***-11rep&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Put all these 16 jump distributions together and plot them.</p>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Two sample Kolmogorov-Smirnov Tests: test if any two jump distributions are from the same distribution or not. First standardize the data.</p>
<table style="width:50%;">
<colgroup>
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th>–</th>
<th>aca</th>
<th>acc</th>
<th>acg</th>
<th>act</th>
<th>cca</th>
<th>ccc</th>
<th>ccg</th>
<th>cct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>aca</td>
<td></td>
</tr>
<tr class="even">
<td>acc</td>
<td>0.403(1.3e-14)</td>
</tr>
<tr class="odd">
<td>acg</td>
<td>0.2836(1.91e-7)</td>
<td>0.393(6.5e-14)</td>
</tr>
<tr class="even">
<td>act</td>
<td>0.3781(6.6e-13)</td>
<td><strong>0.1194(0.1139)</strong></td>
<td>0.3632(6.1e-12)</td>
</tr>
<tr class="odd">
<td>cca</td>
<td><strong>0.0995(0.2727)</strong></td>
<td>0.3632(6.1e-12)</td>
<td>0.2438(1.3e-5)</td>
<td>0.3234(1.5e-9)</td>
</tr>
<tr class="even">
<td>ccc</td>
<td>0.4279(2.2e-16)</td>
<td><strong>0.0896(0.3957)</strong></td>
<td>0.398(2.98e-14)</td>
<td>0.1542(0.0168)</td>
<td>0.3582(1.3e-11)</td>
</tr>
<tr class="odd">
<td>ccg</td>
<td>0.2935(6e-8)</td>
<td>0.393(6.5e-14)</td>
<td><strong>0.0597(0.8663)</strong></td>
<td>0.3781(6.6e-13)</td>
<td>0.3532(2.6e-11)</td>
<td>0.4(2.98e-14)</td>
</tr>
<tr class="even">
<td>cct</td>
<td>0.4129(2.5e-15)</td>
<td><strong>0.1493(0.0227)</strong></td>
<td>0.4378(&lt;2.2e-16)</td>
<td><strong>0.1294(0.0693)</strong></td>
<td>0.9552(&lt;2.2e-16)</td>
<td><strong>0.1(0.22)</strong></td>
<td>0.4(&lt;2.2e-16)</td>
</tr>
<tr class="odd">
<td>gca</td>
<td>0.1791(0.0032)</td>
<td>0.3582(1.3e-11)</td>
<td>0.2687(1e-6)</td>
<td>0.3134(5.3e-9)</td>
<td>0.4726(&lt;2.2e-16)</td>
<td>0.29(6e-8)</td>
<td>0.3(1e-7)</td>
<td>0.3(3e-8)</td>
</tr>
<tr class="even">
<td>gcc</td>
<td>0.5473(2.2e-16)</td>
<td><strong>0.1592(0.0123)</strong></td>
<td>0.5522(&lt;2.2e-16)</td>
<td>0.2189(0.0001)</td>
<td>0.9552(&lt;2.2e-16)</td>
<td>0.25(7.9e-6)</td>
<td>0.6(&lt;2.2e-16)</td>
<td>0.2(0.0001)</td>
</tr>
<tr class="odd">
<td>gcg</td>
<td><strong>0.1343(0.0532)</strong></td>
<td>0.4876(&lt;2.2e-16)</td>
<td>0.2637(1.7e-6)</td>
<td>0.4279(2.2e-16)</td>
<td>0.1642(0.0089)</td>
<td>0.42(4.4e-16)</td>
<td>0.2(3e-5)</td>
<td>0.5(&lt;2.2e-16)</td>
</tr>
<tr class="even">
<td>gct</td>
<td>0.4328(&lt;2.2e-16)</td>
<td><strong>0.1144(0.1438)</strong></td>
<td>0.3881(1.4e-13)</td>
<td><strong>0.1443(0.0305)</strong></td>
<td>0.3781(6.6e-13)</td>
<td>0.18(0.002)</td>
<td>0.4(1e-13)</td>
<td>0.2(0.002)</td>
</tr>
<tr class="odd">
<td>tca</td>
<td>0.1642(0.0089)</td>
<td>0.3234(1.5e-9)</td>
<td>0.2388(2.1e-5)</td>
<td>0.2687(1e-6)</td>
<td><strong>0.15(0.02)</strong></td>
<td>0.32(1.5e-9)</td>
<td>0.2(3e-5)</td>
<td>0.3(3e-8)</td>
</tr>
<tr class="even">
<td>tcc</td>
<td>0.5224(&lt;2.2e-16)</td>
<td>0.204(0.0005)</td>
<td>0.4776(&lt;2.2e-16)</td>
<td>0.2438(1.3e-5)</td>
<td>0.47(&lt;2.2e-16)</td>
<td>0.17(0.0045)</td>
<td>0.5(&lt;2e-16)</td>
<td>0.2(3e-5)</td>
</tr>
<tr class="odd">
<td>tcg</td>
<td>0.2537(4.8e-6)</td>
<td>0.4677(&lt;2.2e-16)</td>
<td><strong>0.1443(0.0305)</strong></td>
<td>0.4279(2.2e-16)</td>
<td>0.22(0.0001)</td>
<td>0.4(2.9e-14)</td>
<td><strong>0.2(0.02)</strong></td>
<td>0.5(&lt;2e-16)</td>
</tr>
<tr class="even">
<td>tct</td>
<td>0.5025(&lt;2.2e-16)</td>
<td><strong>0.1443(0.0305)</strong></td>
<td>0.4179(1.1e-15)</td>
<td>0.1791(0.0032)</td>
<td>0.44(&lt;2.2e-16)</td>
<td>0.2(0.0002)</td>
<td>0.4(1e-15)</td>
<td>0.2(0.005)</td>
</tr>
</tbody>
</table>
<table style="width:50%;">
<caption>K-S test for pairwise jump distributions of 16 possible base compositions from combined 48ng samples of 11 replicates of plus strand.</caption>
<colgroup>
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th>–</th>
<th>gca</th>
<th>gcc</th>
<th>gcg</th>
<th>gct</th>
<th>tca</th>
<th>tcc</th>
<th>tcg</th>
<th>tct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>gca</td>
<td></td>
</tr>
<tr class="even">
<td>gcc</td>
<td>0.5(&lt;2.2e-16)</td>
</tr>
<tr class="odd">
<td>gcg</td>
<td>0.2(3e-5)</td>
<td>0.6(&lt;2.2e-16)</td>
</tr>
<tr class="even">
<td>gct</td>
<td>0.4(6e-12)</td>
<td>0.2(0.003)</td>
<td>0.5(&lt;2.2e-16)</td>
</tr>
<tr class="odd">
<td>tca</td>
<td><strong>0.1(0.2)</strong></td>
<td>0.4(4e-16)</td>
<td>0.2(0.0003)</td>
<td>0.4(1e-11)</td>
</tr>
<tr class="even">
<td>tcc</td>
<td>0.4(&lt;2.2e-16)</td>
<td>0.1(0.04)</td>
<td>0.5(&lt;2.2e-16)</td>
<td><strong>0.1(0.02)</strong></td>
<td>0.4(&lt;2.2e-16)</td>
</tr>
<tr class="odd">
<td>tcg</td>
<td>0.2(2e-5)</td>
<td>0.6(&lt;2.2e-16)</td>
<td>0.2(0.0005)</td>
<td>0.4(1e-13)</td>
<td>0.2(0.0003)</td>
<td>0.5(&lt;2.2e-16)</td>
</tr>
<tr class="even">
<td>tct</td>
<td>0.4(3e-14)</td>
<td>0.1(0.05)</td>
<td>0.6(&lt;2.2e-16)</td>
<td>0.2(0.0001)</td>
<td>0.4(3e-14)</td>
<td>0.2(0.003)</td>
<td>0.5(&lt;2.2e-16)</td>
</tr>
</tbody>
</table>
<ul>
<li>this is only for combing 11 replicates plus strand.</li>
<li>most pair whose p values are large to be not significant (&gt;0.05) have the same base after 5hmC base</li>
</ul>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /> This figure shows the ratio of reads within 20 bp window in the center with the center site being ca, cc, cg, ct, in plus strand. Obviously, over 99% reads are generated from the window with cg in the center.</p>
<p>Investigate the number of CA, CC, CG, CT’s in the peak windows.</p>
<pre class="r"><code>total=matrix(nrow=4, ncol=1)
total[,1]=c(2936943, 2318330, 2135195, 3507124)/sum(c(2936943, 2318330, 2135195, 3507124))
rownames(total)=c(&quot;CA&quot;, &quot;CC&quot;, &quot;CG&quot;, &quot;CT&quot;)
barplot(total[,1]/sum(total[,1]), ylim=c(0,0.5), main=&quot;48ng_11rep_merge.minus.FDR0.05.corrected&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="reads-depletion" class="section level3">
<h3>reads depletion</h3>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-at-cis-regulatory-regions" class="section level3">
<h3>enrichment at cis-regulatory regions</h3>
<p>Combine 11 48ng replicates together and look at plus and minus strand.</p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0070, 0.0032)
percent[,2]=c(0.0139, 0.0034)

percent[,3]=c(0.0069, 0.0026)
percent[,4]=c(0.0042, 0.0021)

percent[,5]=c(0.0005, 0.0011)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.02),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;combine 11 48ng replicates minus (FDR0.05)&quot;, xlab=&quot;&quot;, cex.names=1)
text(2, 0.0075, &quot;2.21(p&lt;2.2e-16)&quot;)
text(5, 0.0145, &quot;3.87(p&lt;2.2e-16)&quot;)
text(8, 0.0075, &quot;2.60(p&lt;2.2e-16)&quot;)
text(11, 0.005, &quot;1.99(p&lt;2.2e-16)&quot;)
text(14, 0.0015, &quot;0.46 (p=1)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0068, 0.0032)
percent[,2]=c(0.0139, 0.0034)

percent[,3]=c(0.0070, 0.0026)
percent[,4]=c(0.0042, 0.0021)

percent[,5]=c(0.0006, 0.0011)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.02),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;combine 11 48ng replicates plus (FDR0.05)&quot;, xlab=&quot;&quot;, cex.names=1)
text(2, 0.0075, &quot;2.21(p&lt;2.2e-16)&quot;)
text(5, 0.0145, &quot;3.95(p&lt;2.2e-16)&quot;)
text(8, 0.0075, &quot;2.60(p&lt;2.2e-16)&quot;)
text(11, 0.005, &quot;1.99(p&lt;2.2e-16)&quot;)
text(14, 0.0015, &quot;0.49 (p=1)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-40-1.png" alt="We combined eleven 48ng replicates, called peak windows of 20 bp, selecetd 20 bp windows with FDR 0.05. We randomly selected same amount of 20 bp windows as  the chosen windows in the whole genome. The proportion of chosen peak windows (red) and randon windows (blue) overlapping with different regions are calculated and the corresponding enrichments are shown above the red bar" width="672" />
<p class="caption">
We combined eleven 48ng replicates, called peak windows of 20 bp, selecetd 20 bp windows with FDR 0.05. We randomly selected same amount of 20 bp windows as the chosen windows in the whole genome. The proportion of chosen peak windows (red) and randon windows (blue) overlapping with different regions are calculated and the corresponding enrichments are shown above the red bar
</p>
</div>
</div>
<div id="peak-distribution" class="section level3">
<h3>peak distribution</h3>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>The peaks are from combined data of 11 48ng replicates of plus strand at FDR0.05.</p></li>
<li><p>Each peak window is counted once, start from CTCF, then enhancer, h3k27ac, h3k5me3, PoIII. For example, first count how many peak windows are overlaping with CTCF, then exclude these overlapped windows, use the rest to overlap with enhancer, and so on.</p></li>
<li><p>all these peaks overlapping with those regions only take about 3% of all peaks called.</p></li>
</ul>
<div id="section" class="section level101">
<p></p>
</div>
</div>
</div>
<div id="ng-5hmc-replicate" class="section level2">
<h2>24ng 5hmC replicate</h2>
<p>There are 5 24ng 5hmC replicates.</p>
<table>
<caption>number of reads in each replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>3,613,864</td>
<td>3,621,005</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>2,654,892</td>
<td>2,660,120</td>
</tr>
<tr class="odd">
<td>rep 3</td>
<td>4,634,559</td>
<td>4,645,750</td>
</tr>
<tr class="even">
<td>rep 4</td>
<td>1,846,242</td>
<td>1,853,346</td>
</tr>
<tr class="odd">
<td>rep 5</td>
<td>5,072,002</td>
<td>5,085,060</td>
</tr>
</tbody>
</table>
<pre class="r"><code>win.minus=c(255946, 1255527, 498618, 1474825, 555722)
win.plus=c(255426, 1262831, 499772, 1481412, 555236)
plot(win.minus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;replicates&quot;, ylab=&quot;Number of peak windows&quot;, col=1, main=&quot;Number of peak windows (20bp, FDR0.05)&quot;, cex=0.8)
lines(win.plus, col=2, type=&quot;o&quot;)
legend(1, 1500000, c(&quot;minus strand&quot;, &quot;plus strand&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1 &quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="comparison-with-dalis-peaks-1" class="section level3">
<h3>Comparison with Dali’s peaks</h3>
<pre class="r"><code>jump.rate.minus=c(0.1248, 0.0639, 0.1255, 0.0829, 0.1264)
jump.rate.plus=c(0.1254, 0.0646, 0.1258, 0.0833, 0.1265)
dali.rate.minus=c(0.3455, 0.6132, 0.5248, 0.7351, 0.5625)
dali.rate.plus=c(0.3465, 0.6181, 0.5281, 0.7390, 0.5610)
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali&#39;s peaks&quot;, cex=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(dali.rate.minus, col=3, type=&quot;o&quot;)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 1, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;dali.rate.minus&quot;, &quot;dali.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-strong-tab-seq" class="section level3">
<h3>Comparison with strong TAB-seq</h3>
<pre class="r"><code>jump.rate.minus=c(0.2610, 0.1230, 0.2865, 0.1672, 0.2730)
jump.rate.plus=c(0.2612, 0.1234, 0.2891, 0.1686, 0.2734)
TAB.rate.minus=c(0.0422, 0.0905, 0.0881, 0.1437, 0.0930)
TAB.rate.plus=c(0.0421, 0.0913, 0.0894, 0.1458, 0.0932)
plot(jump.rate.minus, ylim=c(0, 0.6), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks&quot;, cex.main=0.9)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(TAB.rate.minus, col=3, type=&quot;o&quot;)
lines(TAB.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.6, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;StrongTAB.rate.minus&quot;, &quot;StrongTAB.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-rep4-minus" class="section level3">
<h3>enrichment-rep4 minus</h3>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(122190, 29091)
count[,2]=1474825-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;hmC_Seal&quot;, main=&quot;24ng rep4 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(246529, 17073)
count[,2]=1474825-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;strong TAB-seq&quot;, main=&quot;24ng rep4 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="section-1" class="section level99">
<p></p>
</div>
</div>
</div>
<div id="ng-5hmc-replicate-1" class="section level2">
<h2>12ng 5hmC replicate</h2>
<p>There are 5 12ng 5hmC replicates.</p>
<table>
<caption>number of reads in each replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>1,538,440</td>
<td>1,543,973</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>2,817,848</td>
<td>2,829,593</td>
</tr>
<tr class="odd">
<td>rep 3</td>
<td>1,805,014</td>
<td>1,813,743</td>
</tr>
<tr class="even">
<td>rep 4</td>
<td>1,962,631</td>
<td>1,976,519</td>
</tr>
<tr class="odd">
<td>rep 5</td>
<td>1,365,146</td>
<td>1,375,074</td>
</tr>
</tbody>
</table>
<pre class="r"><code>win.minus=c(1144297, 1008422, 1436959, 1197291)
win.plus=c(1312102, 1036507, 1441863, 1200396)
plot(win.minus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;replicates&quot;, ylab=&quot;Number of peak windows&quot;, col=1, main=&quot;Number of peak windows (20bp, FDR0.05)&quot;, cex=0.8)
lines(win.plus, col=2, type=&quot;o&quot;)
legend(1, 1420000, c(&quot;minus strand&quot;, &quot;plus strand&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1 &quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,4), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="comparison-with-dalis-peaks-2" class="section level3">
<h3>Comparison with Dali’s peaks</h3>
<pre class="r"><code>jump.rate.minus=c(0.0876, 0.0665, 0.0734, 0.0933)
jump.rate.plus=c(0.0858, 0.0672, 0.0734, 0.0938)
dali.rate.minus=c(0.6385, 0.5558, 0.6989, 0.7158)
dali.rate.plus=c(0.7222, 0.5711, 0.7026, 0.7202)
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali&#39;s peaks&quot;, cex=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(dali.rate.minus, col=3, type=&quot;o&quot;)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.5, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;dali.rate.minus&quot;, &quot;dali.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,4), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-strong-tab-seq-1" class="section level3">
<h3>Comparison with strong TAB-seq</h3>
<pre class="r"><code>jump.rate.minus=c(0.1182, 0.0822, 0.0932, 0.1250)
jump.rate.plus=c(0.1180, 0.0826, 0.0933, 0.1254)
TAB.rate.minus=c(0.0776, 0.0472, 0.0765, 0.0861)
TAB.rate.plus=c(0.0888, 0.0487, 0.0767,0.0865)
plot(jump.rate.minus, ylim=c(0, 0.4), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks&quot;, cex.main=0.9)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(TAB.rate.minus, col=3, type=&quot;o&quot;)
lines(TAB.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.4, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;StrongTAB.rate.minus&quot;, &quot;StrongTAB.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,4), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-rep-5-minus-strand" class="section level3">
<h3>enrichment-rep 5 minus strand</h3>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(111734, 23171)
count[,2]=1197291-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;hmC_Seal&quot;, main=&quot;12ng rep5 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(149717, 31057)
count[,2]=1197291-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;strong TAB-seq&quot;, main=&quot;12ng rep5 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="section-2" class="section level101">
<p></p>
</div>
</div>
</div>
<div id="ng-5hmc-replicate-2" class="section level2">
<h2>6ng 5hmC replicate</h2>
<div id="reads-distribution-1" class="section level3">
<h3>reads distribution</h3>
<pre class="r"><code>reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), &quot;&gt;10&quot;); reads[,1]=c(473607, 8448, 370, 60, 30, 13, 8, 7, 5, 6, 94)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of called peak windows at FDR0.05 (rep1 minus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There are 5 6ng 5hmC replicates.</p>
<table>
<caption>number of reads in each replicate</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>506,562</td>
<td>510,127</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>900,135</td>
<td>902,501</td>
</tr>
<tr class="odd">
<td>rep 3</td>
<td>1,049,521</td>
<td>1,051,758</td>
</tr>
<tr class="even">
<td>rep 4</td>
<td>785,990</td>
<td>792,942</td>
</tr>
<tr class="odd">
<td>rep 5</td>
<td>705,891</td>
<td>713,330</td>
</tr>
</tbody>
</table>
<pre class="r"><code>win.plus=c(482648, 838387, 966623, 736000, 665408)
win.minus=c(482398, 836889, 964130, 735655, 665737)
plot(win.plus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;replicates&quot;, ylab=&quot;# windows of 20bp&quot;, col=1, main=&quot;Number of peak windows (20bp, FDR0.05)&quot;, cex=0.5)
lines(win.minus, col=2, type=&quot;o&quot;)
legend(2.5, 700000, c(&quot;plus_FDR0.05&quot;, &quot;minus_FDR0.05&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep 1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-53-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-dalis-peaks-3" class="section level3">
<h3>Comparison with Dali’s peaks</h3>
<pre class="r"><code>jump.rate.minus=c(0.0934, 0.0771, 0.0764, 0.0917, 0.0922)
jump.rate.plus=c(0.0926, 0.0773, 0.0765, 0.0913, 0.0920)
dali.rate.minus=c(0.4538, 0.5586, 0.5996, 0.5753, 0.5469)
dali.rate.plus=c(0.4523, 0.5612, 0.6015, 0.5719, 0.5444)
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali&#39;s peaks&quot;, cex=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(dali.rate.minus, col=3, type=&quot;o&quot;)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 1, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;dali.rate.minus&quot;, &quot;dali.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;rep5&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-strong-tab-seq-2" class="section level3">
<h3>Comparison with strong TAB-seq</h3>
<pre class="r"><code>jump.rate.minus=c(0.1653, 0.1413, 0.1411, 0.1720, 0.1724)
jump.rate.plus=c(0.1657, 0.1410, 0.1409, 0.1721, 0.1723)
TAB.rate.minus=c(0.0492, 0.0715, 0.0818, 0.0771, 0.0704)
TAB.rate.plus=c(0.0492, 0.0715, 0.0820, 0.0769, 0.0699)
plot(jump.rate.minus, ylim=c(0, 0.6), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;number of reads&quot;, ylab=&quot;overlapping rate&quot;, col=1, main=&quot;Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks&quot;, cex.main=0.9)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
lines(TAB.rate.minus, col=3, type=&quot;o&quot;)
lines(TAB.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.6, c(&quot;jump.rate.minus&quot;, &quot;jump.rate.plus&quot;, &quot;StrongTAB.rate.minus&quot;, &quot;StrongTAB.rate.plus&quot;), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-55-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-rep-5-minus-strand-1" class="section level3">
<h3>enrichment-rep 5 minus strand</h3>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(61404, 12907)
count[,2]=665737-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;hmC_Seal&quot;, main=&quot;6ng rep5 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>count=matrix(nrow=2, ncol=2)
count[,1]=c(114794, 17049)
count[,2]=665737-count[,1]
barplot(count[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;strong TAB-seq&quot;, main=&quot;6ng rep5 minus strand&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-57-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="section-3" class="section level101">
<p></p>
</div>
</div>
</div>
<div id="point4ng-5hmc-replicate" class="section level2">
<h2>2point4ng 5hmC replicate</h2>
<table>
<caption>number of reads in every replicate.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>270,885</td>
<td>272,807</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>165,553</td>
<td>167,277</td>
</tr>
<tr class="odd">
<td>rep 3</td>
<td>279,404</td>
<td>281,109</td>
</tr>
<tr class="even">
<td>rep 4</td>
<td>213,739</td>
<td>215,604</td>
</tr>
<tr class="odd">
<td>combine</td>
<td>929,581</td>
<td>926,797</td>
</tr>
</tbody>
</table>
<pre class="r"><code>win.plus2=c(257809, 152655, 264996, 202465, 505788) # FDR0.05 (~pvalue cutoff 0.01)
win.minus2=c(257918, 152853, 264218, 202051, 521062)
plot(win.plus2, type=&quot;o&quot;, ylim=c(150000, 530000), xaxt=&#39;n&#39;, xlab=&quot;replicates&quot;, ylab=&quot;# windows of 20bp&quot;, col=1, main=&quot;Number of peak windows&quot;, cex=0.5)
lines(win.minus2, col=2, type=&quot;o&quot;)
legend(1, 400000, c(&quot;plus_FDR0.05&quot;, &quot;minus_FDR0.05&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep 1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="reads-distribution-2" class="section level3">
<h3>Reads distribution</h3>
<pre class="r"><code>reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), &quot;&gt;10&quot;); reads.plus[,1]=c(253804, 3709, 159, 31, 16, 11, 1, 6, 7, 4, 61)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of windows with reads (rep1 plus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), &quot;&gt;10&quot;); reads.plus[,1]=c(371791, 128383, 4260, 985, 115, 67, 27, 14, 11, 10, 125)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of windows with reads (combine plus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-60-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads.minus=matrix(nrow=11, ncol=1)
rownames(reads.minus)=c(seq(1,10), &quot;&gt;10&quot;); reads.minus[,1]=c(253875, 3720, 166, 31, 21, 8, 7, 3, 2, 3, 82)
barplot(reads.minus[,1]/sum(reads.minus[,1]), ylim=c(0,1), ylab=&quot;Freq&quot;, main=&quot;Frequency of called peak windows at FDR0.05  (rep1 minus)&quot;, xlab=&quot;Number of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-61-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-dalis-peaks-4" class="section level3">
<h3>Comparison with Dali’s peaks</h3>
<pre class="r"><code>jump.rate.minus=c(0.0854, 0.0711, 0.0830, 0.0809, 0.0829)
jump.rate.plus=c(0.0852, 0.0713, 0.0818, 0.0793, 0.0822)
dali.rate.minus=c(0.2711, 0.1481, 0.2712, 0.2137, 0.4154)
dali.rate.plus=c(0.2692, 0.1490, 0.2674, 0.2093, 0.4014)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of Jump-seq&quot;, col=1, main=&quot;20bp, FDR&gt;0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of nano hmC_seal&quot;, col=3, main=&quot;20bp, FDR&gt;0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(3,4), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)
mtext(&quot;overlap between Jump-seq &amp; nano hmC_seal&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combine.minus=c(0.0818, 0.0840, 0.1435, 0.1059, 0.1000, 0.0641, 0.0606, 0.0588, 0, 0, 0)
plot(seq(1,11), combine.minus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;Number of reads&quot;, ylim=c(0, 0.2), ylab=&quot;Proportion of Jump-seq (combine minus)&quot;)
axis.labels &lt;-c(seq(1,10), &quot;&gt;10&quot;)
abline(h=0.0194, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-63-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>This is the enrichment of peak windows at different reads counts for combined 2.4ng replicates.</li>
<li>This is not a good measure since the number of widnows with large number of reads usually is small, say 2 windows, the overlapping rate with nano hmc_seal is not so reliable, (with large variance).</li>
<li>red line is background rate, 1.94%.</li>
</ul>
<pre class="r"><code>combine.plus=c(0.0814, 0.0828, 0.1254, 0.1289, 0.1478, 0.1045, 0.0741, 0, 0.0909, 0, 0.0080)
plot(seq(1,11), combine.plus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;Number of reads&quot;, ylim=c(0, 0.2), ylab=&quot;Proportion of Jump-seq (combine plus)&quot;)
axis.labels &lt;-c(seq(1,10), &quot;&gt;10&quot;)
abline(h=0.0194, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-64-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-with-strong-tab-seq-3" class="section level3">
<h3>Comparison with strong TAB-seq</h3>
<pre class="r"><code>jump.rate.minus=c(0.1543, 0.1295, 0.1544, 0.1483, 0.1489)
jump.rate.plus=c(0.1564, 0.1292, 0.1547, 0.1491, 0.1496)
TAB.rate.minus=c(0.0246, 0.0122, 0.0253, 0.0186, 0.0477)
TAB.rate.plus=c(0.0249, 0.0122, 0.0253, 0.0187, 0.0462)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of Jump-seq&quot;, col=1, main=&quot;20bp, FDR&gt;0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(1,2), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)


plot(TAB.rate.minus, ylim=c(0,1), type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;&quot;, ylab=&quot;Proportion of TAB-seq&quot;, col=3, main=&quot;20bp, FDR&gt;0.05&quot;, cex.main=0.8, cex.lab=0.8)
lines(TAB.rate.plus, col=4, type=&quot;o&quot;)
legend(1, 0.7, c(&quot;minus&quot;, &quot;plus&quot;), col=c(3,4), lty=c(1,1))
axis.labels &lt;-c(&quot;rep1&quot;, &quot;rep2&quot;, &quot;rep3&quot;, &quot;rep4&quot;, &quot;combine&quot;)
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)
mtext(&quot;overlap between Jump-seq &amp; TAB-seq&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combine.minus=c(0.1462, 0.1519, 0.2660, 0.2118, 0.2357, 0.0769, 0.2121, 0.1176, 0.0769, 0.3333, 0.2349)
plot(seq(1,11), combine.minus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;Number of reads&quot;, ylim=c(0, 0.4), ylab=&quot;Proportion of Jump-seq (combine minus)&quot;)
axis.labels &lt;-c(seq(1,10), &quot;&gt;10&quot;)
abline(h=0.0258, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-66-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combine.plus=c(0.1466, 0.1532, 0.2643, 0.2609, 0.2174, 0.1642, 0.1481, 0.3571, 0.1818, 0.1000, 0.2960)
plot(seq(1,11), combine.plus, type=&quot;o&quot;, xaxt=&#39;n&#39;, xlab=&quot;Number of reads&quot;, ylim=c(0, 0.4), ylab=&quot;Proportion of Jump-seq (combine plus)&quot;)
axis.labels &lt;-c(seq(1,10), &quot;&gt;10&quot;)
abline(h=0.0258, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-2.4ng-combine-minus-strand" class="section level3">
<h3>enrichment-2.4ng combine minus strand</h3>
<pre class="r"><code>count1=matrix(nrow=2, ncol=2)
count1[,1]=c(43213, 10092)
count1[,2]=521062-count1[,1]
barplot(count1[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;hmC_Seal&quot;, main=&quot;2.4ng (combine)&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>count2=matrix(nrow=2,ncol=2)
count2[,1]=c(98049, 13382)
count2[,2]=521062-count2[,1]
barplot(count2[,1], beside=F, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Observed&quot;, &quot;Expected&quot;), xlab=&quot;strong TAB-seq&quot;, main=&quot;2.4ng (combine)&quot;, ylab=&quot;Number of peak windows&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-69-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="jump-distribution-2.4ng" class="section level3">
<h3>jump distribution-2.4ng</h3>
<pre class="r"><code>minus.2.4ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\CHe-lu-2.4ng_mixed_with_other_library.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_minus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)-1,minus.2.4ng[,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.2.4ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
mtext(&quot;CHe-lu-2.4ng_mixed_with_other_library.minus&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-70-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plus.2.4ng=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\CHe-lu-2.4ng_mixed_with_other_library.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_plus_with_1_coverage.ave.perbase&quot;, header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)+1,plus.2.4ng[,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
plot(seq(-10, 10, by=1)+1,plus.2.4ng[91:111,1], type=&quot;l&quot;, ylab=&quot;coverage&quot;, xlab=&quot;Distance&quot;, main=&quot;&quot;)
abline(v=0, col=2)
mtext(&quot;CHe-lu-2.4ng_mixed_with_other_library.plus&quot;, outer = TRUE, cex = 1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-71-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-at-cis-regulatory-regions-1" class="section level3">
<h3>enrichment at cis-regulatory regions</h3>
<p>download cis-regulatory peaks at <a href="http://chromosome.sdsc.edu/mouse/download.html">Ren lab</a>. They are sinlge base peaks and extended to 200 bp windows.</p>
<pre class="r"><code>num.base=matrix(nrow=5, ncol=1)
rownames(num.base)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;); num.base[,1]=c(38714, 42477, 32109, 25911, 13398)
barplot(num.base[,1]/sum(num.base[,1]), ylab=&quot;Freq&quot;, main=&quot; The ratio of peaks in different regions&quot;, xlab=&quot;&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-72-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0068, 0.0032)
percent[,2]=c(0.0082, 0.0034)

percent[,3]=c(0.0048, 0.0026)
percent[,4]=c(0.0028, 0.0021)

percent[,5]=c(0.0064, 0.0011)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.01),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;combine minus (FDR0.05)&quot;, xlab=&quot;&quot;, cex.names=1)
text(2.1, 0.007, &quot;2.11(p=1.99e-13)&quot;)
text(5, 0.0085, &quot;2.40(p&lt;2.2e-16)&quot;)
text(8, 0.005, &quot;1.87(p=9.37e-8)&quot;)
text(11, 0.003, &quot;1.33(p=0.0325)&quot;)
text(14, 0.0068, &quot;5.77 (p&lt;2.2e-16)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-73-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0072, 0.0032)
percent[,2]=c(0.0079, 0.0034)

percent[,3]=c(0.0039, 0.0026)
percent[,4]=c(0.0020, 0.0021)

percent[,5]=c(0.0057, 0.0011)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.01),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;combine plus (FDR0.05)&quot;, xlab=&quot;&quot;, cex.names=1)
text(2, 0.0075, &quot;2.21(p=8.17e-13)&quot;)
text(4, 0.0085, &quot;2.32(p=2.8e-15)&quot;)
text(8, 0.005, &quot;1.50 (p=0.0029)&quot;)
text(11, 0.003, &quot;0.96(p=0.6208)&quot;)
text(14, 0.006, &quot;5.17 (p&lt;2.2e-16)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-74-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0053, 0.0032)
percent[,2]=c(0.0084, 0.0035)

percent[,3]=c(0.0053, 0.0026)
percent[,4]=c(0.0036, 0.0021)

percent[,5]=c(0.0010, 0.0012)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.01),
ylab=&quot;Proportion of Jump-seq windows&quot;, xlab=&quot;&quot;, cex.names=1, main=&quot;combine minus (one reads window)&quot;)
# do one side &quot;greater&quot; fisher exact test 
text(2, 0.006, &quot;1.67(p&lt;2.2e-16)&quot;)
text(4, 0.009, &quot;2.40(p&lt;2.2e-16)&quot;)
text(8, 0.006, &quot;2.02 (p&lt;2.2e-16)&quot;)
text(11, 0.004, &quot;1.71(p&lt;2.2e-16)&quot;)
text(14, 0.0015, &quot;0.89 (p=0.9776)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-75-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0254, 0.0137)
percent[,2]=c(0.0829, 0.0160)

percent[,3]=c(0.0186, 0.0117)
percent[,4]=c(0.0486, 0.0100)

percent[,5]=c(0.0027, 0.0049)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Nano_hmC seal peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.1),
ylab=&quot;Proportion of Nano_hmC seal peaks&quot;, main=&quot;&quot;, xlab=&quot;&quot;, cex.names=1)
# do one side &quot;greater&quot; fisher exact test 
text(2, 0.04, &quot;1.85(p&lt;2.2e-16)&quot;)
text(4, 0.09, &quot;5.18(p&lt;2.2e-16)&quot;)
text(8, 0.03, &quot;1.59 (p&lt;2.2e-16)&quot;)
text(11, 0.06, &quot;4.86(p&lt;2.2e-16)&quot;)
text(14, 0.01, &quot;0.55 (p=1)&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-76-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="peak-distribution-1" class="section level3">
<h3>peak distribution</h3>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-77-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>peak windows in all these regions take &lt;3% of all peaks called.</li>
</ul>
</div>
<div id="reads-depletion-1" class="section level3">
<h3>reads depletion</h3>
<p>Reads distribution of 200bp window with CTCF at the center.<br />
<img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-78-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-79-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="section-4" class="section level100">
<p></p>
</div>
</div>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<div id="overlap-pattern-at-different-genomic-concentration" class="section level3">
<h3>Overlap pattern at different genomic concentration</h3>
<ul>
<li>Jump-seq overlaps with Dali’s peaks</li>
</ul>
<p>Horizontal line is genome-wide overlap rate of 0.0203.</p>
<pre class="r"><code>jump.rate=c(0.0829, 0.0922, 0.0933, 0.1264, 0.1397)
dali.rate=c(0.4154, 0.5469, 0.7158, 0.5625, 0.9189)
plot(jump.rate, type=&quot;o&quot;, ylim=c(0,1), xaxt=&#39;n&#39;, xlab=&quot;samples&quot;, ylab=&quot;overlap rate&quot;, col=1, main=&quot;&quot;, cex=0.8)
lines(dali.rate, col=2, type=&quot;o&quot;)
abline(h=0.0203, col=3)
legend(1, 1, c(&quot;Jump-seq&quot;, &quot;Dali&#39;s peak&quot;, &quot;background rate&quot;), col=c(1,2, 3), lty=c(1,1, 1))
axis.labels &lt;-c(&quot;2.4ng combine &quot;, &quot;6ng&quot;, &quot;12ng&quot;, &quot;24ng&quot;, &quot;48ng(combine)&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-80-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Jump-seq overlaps with strong TAB-seq</li>
</ul>
<p>Horizontal line is the genome-wide overlap rate of 0.0258.</p>
<pre class="r"><code>jump.rate=c(0.1489, 0.1724, 0.1250, 0.2730, 0.2432)
TAB.rate=c(0.0477, 0.0704, 0.0861, 0.0930, 0.4693)
plot(jump.rate, type=&quot;o&quot;, ylim=c(0,0.5), xaxt=&#39;n&#39;, xlab=&quot;samples&quot;, ylab=&quot;overlap rate&quot;, col=1, main=&quot;&quot;, cex=0.8)
lines(TAB.rate, col=2, type=&quot;o&quot;)
abline(h=0.0260, col=3) # background rate
legend(1, 0.5, c(&quot;Jump-seq&quot;, &quot;strong TAB-seq&quot;, &quot;background rate&quot;), col=c(1,2,3), lty=c(1,1, 1))
axis.labels &lt;-c(&quot;2.4ng combine &quot;, &quot;6ng&quot;, &quot;12ng&quot;, &quot;24ng&quot;, &quot;48ng(combine)&quot;)
axis(1, at=seq(1,5), labels=axis.labels)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-81-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichemnt-analysis" class="section level3">
<h3>Enrichemnt analysis</h3>
<pre class="r"><code>percent=matrix(nrow=2, ncol=10)
percent[,1]=c(43213, 10092)/521062
percent[,2]=c(98049, 13382)/521062

percent[,3]=c(61404, 12907)/665737
percent[,4]=c(114794, 17049)/665737

percent[,5]=c(111734, 23171)/1197291
percent[,6]=c(149717, 31057)/1197291

percent[,7]=c(122190, 28799)/1474825
percent[,8]=c(246529, 37851)/1474825

percent[,9]=c(529216, 73689)/3789453
percent[,10]=c(921479, 98580)/3789453
colnames(percent)=rep(c(&quot;2.4ng&quot;, &quot;6ng&quot;, &quot;12ng&quot;, &quot;24ng&quot;, &quot;48ng&quot;), each=2)

par(mfrow=c(1,2))
barplot(percent[,c(1,3,5,7,9)], beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.3),
xlab=&quot;Figure 1A&quot;, main=&quot;&quot;, ylab=&quot;Proportion of Nano-seal peaks&quot;, cex.names=0.5)

barplot(percent[,c(2,4,6,8,10)], beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.4),
xlab=&quot;Figure 1B&quot;, main=&quot;&quot;, ylab=&quot;Proportion of Tab-seq peaks&quot;, cex.names=0.5)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-82-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ng-5hmc-replicate-3" class="section level2">
<h2>0.6ng 5hmC replicate</h2>
<div id="reads-information-and-peak-windows" class="section level3">
<h3>reads information and peak windows</h3>
<table>
<caption>number of reads in every replicate.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>22,997</td>
<td>23,664</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>29,836</td>
<td>29,848</td>
</tr>
</tbody>
</table>
<table>
<caption>number of peak windows with at least one reads in every replicate.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>20,511</td>
<td>21,188</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>27,154</td>
<td>27,392</td>
</tr>
</tbody>
</table>
<div class="figure">
<img src="figure/5hmC/0.6ng.onereplicate.png" alt="Whole genome reads distribution of one replicate-plus+minus" />
<p class="caption">Whole genome reads distribution of one replicate-plus+minus</p>
</div>
</div>
<div id="overlap-with-nano-seal-sites" class="section level3">
<h3>overlap with Nano seal sites</h3>
<table>
<caption>Ratio of jump-seq windows with at least one reads overlapping with Nano seal peaks. The background rate is 0.0194.</caption>
<thead>
<tr class="header">
<th>rep</th>
<th>plus</th>
<th>minus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rep 1</td>
<td>0.0582</td>
<td>0.0558</td>
</tr>
<tr class="even">
<td>rep 2</td>
<td>0.0760</td>
<td>0.0752</td>
</tr>
</tbody>
</table>
</div>
<div id="enrichment-in-cis-regulatory-regions" class="section level3">
<h3>Enrichment in cis-regulatory regions</h3>
<pre class="r"><code>percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0049, 0.0032)
percent[,2]=c(0.0056, 0.0035)

percent[,3]=c(0.0033, 0.0026)
percent[,4]=c(0.0021, 0.0021)

percent[,5]=c(0.0034, 0.0012)
colnames(percent)=c(&quot;ctcf&quot;, &quot;enhancer&quot;, &quot;h3k27ac&quot;, &quot;h3k4me3&quot;, &quot;polII&quot;)

barplot(percent, beside=T, col=c(&quot;red&quot;, &quot;darkblue&quot;), legend=c(&quot;Jump-seq peaks&quot;, &quot;Random windows&quot;), ylim=c(0,0.01),
ylab=&quot;Proportion of Jump-seq windows&quot;, main=&quot;20160811_5hmC_Jump_Seq_0.6ng.minus&quot;, xlab=&quot;&quot;, cex.names=1)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-83-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>cis-regulatory regions are extended 200bp windows.</li>
<li>jump peaks are 20 bp windows with at least one reads.</li>
</ul>
</div>
</div>
<div id="hmc-spike-in" class="section level2">
<h2>5hmC spike in</h2>
<div id="number-of-reads-versus-5hmc-percentage-only-consider-unique-reads" class="section level3">
<h3>number of reads versus 5hmC percentage-only consider unique reads</h3>
<pre class="r"><code>num.reads.plus=c(61720, 49782, 57708,  140773, 97274, 146991, 155893, 156222)
num.reads.minus=c(256, 117, 82, 290, 197, 339, 390, 301)
num.reads=num.reads.plus+num.reads.minus
hmc.ratio=c(0, 0.01, 0.05, 0.1, 0.25, 0.5, 0.75, 1)
plot(hmc.ratio, num.reads, type=&quot;o&quot;, xlab=&quot;5hmC percentage&quot;, ylab=&quot;Number of reads in spike in&quot;, main=&quot;20170531-200 bp spike in sequence&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-84-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>num.reads.plus=c(51218, 42860, 52078, 131192, 91995, 141295, 149531, 151531)
num.reads.minus=c(264, 98, 96, 267, 184, 335, 424, 286)
num.reads=num.reads.plus+num.reads.minus
hmc.ratio=c(0, 0.01, 0.05, 0.1, 0.25, 0.5, 0.75, 1)
plot(hmc.ratio, num.reads, type=&quot;o&quot;, xlab=&quot;5hmC percentage&quot;, ylab=&quot;Number of reads in spike in&quot;, main=&quot;20170605-200 bp spike in sequence&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-85-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Only focus on reads from 96 base to 102 base.</p>
<pre class="r"><code>num.reads.plus1=c(6394, 8485, 11180,  22850, 18627, 21870, 22936, 23313)
num.reads.plus2=c(5392, 7629, 10587, 21930, 18126, 21722, 22765, 23269)
hmc.ratio=c(0, 0.01, 0.05, 0.1, 0.25, 0.5, 0.75, 1)
plot(hmc.ratio, num.reads.plus1, type=&quot;o&quot;, xlab=&quot;5hmC percentage&quot;, ylab=&quot;Number of reads in spike in&quot;, main=&quot;reads in base &gt;=96 &amp; &lt;=102&quot;, col=1, ylim=c(5700, 26500))
lines(hmc.ratio, num.reads.plus2, col=2, type=&quot;o&quot;)
legend(0.8, 20000, c(&quot;plus1&quot;, &quot;plus2&quot;), col=c(1,2), lty=c(1,1))</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-86-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spike_in_zero=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20170531\\CHe-LH-lu-1_S1_L006_R1_001.160811_5hmc_spikein.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_160811_5hmc_spikein_plus_with_1_coverage.ave.perbase&quot;, header=F)[[1]]
par(mfrow=c(1,2))
plot(seq(1,200)+1, spike_in_zero, type=&quot;l&quot;, main=&quot;0% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)
plot(seq(96, 102), spike_in_zero[97:103], type=&quot;o&quot;, main=&quot;0% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-87-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spike_in_zero=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20170531\\CHe-LH-lu-8_S8_L006_R1_001.160811_5hmc_spikein.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_160811_5hmc_spikein_plus_with_1_coverage.ave.perbase&quot;, header=F)[[1]]
par(mfrow=c(1,2))
plot(seq(1,200)+1, spike_in_zero, type=&quot;l&quot;, main=&quot;100% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)
plot(seq(96, 102), spike_in_zero[97:103], type=&quot;o&quot;, main=&quot;100% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-88-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># there is one base shift </code></pre>
</div>
<div id="allow-multiple-reads" class="section level3">
<h3>allow multiple reads</h3>
<p>Only focus on reads from 96 base to 102 base.</p>
<pre class="r"><code>full.num.reads.plus1=c(34536, 103817, 282178,  1398644, 1374788, 2427158, 3002380, 3198060)
full.num.reads.plus2=c(21389, 84709, 241444, 1184547, 1171380, 2109440, 2578213, 2757798)

num.reads.plus1=c(34536, 103817, 282178, 1374788, 2427158, 3002380, 3198060)
num.reads.plus2=c(21389, 84709, 241444,  1171380, 2109440, 2578213, 2757798)
hmc.ratio=c(0, 0.01, 0.05,  0.25, 0.5, 0.75, 1)
plot(hmc.ratio, num.reads.plus1, type=&quot;o&quot;, xlab=&quot;5hmC percentage&quot;, ylab=&quot;Number of reads in spike in&quot;, main=&quot;reads in base &gt;=96 &amp; &lt;=102&quot;, col=1, ylim=c(21000, 3200000), xaxt=&quot;n&quot;)
axis(1, at =hmc.ratio, las=0, cex.axis=0.5)
lines(hmc.ratio, num.reads.plus2, col=2, type=&quot;o&quot;)
legend(0.8, 1000000, c(&quot;plus1&quot;, &quot;plus2&quot;), col=c(1,2), lty=c(1,1))</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-89-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-90-1.png" alt="The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in" width="672" />
<p class="caption">
The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in
</p>
</div>
<p>Fold change can be &gt;100.</p>
<pre class="r"><code>spike_in_zero=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20170605\\CHe-LH-lu-1_S1_L007_R1_001.160811_5hmc_spikein.umi_encoded_adaptor_removed_no_mismatch.sorted_160811_5hmc_spikein_plus_with_1_coverage.ave.perbase&quot;, header=F)[[1]]
par(mfrow=c(1,2))
plot(seq(1,200)+1, spike_in_zero, type=&quot;l&quot;, main=&quot;0% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)
plot(seq(96, 102), spike_in_zero[97:103], type=&quot;o&quot;, main=&quot;0% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-91-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spike_in_zero=read.table(&quot;D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20170605\\CHe-LH-lu-8_S8_L007_R1_001.160811_5hmc_spikein.umi_encoded_adaptor_removed_no_mismatch.sorted_160811_5hmc_spikein_plus_with_1_coverage.ave.perbase&quot;, header=F)[[1]]
par(mfrow=c(1,2))
plot(seq(1,200)+1, spike_in_zero, type=&quot;l&quot;, main=&quot;100% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)
plot(seq(96, 102), spike_in_zero[97:103], type=&quot;o&quot;, main=&quot;100% hmC model&quot;, xlab=&quot;Base position&quot;, ylab=&quot;Num of reads&quot;)</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-92-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># there is one base shift </code></pre>
</div>
</div>
<div id="ngmes-correlation" class="section level2">
<h2>0.5ngmES-correlation</h2>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-93-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>With each window length, use the the windows with at least one reads, calculate the pairwise correlation between two replicates using reads number in the window, then average the correlation across all possible pairwise combinations (here only randomly pick 10 pairs).</li>
</ul>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-94-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-95-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.6931962</code></pre>
<pre><code>Warning: NAs introduced by coercion

Warning: NAs introduced by coercion</code></pre>
<pre><code>[1] 0.9516026</code></pre>
<p><img src="figure/JumpSeq-analysis.Rmd/unnamed-chunk-96-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.4.3  backports_1.0.5 magrittr_1.5    rprojroot_1.3-1
 [5] tools_3.4.3     htmltools_0.3.5 yaml_2.1.16     Rcpp_0.12.10   
 [9] stringi_1.1.5   rmarkdown_1.8   highr_0.6       knitr_1.17     
[13] git2r_0.21.0    stringr_1.2.0   digest_0.6.12   evaluate_0.10  </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
