---
title: "5mC data analysis"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
fig_caption: yes
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->
Samples are [here][allsample]: and original samples are [here][orighere]  

[allsample]: https://docs.google.com/spreadsheets/d/1FHkVSgTCJDY19VcFg2goat4rXJfxofCn3CFRV-gCbfo/edit#gid=271045416
[orighere]: https://docs.google.com/spreadsheets/d/1WesFCQTrFplnXRKCHmhqtm1529DHc5Sc2tX7GkRa_-U/edit#gid=1664366512


## 48ng 5mC replicates 
There are two replicates. 

  


### reads distribution 


rep | minus | plus
----|-------|----------
rep1| 4,270,565|4,246,992
rep2|5,707,602|5,686,410
Table: number of reads in every replicate

Peak windows with reads greater than 10 are put together into category ">10". 
```{r, echo=T}
reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), ">10"); reads[,1]=c(3338973, 312411, 45971, 8205, 1741, 435, 202, 83, 52, 45, 393)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of peak windows (rep 1 minus)", xlab="Number of reads")
```




```{r, echo=T}
reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), ">10"); reads[,1]=c(4066783, 506148, 99300, 22528, 5818, 1551, 563, 215, 130, 62, 639)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of peak windows (rep 2 plus)", xlab="Number of reads")
```

rep | minus | plus
----|-------|----------
rep1| 33,312|34,307
rep2|94,687|94,320
Table: number of peak windows at FDR 0.05 (~ pvalue $1.2\times 10^{-5}$) in every replicate


rep | minus | plus
----|-------|----------
rep1| 56,390|57,010
rep2|134,018|134,260
Table: number of peak windows at FDR 0.1 (~ pvalue $3.5\times 10^{-5}$) in every replicate

rep | minus | plus
----|-------|----------
rep1| 181,825|188,726
rep2|188,726|431,088
Table: number of peak windows at FDR 0.5 (~ pvalue $1.5\times 10^{-3}$) in every replicate


#### use p value as cutoffs 
rep | minus | plus
----|-------|----------
rep1| 237,674|238,205
rep2|356,244|358,641
Table: number of peak windows with  pvalue less than $10^{-3}$ in every replicate


rep | minus | plus
----|-------|----------
rep1| 13,341|13,559
rep2|29,038|28,901
Table: number of peak windows with  pvalue less than $10^{-6}$ in every replicate

### Overlap with other bisulfite peaks  
 download bisulfite peaks at this [site][site]. Use the data _GSM1173118_E14-d0.CPG.methylPattern.txt_ (denoted as E14-d0). It has a total of 3605387 CG sites, with varying methylation levels between 0 and 1. Use 0.9 as threshold to choose CG sites as peaks, which takes 64.55%.    
 
 [site]:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48229
 


rep | minus | plus
----|-------|----------
rep1| 0.0250|0.0254
rep2|0.0216|0.0221
Table: Proportion of 5mC data (FDR0.05) overlapping with E14_d0 and the background rate is 0.0157. 



Got methylated cytosines in Tab-seq paper from Gary Hon. It has 35,452,294 sites, including plus strand and minus strand. Extending the site by one base upstream and one base downstream to form 3 bp windows. All 2,057,636 strong Tab-seq sites are included in  35,452,294 methylated cytosines.    

rep | minus | plus
----|-------|----------
rep1| 0.8470|0.8520
rep2|0.8597|0.8618
Table: Proportion of 5mC data (FDR0.05) overlapping with methylated cytosines and the background rate is 0.1450. 


rep | minus | plus
----|-------|----------
rep1| 0.7890|0.7882
rep2|0.8023|0.8034
Table: Proportion of 5mC data (p value less than $10^{-3}$) overlapping with methylated cytosines and the background rate is 0.1450. 

rep | minus | plus
----|-------|----------
rep1| 0.8374|0.8403
rep2|0.8598|0.8584
Table: Proportion of 5mC data (p value less than $10^{-6}$) overlapping with methylated cytosines and the background rate is 0.1450.

## 24ng 5mC replicates 

### reads distribution 

rep | minus | plus
----|-------|----------
rep1| 4,479,619|4,456,960
rep2|1,848,313|1,833,769
Table: number of reads in every replicate


### overlap with 5mC peaks 
rep | minus | plus
----|-------|----------
rep1| 0.8511|0.8482
rep2|0.8441|0.8651
Table: Proportion of 5mC data (p value less than $10^{-6}$) overlapping with methylated cytosines and the background rate is 0.1450.

## 12ng 5mC replicates 

### reads distribution 

rep | minus | plus
----|-------|----------
rep1| 2,954,172|2,927,975
rep2|1,006,187|986,516
Table: number of reads in every replicate


### overlap with 5mC peaks 
rep | minus | plus
----|-------|----------
rep1| 0.8051|0.8356
rep2|0.8160|0.8403
Table: Proportion of 5mC data (p value less than $10^{-6}$) overlapping with methylated cytosines and the background rate is 0.1450.

## 6ng 5mC replicates 
### reads distribution 

rep | minus | plus
----|-------|----------
rep1| 2,049,655|2,025,183
rep2|330,093|845,322
Table: number of reads in every replicate


### overlap with 5mC peaks 
rep | minus | plus
----|-------|----------
rep1| 0.8069|0.8473
rep2|0.7305|0.8704
Table: Proportion of 5mC data (p value less than $10^{-6}$) overlapping with methylated cytosines and the background rate is 0.1450.

## 2point4ng 5mC replicates

### reads distribution 

rep | minus | plus
----|-------|----------
rep1| 562,794|550,530
rep2|231,751|224,961
Table: number of reads in every replicate


rep | minus | plus
----|-------|----------
rep1| 494| 552
rep2|345|469
Table: number of peak windows at FDR 0.05 in every replicate. 

### Overlap with other bisulfite peaks

#### overlap with CpG's from Gary Hon.  

rep | minus | plus
----|-------|----------
rep1| 0.7895|0.7826
rep2|0.6174|0.6397
Table: Proportion of 5mC data (FDR0.05) overlapping with methylated cytosines and the background rate is 0.1450. 


## Summary 

### Enrichment 

```{r, echo=T}
percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.7895, 0.1450)
percent[,2]=c(0.8069, 0.1450)

percent[,3]=c(0.8051, 0.1450)
percent[,4]=c(0.8511, 0.1450)

percent[,5]=c(0.8374, 0.1450)
colnames(percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")

barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,1.2),
ylab="Proportion of Jump-seq windows", main="Enrichment with methylated cytosine from Gary Hon (Data:rep 1 minus)", xlab="", cex.names=0.7)
```

```{r, echo=F, fig.cap="Enrichment of methylated cytosine from Gary Hon  at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. "}
minus.percent=matrix(nrow=2, ncol=5)
minus.percent[,1]=c(0.7895, 0.1450)
minus.percent[,2]=c(0.8069, 0.1450)
minus.percent[,3]=c(0.8051, 0.1450)
minus.percent[,4]=c(0.8511, 0.1450)
minus.percent[,5]=c(0.8374, 0.1450)
colnames(minus.percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")

plus.percent=matrix(nrow=2, ncol=5)
plus.percent[,1]=c(0.7826, 0.1450)
plus.percent[,2]=c(0.8473, 0.1450)
plus.percent[,3]=c(0.8355, 0.1450)
plus.percent[,4]=c(0.8482, 0.1450)
plus.percent[,5]=c(0.8403, 0.1450)
colnames(plus.percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")
percent=(plus.percent+minus.percent)/2
#pdf("D:\\Dropbox\\JumpSeqProject\\Figure\\Enrichemnt_5mC.pdf")
barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,1.2),
ylab="Proportion of Jump-seq windows", main="", xlab="", cex.names=0.7)
#dev.off()
```


* cutoff used for peak windows: 2.4ng (FDR0.05), others use p values less than $10^{-6}$. 

### Sensitivity 
```{r sensitivity, echo=F, fig.cap="Proportion of  methylated cytosine from Gary Hon  at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng.}
percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.000026, 0.000029)
percent[,2]=c(0.000195, 0.000334)
percent[,3]=c(0.000313, 0.000553)
percent[,4]=c(0.000503, 0.000704)
percent[,5]=c(0.000776, 0.000786)
colnames(percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")
col.mean=colMeans(percent)
barplot(col.mean, col="darkblue", ylab="Proportion of Tab-seq methylated cytosine")
```


## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
