---
title: "FinalAnalysis"
author: "Shengtong"
date: "July 25, 2018"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE}
knitr::read_chunk("chunks.R")
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
set.seed(123)
```



## Proof of  concept
`


### Accuracy of Jump-seq assessed with synthetic DNA

```{r, echo=F, fig.cap="Fig 1.1, Number of reads (in million on y axis) at different sites, 5hmC (red) and unmodified C (blue), whose numbers are in brackets."}
CPM=matrix(nrow=1, ncol=2)
CPM[1,]=c(22, 0.78)
colnames(CPM)=c("5hmC", "Unmodified C")
barplot(CPM, col=c("red", "blue"), ylim=c(0, 24), ylab="Number of reads (million)", beside=T)
text(1.5, 23, "22.00 (1)")
text(3.5, 2, "0.78 (72)")
```




###  Quantification of 5hmC: linear correlation of read counts vs. 5hmC amount


```{r, echo=F, fig.cap=" Fig 1.2: The fitted regression line (red) on the fold change of number of reads (point) at different hmC proportions in spike in"}
hmc.ratio=c(0, 0.01, 0.05,  0.25, 0.5, 0.75, 1)
num.reads.plus1=c(34536, 103817, 282178, 1374788, 2427158, 3002380, 3198060)

fit=lm(num.reads.plus1/num.reads.plus1[1] ~ hmc.ratio)

#pdf("D:\\Dropbox\\JumpSeqProject\\Figure\\spike.in.regression.pdf")
plot(hmc.ratio, num.reads.plus1/num.reads.plus1[1], xlim=c(0,1), ylim=c(0, 110), type="p", ylab="Fold change of reads count", xlab="Proportion of hmC", xaxt="n", main="", pch=16)
axis(1, at =hmc.ratio, las=0, cex.axis=0.5)
lines(hmc.ratio, fit$coefficients[1]+fit$coefficients[2]*hmc.ratio, col=2, type="l")
r.square=summary(fit)$r.squared
legend(0.01,100, expression(paste(R^2, "=0.94")))
#dev.off()

#legend(0.75, 40, c("observed", "fitted"), col=c(1,2), lty=c(1,1))
```



### Visualization 
Figure: IGV view, show peaks at different DNA levels, also show peaks from Tab-seq or Nano-seal. See Figure 1B of Nano-seal paper. 


### Spatial distribution 


The reference is GSM882244_FDR_0.0484. First got non-overlapping 5hmC regions and make sure each window has only one 5hmC site. For each window, count the reads depth at every site, then add up reads depth of these non-overlapping windows. Reads distribution is the accumulative reads depth at every base of all non-overlapping windows. Input Bam file is the replicate with largest number of reads.  



```{r, echo=F}
plus.48ng=read.table("../data/20161115_5hmC_Jump_Seq_2hmC_spike_in.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_plus_with_1_coverage.ave.perbase", header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)+1,plus.48ng[,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
mtext("20161115_5hmC_Jump_Seq_2hmC_spike_in.plus", outer = TRUE, cex = 1)

```




## Self-quantification
```{r, eval=F, echo=F}
#Correlation: 0.6 ng - 48 ng. Both 5mC and 5hmC. 
#Comparison across replicates and different DNA levels
#Figure: heat-map to summarize the results. See Figure S1C. 
#Estimate precision and sensitivity of 0.6 ng vs. 48 ng. 
#Figure: barplots, show precision and sens. at different DNA levels (using sites at 48ng as true sites). 
```

### corrleation among replicates 

```{r, echo=F, fig.cap="Reads correlations in 1kb windows between two 5hmC replicates. log2 transformation in calculation of correlations  of number of reads", warning=F, cache=T}
#pdf("../../Figure/correlation_two_5hmc.pdf")
reads=read.table("../../5hmC/mm9.genome.1kb.window.bed.CHe-Lu-1_S12_L005_R1_001.plus.21.bp.bed.He-Lu-lu-1-48ng_S1_L006_R1_001.plus.21.bp.win.bed", header=F)
reads1=as.numeric(as.character(reads[,4]))
reads2=as.numeric(as.character(reads[,5]))
reads1[is.na(reads1)]=0
reads2[is.na(reads2)]=0
reads.combine=cbind(reads1, reads2)
reads.effect=reads.combine[which(rowSums(reads.combine) > 0),] 
plot(log2(reads.effect[,1]), log2(reads.effect[,2]), pch=16, xlab="log2(reads) Rep 1", ylab="log2(reads) Rep 2", main="48ng 5hmC")
legend(0, 12, c("r=0.94"))
#dev.off()
```



```{r, echo=F, warning=F, eval=F}
reads1=as.numeric(as.character(read.table("../../5hmC/first.six.sample.sorted.bam.cove.base.level.extend.bed.with.reads.second.six.sample.sorted.bam.cove.base.level.extend.bed.with.reads.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("..\\5hmC\\first.six.sample.sorted.bam.cove.base.level.extend.bed.with.reads.second.six.sample.sorted.bam.cove.base.level.extend.bed.with.reads.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
plot(log2(reads1), log2(reads2), pch=16, xlab="Combined sample 1", ylab="Combined sample 2", main="48ng combine 5hmC")
legend(0, 15, c("r=0.98"))

```



```{r, echo=F, warning=F}
reads1=as.numeric(as.character(read.table("../../5hmC/correlation/20160601_5hmC_Jump_Seq_24ng.minus.FDR0.05.corrected.txt.bed.He-lu-7_S9_L006_R1_001.minus.FDR0.05.corrected.txt.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("../../5hmC/correlation/20160601_5hmC_Jump_Seq_24ng.minus.FDR0.05.corrected.txt.bed.He-lu-7_S9_L006_R1_001.minus.FDR0.05.corrected.txt.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="24ng 5hmC")
legend(2, 12, c("r=0.99"))

```


```{r, echo=F, warning=F}
reads1=as.numeric(as.character(read.table("../../5hmC/correlation/20160428_5hmC_Jump_Seq_NPM_12ng.minus.FDR0.05.txt.bed.CHe-Lu-3_S14_L005_R1_001.minus.FDR0.05.txt.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("../../5hmC/correlation/20160428_5hmC_Jump_Seq_NPM_12ng.minus.FDR0.05.txt.bed.CHe-Lu-3_S14_L005_R1_001.minus.FDR0.05.txt.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="12ng 5hmC")
legend(2, 12, c("r=-0.09"))

```




```{r, echo=F, warning=F}
reads1=as.numeric(as.character(read.table("../../5hmC/correlation/20160428_5hmC_Jump_Seq_NPM_6ng.minus.FDR0.05.txt.bed.CHe-Lu-6ng_L008_R1_001.minus.FDR0.05.txt.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("../../5hmC/correlation/20160428_5hmC_Jump_Seq_NPM_6ng.minus.FDR0.05.txt.bed.CHe-Lu-6ng_L008_R1_001.minus.FDR0.05.txt.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="6ng 5hmC")
legend(2, 12, c("r=0.95"))

```



```{r, echo=F, warning=F}
reads1=as.numeric(as.character(read.table("../../5hmC/correlation/CHe-lu-2.4ng_mixed_with_other_library.minus.FDR0.05.corrected.txt.bed.He-Lu-10_2.4ng-S5_L001_R1_001.minus.FDR0.05.corrected.txt.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("../../5hmC/correlation/CHe-lu-2.4ng_mixed_with_other_library.minus.FDR0.05.corrected.txt.bed.He-Lu-10_2.4ng-S5_L001_R1_001.minus.FDR0.05.corrected.txt.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="2.4ng 5hmC")
legend(2, 10, c("r=0.99"))

```



![Six samples of 12 samplers are combined as combined sample 1 and the rest six are combined sample 2. Reads in 20 bp windows are used to calculate correlations between these two combined samples. log2 transformation is used for reads on both x and y axis.](figure/correltion-combined-5hmc.png)




```{r, echo=F, eval=F}
0.6ng: rep 1: 49/334;  rep 2: 57/297
```

![ Venn diagram of two 0.6ng 5hmC replicates. For each replicate, 1kb windows  are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle.  ](figure/VennDiagram_5hmC_0.6ng_tworeplicate.png)



### correlation between different DNA levels 

```{r, echo=F, cache=T, eval=F}
0.6ng vs 48ng : 0.6ng: 94/334; 48ng: 79/195809 
```

![ Venn diagram of  0.6ng 5hmC  and 48ng 5hmC. For each sample, 1kb windows  are called at FDR 0.05. The proportion of peak windows between two replicates are presented inside the circle.  ](figure/VennDiagram_5hmC_0.6ng_48ng.png)





## Comparison with other methods 

### correlation between Jump-seq and nano-seal

```{r, echo=F, eval=F}
reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\Dali Han\\GSM2166069_mESC_50ng_1.ucsc.bedGraph.noheader.48ng_11rep_merge.mm9.last2.col", header=F)
reads1=as.numeric(as.character(reads[,1]))
reads2=as.numeric(as.character(reads[,2]))
reads22=reads2[is.na(reads2)]
```

The correlation between combined 48ng 5hmC sample and 50ng  nano-seal sample is 0.08. The calculation is problematic as jump-seq is using the actual number of reads while nano-seal uses average number across cells.    

### Comparison of Jump vs nano-seal with strong TAB-seq as gold standard 


```{r, echo=F}
percent=matrix(nrow=2, ncol=6)
percent[,1]=c(0.7186, 0.9186)
percent[,2]=c(0.7786, 0.9186)

percent[,3]=c(0.9356, 0.9186)
percent[,4]=c(0.9598, 0.9186)

percent[,5]=c(0.9560, 0.9186)
percent[,6]=c(0.9504, 0.9186)
colnames(percent)=c("0.6ng", "2.4ng", "6ng", "12ng", "24ng", "48ng")
bar.centers=barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Nano-seal peaks"), ylim=c(0,1.5),xlab="", main="", ylab="Proportion of peaks overlapping with TAB-seq peaks", cex.names=0.5)

```

* JumpSeq: 1kb windows are called at FDR 0.05. To make a fair comparison, we trim the jump-seq peak windows to be of the same length as nano_seq peaks.   


### Enrichments 

```{r, echo=F, fig.cap="For jump-seq, we called 1kb windows at FDR0.05 as peaks and computed the proportion of peaks (in red) overlapping with Nano-seal peaks and Tab-seq peaks. Randomly pick same number of 1kb windows in the whole genome as jump-seq peaks at different DNA levels and calculate its proportion of windows (in blue) overlapping with Nano-seal peaks and Tab-seq peaks extended to 1kb. "}
percent=matrix(nrow=2, ncol=12)

percent[,1]=c(0.0808, 0.0365) # odd rows for nano-seal and even for TAB-seq
percent[,2]=c(0.7172, 0.4827)

percent[,3]=c(0.1867, 0.0365)
percent[,4]=c(0.7942,0.4827)

percent[,5]=c(0.4099, 0.0365)
percent[,6]=c(0.9439, 0.4827)

percent[,7]=c(0.3727, 0.0365)
percent[,8]=c(0.9664, 0.4827)

percent[,9]=c(0.2914, 0.0365)
percent[,10]=c(0.9646,0.4827)

percent[,11]=c(0.2872, 0.0365)
percent[,12]=c(0.9608,0.4827)
colnames(percent)=rep(c("0.6", "2.4ng", "6ng", "12ng", "24ng", "48ng"), each=2)

par(mfrow=c(1,2))
barplot(percent[,c(1,3,5,7,9, 11)], beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,0.7),
xlab="Figure A", main="", ylab="Proportion of peaks overlapping with Nano-seal peaks", cex.names=0.5)

barplot(percent[,c(2,4,6,8,10, 12)], beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,1.5),
xlab="Figure B", main="", ylab="Proportion of paeks overlapping with Tab-seq peaks", cex.names=0.5)
```

* Jump-seq; 1kb window are called at FDR 0.05 

### Sensitvity 

```{r, echo=F, fig.cap="For jump-seq, 1kb windows are called at FDR0.05 as peaks. At different DNA levels, the proporttion of Tab-seq peaks overlapping with Jump-seq peaks are calculated. ", eval=F}
percent=matrix(nrow=2, ncol=6)
percent[,1]=c(0.0002, 0.0002)
percent[,2]=c(0.0004, 0.0004)
percent[,3]=c(0.0050, 0.0058)
percent[,4]=c(0.0652, 0.0665)
percent[,5]=c(0.2284, 0.2303)
percent[,6]=c(0.2558, 0.2575)
colnames(percent)=c("0.6ng", "2.4ng", "6ng", "12ng", "24ng", "48ng")
col.mean=colMeans(percent)

#normalized.mean=col.mean/max(col.mean)
barplot(col.mean , col="darkblue", ylab="Proportion of Tab-seq peaks recovered by Jump-seq", cex.lab=0.9)
#tab_seq_overlap=round(col.mean*2057636)
percent=matrix(nrow=2, ncol=6)
percent[,1]=c(0.0002, 0.0002)
percent[,2]=c(0.0004, 0.0003)
percent[,3]=c(0.009, 0.0105)
percent[,4]=c(0.1126, 0.1153)
percent[,5]=c(0.3447, 0.3466)
percent[,6]=c(0.3842, 0.3855)
colnames(percent)=c("0.6ng", "2.4ng", "6ng", "12ng", "24ng", "48ng")
col.mean=colMeans(percent)
#normalized.mean=col.mean/max(col.mean)
barplot(col.mean, col="darkblue", ylab="Proportion of nano-seal  peaks recovered by Jump-seq", cex.lab=0.9)
#nano_seal_overlap=round(col.mean*64865)
```


```{r, echo=F}
overlap=matrix(nrow=2, ncol=6)
tab_by_jump=c(832, 1116, 19686, 219644, 649783, 715179)
tab_by_nano=rep(465501, 6)
overlap[1,]=tab_by_jump
overlap[2,]=tab_by_nano
colnames(overlap)=c("0.6ng", "2.4ng", "6ng", "12ng", "24ng", "48ng")
rownames(overlap)=c("Jump_seq", "Nano_seal (50ng)")
kable(overlap, caption="Number of Tab-seq peaks recovered by Jump_seq and nano-seal")

#pdf("C:\\Shengtong\\Research\\Jump-Seq\\5hmC\\jump-seq-vs-nano.pdf")
bar.centers=barplot(overlap[,6], beside=T,   col=c("red", "darkblue"), 
ylab="# Tab-seq peaks overlaping with other peaks", main="48ng Jump-seq vs 50ng nano-seal", xlab="", cex.names=0.7, legend=c("Jump-seq peaks", "nano-seal peaks"), ylim=c(0, 760000))
text(bar.centers[1,1], 740000, "715179")
text(bar.centers[2,1], 490000, "465501")
#dev.off()
```


To be  fair, 1kb Jump-sea peaks are trimed to be of the same length as nano-seal peaks. 
 









### Overlap with other peaks by Venn diagram 

If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of  which 686652 windows are overlapping with strong TAB-seq windows which are extended 10 bp downtream and 10 bp upstream. for strong TAB-seq, there are 2057636 20bp extended windows and 763937 are overlapping with Jump-seq windows.   



![ Venn diagram of 5hmC and TAB-seq. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with TAB-seq and the proportion of peaks of TAB-seq overlapping with jump-seq are presented inside the circle.  ](figure/VennDiagram.png)



If we combine eleven 48ng 5hmC samples together, there are 2367316 20bp windows with FDR 0.05, of  which 386331 (16.32%) windows are overlapping with hmC seal. For hmC seal,  there are 64865 peaks and 58280 (89.85%) are overlapping with 5hmC peaks. 

![ Venn diagram of 5hmC and hmC-seal. 48ng samples are combined together and 20 bp windows are called at FDR 0.05. The proportion of peak windows of jump-seq overlapping with hmC-seal  and the proportion of peaks of hmC-seal overlapping with jump-seq are presented inside the circle. ](figure/VennDiagram_hmc_hmcseal.png)



 If we use single 48gn 5hmC sample, then 28095/97548=28.80% 1kb windows are overllaping with nano-seal peaks and 24920/64865=38.42% nano-seal peaks overlapping with jump-seq.  
















## BiS-seq (5mC) 

 
 
### Enrichment

```{r, echo=F, fig.cap="Enrichment of methylated cytosine from Gary Hon  at different genomic concentrations. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng. Number of peak windows are shown over the bar."}
minus.percent=matrix(nrow=2, ncol=5)
minus.percent[,1]=c(0.7895, 0.1450)
minus.percent[,2]=c(0.8069, 0.1450)
minus.percent[,3]=c(0.8051, 0.1450)
minus.percent[,4]=c(0.8511, 0.1450)
minus.percent[,5]=c(0.8374, 0.1450)
colnames(minus.percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")

plus.percent=matrix(nrow=2, ncol=5)
plus.percent[,1]=c(0.7826, 0.1450)
plus.percent[,2]=c(0.8473, 0.1450)
plus.percent[,3]=c(0.8355, 0.1450)
plus.percent[,4]=c(0.8482, 0.1450)
plus.percent[,5]=c(0.8403, 0.1450)
colnames(plus.percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")
percent=(plus.percent+minus.percent)/2


bar.centers=barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,1.3),
ylab="Proportion of peaks overlapping with methylated cytosine", main="", xlab="", cex.names=0.7)
num.peaks=c(552,5666,10032, 30801, 93420)
above.dist=percent[1,]+0.04
for (i in 1:5)
  text(bar.centers[1,i], above.dist[i], num.peaks[i])
```


* cutoff used for peak windows: 2.4ng (FDR0.05), others use p values less than $10^{-6}$. 




### Sensitivity 


```{r, echo=F, fig.cap="Normalized proportion of  methylated cytosine from Gary Hon  at different genomic concentrations recovered by Jump-seq. Peak windows with p value less than 10^{-6} is used for 48ng, 24ng, 12ng, 6ng and FDR0.05 is used for 2.4ng"}
percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.000026, 0.000029)
percent[,2]=c(0.000195, 0.000334)
percent[,3]=c(0.000313, 0.000553)
percent[,4]=c(0.000503, 0.000704)
percent[,5]=c(0.000776, 0.000786)
colnames(percent)=c("2.4ng", "6ng", "12ng", "24ng", "48ng")
col.mean=colMeans(percent)
normalized.mean=col.mean/max(col.mean)
barplot(normalized.mean, col="darkblue", ylab="Normalized proportion of methylated cytosine")
```


### Correlation between replicates 

```{r, echo=F, warning=F, fig.cap="Correlation of reads in 20 bp windows between two replicates of 2.4ng 5mC. log2 transformation is used for computing correlations"}
reads1=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_2.4ng.minus.21.bp.win.5prime.pstv.cove.bed.numberC.combine.pvalue.He-Lu-10_5mc-jump-2.4ng-S10_L004_R1_001.minus.21.bp.win.5prime.pstv.cove.bed.numberC.combine.pvalue.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_2.4ng.minus.21.bp.win.5prime.pstv.cove.bed.numberC.combine.pvalue.He-Lu-10_5mc-jump-2.4ng-S10_L004_R1_001.minus.21.bp.win.5prime.pstv.cove.bed.numberC.combine.pvalue.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
#pdf("../../Figure/correlation_two_5mc_2.4ng.pdf")
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="2.4ng 5mC")
legend(0, 12, c("r=0.98"))
#dev.off()
```

```{r, echo=F, warning=F, fig.cap="Correlation of reads in 20 bp windows between two replicates of 6ng 5mC. log2 transformation is used for computing correlations"}
reads1=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_6ng.minus.pvalue10-6.bed.He-Lu-9_5mc-jump-6ng-S9_L004_R1_001.minus.pvalue10-6.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_6ng.minus.pvalue10-6.bed.He-Lu-9_5mc-jump-6ng-S9_L004_R1_001.minus.pvalue10-6.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
#pdf("../../Figure/correlation_two_5mc_2.4ng.pdf")
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="6ng 5mC")
legend(2, 12, c("r=0.91"))
#dev.off()
```



```{r, echo=F, warning=F, fig.cap="Correlation of reads in 20 bp windows between two replicates of 12ng 5mC. log2 transformation is used for computing correlations"}
reads1=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_12ng.minus.pvalue10-6.bed.He-Lu-8_5mc-jump-12ng-S8_L004_R1_001.minus.pvalue10-6.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_12ng.minus.pvalue10-6.bed.He-Lu-8_5mc-jump-12ng-S8_L004_R1_001.minus.pvalue10-6.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
#pdf("../../Figure/correlation_two_5mc_2.4ng.pdf")
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="12ng 5mC")
legend(2, 12, c("r=0.97"))
#dev.off()
```
 



```{r, echo=F, warning=F, fig.cap="Correlation of reads in 20 bp windows between two replicates of 24ng 5mC. log2 transformation is used for computing correlations"}
reads1=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_24ng.minus.pvalue10-6.bed.He-Lu-7_5mC-jump-24ng-S7_L004_R1_001.minus.pvalue10-6.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_24ng.minus.pvalue10-6.bed.He-Lu-7_5mC-jump-24ng-S7_L004_R1_001.minus.pvalue10-6.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
#pdf("../../Figure/correlation_two_5mc_2.4ng.pdf")
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="24ng 5mC")
legend(2, 12, c("r=0.98"))
#dev.off()
```




```{r, echo=F, warning=F, fig.cap="Correlation of reads in 20 bp windows between two replicates of 48ng 5mC. log2 transformation is used for computing correlations"}
reads1=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_48ng.minus.pvalue10-6.bed.He-Lu-6_5mC-jump-48ng-S6_L004_R1_001.minus.pvalue10-6.bed.bed.reads.1", header=F)[,1]))
reads1[is.na(reads1)]=0
reads2=as.numeric(as.character(read.table("C:\\Shengtong\\Research\\Jump-Seq\\5mC\\20160811_5mC_Jump_Seq_48ng.minus.pvalue10-6.bed.He-Lu-6_5mC-jump-48ng-S6_L004_R1_001.minus.pvalue10-6.bed.bed.reads.2", header=F)[,1]))
reads2[is.na(reads2)]=0
#pdf("../../Figure/correlation_two_5mc_2.4ng.pdf")
plot(log2(reads1), log2(reads2), pch=16, xlab="Rep 1", ylab="Rep 2", main="48ng 5mC")
legend(2, 12, c("r=0.93"))
#dev.off()
```



## Enrichment of genomic features
 


```{r, echo=F, fig.cap=" Eleven 48 ng samples are combined together and 20 bp peak windows are called at FDR 0.05. Random windows are randomly selected  in the whole genome with the same amount of peak windows. The proportion of chosen peak windows (red) and random windows (blue) overlapping in  different genomic regions are calculated and the corresponding odds ratio are shown above the red bar"}
par(mfrow=c(1,2))

minus.percent=matrix(nrow=2, ncol=9)
minus.percent[,1]=c(0.0070, 0.0032)
minus.percent[,2]=c(0.0139, 0.0034)
minus.percent[,3]=c(0.0069, 0.0026)
minus.percent[,4]=c(0.0042, 0.0021)
minus.percent[,5]=c(75887/2368196, 140/10000)
minus.percent[,6]=c(30340/2368196, 53/10000)
minus.percent[,7]=c(153280/2368196, 130/10000)
minus.percent[,8]=c(121485/2368196, 265/10000)
minus.percent[,9]=c( 1213246/2368196, 3860/10000)
colnames(minus.percent)=c("ctcf", "enhancer", "h3k27ac", "h3k4me3", "3UTR", "5UTR", "exon", "promoter", "intron")

plus.percent=matrix(nrow=2, ncol=9)
plus.percent[,1]=c(0.0068, 0.0032)
plus.percent[,2]=c(0.0139, 0.0034)
plus.percent[,3]=c(0.0070, 0.0026)
plus.percent[,4]=c(0.0042, 0.0021)
plus.percent[,5]=c(75919/2366463, 140/10000)
plus.percent[,6]=c(29801/2366463, 53/10000)
plus.percent[,7]=c(152850/2366463, 130/10000)
plus.percent[,8]=c(119941/2366463, 265/10000)
plus.percent[,9]=c(1212097/2366463, 3860/10000)
colnames(plus.percent)=c("ctcf", "enhancer", "h3k27ac", "h3k4me3","3UTR", "5UTR", "exon", "promoter", "intron")
ave.percent=(minus.percent+plus.percent)/2


#pdf("D:\\Dropbox\\JumpSeqProject\\Figure\\Enrichment_different_region_48ng_11sampleCombined_FDR0.05.pdf")
barplot(ave.percent[,1:4], beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0, 0.03), 
ylab=" % Jump-seq peaks", main="", xlab="", cex.names=1, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=1)
cex.size=0.7
text(2.2, 0.008, "2.16(p<2.2e-16)", cex=cex.size)
text(4.8, 0.015, "4.09(p<2.2e-16)", cex=cex.size)
text(8.2, 0.008, "2.67(p<2.2e-16)", cex=cex.size)
text(10.8, 0.005, "2.00(p<2.2e-16)", cex=cex.size)
text(14, 0.0015, "0.50(p=1)")
#dev.off()

barplot(ave.percent[,5:8], beside=T, col=c("red", "darkblue"), ylim=c(0, 0.085),
ylab="", main="", xlab="", cex.names=1, xpd = FALSE, las=1)

text(2.2, 0.035, "2.28(p<2.2e-16)", cex=cex.size)
text(5.2, 0.018, "2.42(p=1.426e-13)", cex=cex.size)
text(8.2, 0.068, "4.98(p<2.2e-16)", cex=cex.size)
text(11, 0.055, "1.94(p<2.2e-16)", cex=cex.size)
text(14, 0.002, "0.50(p=1)")
```




```{r, echo=F, fig.cap="For 0.6ng 5hmC sample, 20 bp windows with at least one reads are called peak windows. Random windows are chosen in the whole genome in the same amount of peak windows. The proportion of peak windows and random windows in different genomic regions (200bp bins) are computeed. "}
percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0049, 0.0032)
percent[,2]=c(0.0056, 0.0035)

percent[,3]=c(0.0033, 0.0026)
percent[,4]=c(0.0021, 0.0021)

percent[,5]=c(0.0034, 0.0012)
colnames(percent)=c("ctcf", "enhancer", "h3k27ac", "h3k4me3", "polII")

barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,0.01),
ylab="Proportion of Jump-seq windows", main="", xlab="", cex.names=1)
text(2.0, 0.0052, "1.56(p=3.42e-05)", cex=0.8)
text(4.5, 0.006, "1.65(p=4.46e-07)")
text(7.7, 0.0038, "1.24(p=0.09)")
text(10.7, 0.0026, "0.99(p=1)")
text(14, 0.004, "2.95(p=9.15e-15)", cex=0.8)
```

* cis-regulatory regions are extended 200bp windows. 
* jump peaks are 20 bp windows with at least one reads. 





```{r, echo=F, eval=F}
count=matrix(nrow=2, ncol=2)
count[1,]=c(121485, 2368196)
count[2,]=c(265, 10000)
fisher.test(count)
```

## PN analysis 

For both male and female, at every stage, the nnumber of reads are counted in each region. Within every region, the reads are normalized around zero. To see the overall pattern, in the second figure, reads from all regions are put together. 


### 5hmC 


**Use maximum number of reads as baseline**

```{r, echo=F}
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)
common.second.stage=c("PN5","1cell","2cellearly", "2celllate", "ES")
plot_function=function(hmc_combine, sample.label)
{
scale_hmc_combine=scale(as.matrix(hmc_combine)) 
min.limit=min(scale_hmc_combine)-1; max.limit=max(scale_hmc_combine)+1

female.index=c(seq(1,5), seq(11,15))
male.index=c(seq(6,10), seq(11,15))
######### plot 1*3 figure 
par(mfrow=c(1,3))
plot(scale_hmc_combine[1:5,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=paste(sample.label,  "Maternal", sep="-"), cex.main=0.8, cex.lab=1, lwd=2)
for (i in 2:num.regions)
  lines(scale_hmc_combine[1:5,i], col=i, type="o", lwd=2)
legend(1, max.limit, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.65)
axis.labels <-c("MII", "PN1","PN2","PN3","PN4")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale_hmc_combine[6:10,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main=paste(sample.label, "Paternal", sep="-"), cex.main=0.8, cex.lab=0.6, lwd=2)
for (i in 2:num.regions)
  lines(scale_hmc_combine[6:10,i], col=i, type="o", lwd=2)
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("Sperm", "PN1","PN2","PN3","PN4")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale_hmc_combine[11:15,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main="", cex.main=0.8, cex.lab=0.6, lwd=2)
for (i in 2:num.regions)
  lines(scale_hmc_combine[11:15,i], col=i, type="o", lwd=2)
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN5","1cell","2cellearly", "2celllate", "ES")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)
################ plot with regions combined 
par(mfrow=c(1,1))
hmc_combine_region=rowSums(hmc_combine)
scale_hmc_combine_region=scale(hmc_combine_region)
min.limit=min(scale_hmc_combine_region)-1; max.limit=max(scale_hmc_combine_region)+1
plot(scale_hmc_combine_region[male.index], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main= paste(sample.label, " at different stages", sep=""), cex.main=0.8, cex.lab=1, lty=1, pch=16, lwd=2)
lines(scale_hmc_combine_region[female.index], col=2, type="o", lty=1, pch=16, lwd=2)
legend(2, max.limit, c("Paternal", "Maternal"), col=seq(1,2), lty=rep(1,2), ncol=2, cex=0.65)

axis.labels <-c("MII/Sperm", "PN1","PN2","PN3","PN4", common.second.stage)
axis(1, at=seq(1,length(male.index)), labels=axis.labels, cex.axis=0.8, las=2)
}
```





#### Replicate 1 

```{r, echo=F}
################# replicate 1 ##############
#scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\scale.reads.xiaolong.annotation.csv")
scale.reads=read.csv("../../20180113/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
new.scale.reads=read.csv("../../20180420/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
hmc_female=rbind(scale.reads[5,], scale.reads[12,], scale.reads[14,], scale.reads[16,], scale.reads[18,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]

hmc_male=rbind(new.scale.reads[20,], scale.reads[13,], scale.reads[15,], scale.reads[17,], scale.reads[19,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
rownames(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
first_hmc_combine=rbind(hmc_female[1:5,], hmc_male)
plot_function(first_hmc_combine, "5hmC")
#write.csv(first_hmc_combine, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PNData4Xiaolong\\normalized-with-spikein-5hmC-replicate1.csv")
```

#### Replicate 2 

```{r, echo=F}
############## replicate 2 ####################
second.rep.scale.reads=read.csv("../../20180228/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
new.scale.reads=read.csv( "../../20180420/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
hmc_male=rbind(new.scale.reads[20,], second.rep.scale.reads[10,], second.rep.scale.reads[12,], second.rep.scale.reads[14,], second.rep.scale.reads[16,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
hmc_female=rbind(second.rep.scale.reads[5,], second.rep.scale.reads[9,], second.rep.scale.reads[11,], second.rep.scale.reads[13,], second.rep.scale.reads[15,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]
second_hmc_combine=rbind(hmc_female[1:5,], hmc_male)
#write.csv(second_hmc_combine, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PNData4Xiaolong\\normalized-with-spikein-5hmC-replicate2.csv")
plot_function(second_hmc_combine, "5hmC")

```

### 5mC 


**Use maximum number of reads as baseline**

#### Replicate 1

```{r, echo=F}
############# maternal 
scale.reads=read.csv("../../20180420/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
mc_female=rbind(scale.reads[7,], scale.reads[12,], scale.reads[1,], scale.reads[3,], scale.reads[23,], scale.reads[25,], scale.reads[26,], scale.reads[5,], scale.reads[6,], scale.reads[10,] )
rownames(mc_female)=mc_female[,1]
mc_female=mc_female[,-1]

############### paternal 
scale.reads=read.csv("../../20180420/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
mc_male=rbind(scale.reads[8,], scale.reads[15,], scale.reads[2,], scale.reads[4,],  scale.reads[24,], scale.reads[25,], scale.reads[26,],  scale.reads[5,], scale.reads[6,], scale.reads[10,] )
rownames(mc_male)=mc_male[,1]
mc_male=mc_male[,-1]
first_mc_combine=rbind(mc_female[1:5,], mc_male)
#write.csv(first_mc_combine, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PNData4Xiaolong\\normalized-with-spikein-5mC-replicate1.csv")

plot_function(first_mc_combine, "5mC")




```



#### Replicate 2 

```{r, echo=F}
############ paternal 
scale.reads=read.csv("../../20180227/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
new.scale.reads=read.csv("../../20180420/scale.reads.xiaolong.annotation.use.large.as.baseline.csv")
#mc_male=rbind(scale.reads[5,], scale.reads[9,], scale.reads[11,], scale.reads[13,], scale.reads[15,], scale.reads[1:3,], scale.reads[7,] )
mc_male=rbind(new.scale.reads[8,], scale.reads[9,], scale.reads[11,], scale.reads[13,], scale.reads[15,], scale.reads[1:3,], scale.reads[16,], scale.reads[7,] )
rownames(mc_male)=mc_male[,1]
mc_male=mc_male[,-1]

############### Maternal 
scale.reads=read.csv("../../20180227/scale.reads.xiaolong.annotation.csv")
mc_female=rbind(scale.reads[4,], scale.reads[8,], scale.reads[10,], scale.reads[12,], scale.reads[14,], scale.reads[1:3,], scale.reads[16,], scale.reads[7,] )
rownames(mc_female)=mc_female[,1]
mc_female=mc_female[,-1]
second_mc_combine=rbind(mc_female[1:5,], mc_male)
#write.csv(second_mc_combine, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PNData4Xiaolong\\normalized-with-spikein-5mC-replicate2.csv")

plot_function(second_mc_combine, "5mC")



```



 







