---
title: "5hmC data analysis"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

All samples are [here][allsample]: 48 ng (11); 24 ng (5); 12 ng (5); 6ng (5); 2.4 ng (5); 0.6ng (2). 

[allsample]: https://docs.google.com/spreadsheets/d/1FHkVSgTCJDY19VcFg2goat4rXJfxofCn3CFRV-gCbfo/edit#gid=400442004


Mouse genome mm9 has 2,654,911,517 bases, 507,439,500 (19.11324\%) cytosines and 507,585,491 (19.11873\%) Gumines.

##  48ng 5hmC replicates

There are eleven 48ng 5hmC replicates and each one has two strands, plus+minus. 


rep  | 1 |2 | 3 | 4 | 5| 6| 7| 8| 9| 10| 11 | combine
-----|---|---|---|---|---|---|---|---|---|---|------|----
\#reads(plus) |3,516,149|3,781,024|2,014,714|6,982,592|4,741,743|5,036,004|5,825,862|5,532,109|5,072,780|3,951,163|3,641,213|50,057,101
\#reads(minus)|3,524,627| 3,793,291|2,021,689|6,995,038|4,754,910 |5,048,479|5,834,324|5,546,279|5,059,828|3,935,679|3,631,379|50,183,793
Table: Number of reads in plus and minus strand for each 48 ng replicate. 

### Two large combined replicates from eleven 48 ng 5hmC replicates
Once the base level coverage of each individual replicates is obtained, directly combine some of them to form a synthesized large replicate and the rest form to another such that their coverage, i.e. number of reads are as balanced as possible. In this way, power is expected to be increasing. 

* combined rep 1: rep 4+5+6+7+8
* combined rep 2:rep 1+2+3+9+10+11

Two large combined replicates | \#reads
------------------------------|-----------
rep1.minus | 28,179,030 
rep1.plus | 28,118,310
rep2.minus | 21,966,493
rep2.plus | 21,977,043
rep1+rep2:minus| 50,145,523
rep1+rep2:plus | 50,095,353
Table: Number of reads in each combined replicate. 

##### Comparison 1: Comparison with Dali Han's data
Dali Han's peak: 64865 peak windows with varying window size. 


Originally called 20 bp windows are extended by 2kbp, 1kbp upstream and 1kbp downstream. Overlapping between extended peak windows (at four p value cutoffs $10^{-3}, 10^{-7}, 10^{-15}, 10^{-20}$) and Dali Han's data (window size differs among each other) are summzried below. 

Combined replicate | \#overlap.win | \#.win | ratio
-------------------|---------------|--------|------
rep1.minus.21.bp.enriched.window.p10-3.bed.extend2kbp.bed | 345920 | 1318454 | 0.2623679
rep1.minus.21.bp.enriched.window.p10-7.bed.extend2kbp.bed | 74515 | 236507 | 0.3150647
rep1.minus.21.bp.enriched.window.p10-15.bed.extend2kbp.bed | 9191 | 27656 | 0.3323329
rep1.minus.21.bp.enriched.window.p10-20.bed.extend2kbp.bed | 3686 | 11162 | 0.3302276
rep1.plus.21.bp.enriched.window.p10-3.bed.extend2kbp.bed | 344905 | 1317670 | 0.2617537
rep1.plus.21.bp.enriched.window.p10-7.bed.extend2kbp.bed | 74504 | 236568 | 0.3149369
rep1.plus.21.bp.enriched.window.p10-15.bed.extend2kbp.bed | 9167 | 27662 | 0.3313932
rep1.plus.21.bp.enriched.window.p10-20.bed.extend2kbp.bed | 3554 | 10997 | 0.3231790
rep2:minus.21.bp.enriched.window.p10-3.bed.extend2kbp.bed | 349863 | 1245093 | 0.2809935
rep2:minus.21.bp.enriched.window.p10-7.bed.extend2kbp.bed | 70615 | 199358 | 0.3542120
rep2:minus.21.bp.enriched.window.p10-15.bed.extend2kbp.bed | 8216 | 21511 | 0.3819441
rep2:minus.21.bp.enriched.window.p10-20.bed.extend2kbp.bed | 2851 | 7710 | 0.3697795
rep2:plus.21.bp.enriched.window.p10-3.bed.extend2kbp.bed | 348193 | 1243355 | 0.2800431
rep2:plus.21.bp.enriched.window.p10-7.bed.extend2kbp.bed | 71014 | 200017 | 0.3550398
rep2:plus.21.bp.enriched.window.p10-15.bed.extend2kbp.bed | 8769 | 22941 | 0.3822414
rep2:plus.21.bp.enriched.window.p10-20.bed.extend2kbp.bed | 2792 | 7557 | 0.3694588
Table: Comparison of peak windows between two large replivcates and Dali's data. Column2: total number of peak windows; Column3: number of peak windows overlapping with Dali's peak; Column4: the ratio.   
```{r, echo=T}
rep1.minus=c(0.2624, 0.3151, 0.3323, 0.3302); rep1.plus=c(0.2618, 0.3149, 0.3314, 0.3232)
rep2.minus=c(0.2810, 0.3542, 0.3819, 0.3698); rep2.plus=c(0.2800, 0.3550, 0.3822, 0.3695)
plot(rep1.minus, ylim=c(0.2, 0.4), type="o", xaxt='n', xlab="p value cutoffs", ylab="ratio", col=1, main="Overlapping rate with Dali's peaks")
lines(rep1.plus, col=2, type="o")
lines(rep2.minus, col=3, type="o")
lines(rep2.plus, col=4, type="o")
legend(3, 0.3, c("rep1.minus", "rep1.plus", "rep2.minus", "rep2.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("10^(-3)", "10^(-7)", "10^(-15)", "10^(-20)")
axis(1, at=seq(1,4), labels=axis.labels)
```
##### Comparison 2: Comparison between two large replicates 


minus | $10^{-1}$| $10^{-3}$ | $10^{-5}$ |$10^{-7}$ | $10^{-15}$ | $10^{-20}$
------|----------|-|-----------|------------|-------------|--
rep1 | 0.9905  |0.9388 | 0.8425 |0.7334 | 0.4543 | 0.4056
rep2 |0.9886 | 0.9523 |  0.8961 |0.8091 | 0.5576 | 0.5612
plus |$10^{-1}$| $10^{-3}$ | $10^{-5}$|$10^{-7}$ | $10^{-15}$ | $10^{-20}$
rep1 | 0.9906 | 0.9389 | 0.8426 |0.7337 | 0.4650 | 0.3952
rep2 |0.9887 | 0.9524 | 0.8967 |0.8095 | 0.5397 | 0.5419
Table: Overlapping among peak windows extended by 2kbp for 2 large combined replicates at different p value cutoffs.


Surprisingly, the overlapping rate deceases as the p value cutofff becomes more stringent. 



minus | 0.5| 0.1 | 0.01|0.001 | 0.0001 |$10^{-7}$ |$10^{-10}$
------|----------|-|-----------|------------|----|-----|---
rep1 |0.3955| 0.4118 | 0.3840 | 0.3611| 0.3563| 0.3148 |0.3118
rep2 |0.4985 | 0.3915  |  0.4111 |0.4109 | 0.4023| 0.4196| 0.4231
plus |0.5| 0.1 | 0.01|0.001 | 0.0001 | $10^{-7}$ |$10^{-10}$
rep1 | 0.3954 | 0.4126 | 0.3841 |0.3627 | 0.3572 | 0.3991|0.3115
rep2 |0.4985 | 0.3907 | 0.4116 |0.4128 | 0.4030 | 0.4205| 0.4205
Table: overlapping for originally called 20 bp windows at different FDR levels.

### Downsample
Combine all 48ng replicates together, downsample $40\%, 50\%, \cdots, 90\%$ reads, call peak windows, and see how \#peak.windows (absolute peak windows or overlapping with other data) vary with number of reads.   

```{r, engine = 'bash', eval = FALSE}
samtools merge output.sorted.bam input1.sorted.bam input2.sorted.bam # merge together sorted bam files
```

```{r, engine='bash', eval=F}
samtools view -s 0.25 -b input.all.bam > input_25p.sam # downsample 25% reads 
```

ratio  | plus | minus 
-------|------|-----
40\% |20,027,749 |20,070,771
50\% |25,029,777 |25,091,279
60\% |30,035,435 |30,110,375
70\% |35,040,232|35,127,703
80\% |40,044,961 |40,149,113
90\% |45,048,466 |45,168,687
100\%|50,057,101 | 50,183,793
Table: Actutrally number of reads from each downsample with ratios of  $40\%, 50\%, \cdots, 100\%$. 


#### Reads distribution 
Windows with reads greater than 10 are put together into category 11. 
```{r, echo=T}
reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), ">10"); reads.plus[,1]=c(9371352, 2190106, 772609, 334745, 162092, 84161, 45907, 25962, 15529, 9208, 16960)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of peak windows (40% reads plus)", xlab="Number of reads")
```

```{r, echo=T}
reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), ">10"); reads.plus[,1]=c(14075383, 4568606, 2003109, 1078497, 642833, 411173, 274861, 188434, 133511, 95662, 300518)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of peak windows (100% reads plus)", xlab="Number of reads")
```

For each down sample, call base level coverage, call peak windows and count how many peak windows at specified FDR level. 

```{r, echo=T,  eval =T}
num.win.FDR5per=c(759517, 969627, 1299793, 1599795, 1891138, 2107755, 2368196)
num.reads=c(20070771, 25091279, 30110375, 35127703, 40149113, 45168687, 50183793)
plot(num.reads, num.win.FDR5per, xlab="Number of reads", ylab="# peak windowsof 20 bp at FDR 0.05", type="o", main="minus strands")
```

```{r, echo=T,  eval =T}
num.win.FDR5per=c(760022, 968832, 1309038, 1597802, 1905666, 2124123, 2366463)
num.reads=c(20027749, 25029777, 30035435, 35040232, 40044961, 45048466, 50057101)
plot(num.reads, num.win.FDR5per, xlab="Number of reads", ylab="# peak windowsof 20 bp at FDR 0.05", type="o", main="plus strands")
```

**Based on the current observations, the number of peak windows grows as the read coverage increases, it does not show significant plateau**. 

#### Comparison with Dali's Peaks 
There are 64865 peak windows of varying size in Dali's data. With each down sample, call peak windows of 20bp. 
```{r, echo=T}
jump.rate.minus=c(0.1853, 0.1842, 0.1779, 0.1722, 0.1678, 0.1648, 0.1633)
jump.rate.plus=c(0.1850, 0.1840, 0.1775, 0.1717, 0.1669, 0.1635, 0.1631)
dali.rate.minus=c(0.7455, 0.7967, 0.8420, 0.8654, 0.8816, 0.8898, 0.8988)
dali.rate.plus=c(0.7456, 0.7956, 0.8427, 0.8657, 0.8810, 0.8897, 0.8982)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="", ylab="Proportion of Jump-seq", col=1, main="20bp, FDR 0.05", cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(1,2), lty=c(1,1))
axis.labels <-c("40%", "50%", "60%", "70%", "80%", "90%", "100%")
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type="o", xaxt='n', xlab="", ylab="Proportion of nano hmC_seal", col=3, main="20bp, FDR 0.05", cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("40%", "50%", "60%", "70%", "80%", "90%", "100%")
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)
mtext("overlap between Jump-seq & nano hmC_seal", outer = TRUE, cex = 1)
```

```{r, echo=T}
jump.rate.minus=c(0.1501, 0.1496, 0.1450, 0.1394, 0.1345, 0.1305, 0.1275)
jump.rate.plus=c(0.1495, 0.1488, 0.1447, 0.1392, 0.1344, 0.1304, 0.1274)
dali.rate.minus=c(0.8736, 0.8920, 0.9042, 0.9133, 0.9193, 0.9238, 0.9273)
dali.rate.plus=c(0.8732, 0.8913, 0.9036, 0.9132, 0.9187, 0.9242, 0.9271)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="", ylab="Proportion of Jump-seq", col=1, main="20bp, FDR 0.5", cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(1,2), lty=c(1,1))
axis.labels <-c("40%", "50%", "60%", "70%", "80%", "90%", "100%")
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type="o", xaxt='n', xlab="", ylab="Proportion of nano hmC_seal", col=3, main="20bp, FDR 0.5", cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("40%", "50%", "60%", "70%", "80%", "90%", "100%")
axis(1, at=seq(1,7), labels=axis.labels, cex.axis=0.8)
mtext("overlap between Jump-seq & nano hmC_seal", outer = TRUE, cex = 1)
```



```{r, echo=F,results="hide", eval=F}
par(mfrow=c(1,2))
slices <- c(492278, 2964113) 
lbls <- c("overlap", "non-overlap")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,radius=0.7, labels = lbls, cex=0.7, col=rainbow(length(lbls)),
  	main="Jump-seq", cex.main=0.9, sub="90% reads plus strand")

slices <- c(59368, 5497) 
lbls <- c("overlap", "non-overlap")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,radius=0.7, labels = lbls, cex=0.7, col=rainbow(length(lbls)),
  	main=" hmC_seal", cex.main=0.9, sub="90% reads plus strand")
```



#### Cmparison with strong TAB-seq peaks
There are 2,057,636 strong base level peaks from TAB-seq. To compute overlapping rate, each base is extend by 10bp downstream, 10bp upstream, the same length as the called peak windows.  
```{r, echo=T}
jump.rate.minus=c(0.2678, 0.2677, 0.2560, 0.2630, 0.2566, 0.2492, 0.2432)
jump.rate.plus=c(0.2678, 0.2682, 0.2566, 0.2634, 0.2566, 0.2494, 0.2429)
TAB.rate.minus=c(0.2810, 0.3203, 0.3654, 0.3858, 0.4145, 0.4455, 0.4693)
TAB.rate.plus=c(0.2815, 0.3202, 0.3655, 0.3859, 0.4160, 0.4455, 0.4709)
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks", cex.main=0.9)
lines(jump.rate.plus, col=2, type="o")
lines(TAB.rate.minus, col=3, type="o")
lines(TAB.rate.plus, col=4, type="o")
legend(1, 0.8, c("jump.rate.minus", "jump.rate.plus", "StrongTAB.rate.minus", "StrongTAB.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("40% reads", "50% reads", "60% reads", "70% reads", "80% reads", "90% reads", "100% reads")
axis(1, at=seq(1,7), labels=axis.labels)
```


```{r, echo=T}
ratio=matrix(nrow=2, ncol=2)
ratio[,1]=c(0.9193, 0.4709)
ratio[,2]=c(0.9189, 0.4693)
rownames(ratio)=c("Nano_seal", "Tab-seq")
par(mfrow=c(1,2))
barplot(ratio[,1], beside=F, ylim=c(0,1.2), col=c("red", "darkblue"), xlab=c("plus strand"), main="", ylab="Sensitivity")
text(0.6, 1.0, ratio[1,1])
text(2, 0.6, ratio[2,1])
barplot(ratio[,1], beside=F, ylim=c(0,1.2), col=c("red", "darkblue"), xlab=c("minus strand"), main="", ylab="Sensitivity")
text(0.6, 1.0, ratio[1,2])
text(2, 0.6, ratio[2,2])
```



### enrichment-100% reads minus
To do enrichments analysis, peak windows are called with varying p values. Use FDR 0.05 as threshold to select enriched windows. Randomly pick the same number of windows as enriched windows from the whole genome as background. Perform enrichment analysis for enriched windows versus background windows when overlapping with nano hmC_seal  and strong TAB-seq. FDR 0.05 is used throughout and stated clearly otherwise.   
```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(519216, 73689)
count[,2]=3789453-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="hmC_Seal", main="48ng(combine)  minus strand (FDR 0.05)", ylab="Number of peak windows")

```

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(921479, 98580)
count[,2]=3789453-count[,1]
pval=fisher.test(count, alternative = "greater")$p.value
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="strong TAB-seq", main="48ng(combine) minus strand (FDR 0.05)", ylab="Number of peak windows")

```


### Jump distribution 
The reference is GSM882244_FDR_0.0484. First got non-overlapping 5hmC regions and make sure each window has only one 5hmC site. For each window, count the reads depth at every site, then add up reads depth of these non-overlapping windows. Reads distribution is the accumulative reads depth at every base of all non-overlapping windows. 

```{r, echo=T, eval=T}
minus.48ng=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\20161115_5hmC_Jump_Seq_2hmC_spike_in.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_minus_with_1_coverage.ave.perbase", header=F)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)-1,minus.48ng[,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)

plot(seq(-10, 10, by=1)-1,minus.48ng[91:111,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
mtext("20161115_5hmC_Jump_Seq_2hmC_spike_in.minus", outer = TRUE, cex = 1)

```

```{r, echo=T}
plus.48ng=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\20161115_5hmC_Jump_Seq_2hmC_spike_in.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_plus_with_1_coverage.ave.perbase", header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)+1,plus.48ng[,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
plot(seq(-10, 10, by=1)+1,plus.48ng[91:111,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
mtext("20161115_5hmC_Jump_Seq_2hmC_spike_in.plus", outer = TRUE, cex = 1)

```





#####################################################################################################
## 24ng 5hmC replicate
There are 5 24ng 5hmC replicates. 

rep | plus | minus
-----|--------|------
rep 1|3,613,864| 3,621,005
rep 2| 2,654,892|2,660,120
rep 3|4,634,559 |4,645,750
rep 4|1,846,242|1,853,346
rep 5| 5,072,002|5,085,060
Table: number of reads in each replicate


```{r, echo=T}
win.minus=c(255946, 1255527, 498618, 1474825, 555722)
win.plus=c(255426, 1262831, 499772, 1481412, 555236)
plot(win.minus, type="o", xaxt='n', xlab="replicates", ylab="Number of peak windows", col=1, main="Number of peak windows (20bp, FDR0.05)", cex=0.8)
lines(win.plus, col=2, type="o")
legend(1, 1500000, c("minus strand", "plus strand"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep1 ", "rep2", "rep3", "rep4", "rep5")
axis(1, at=seq(1,5), labels=axis.labels)
```


### Comparison with Dali's peaks

```{r, echo=T}
jump.rate.minus=c(0.1248, 0.0639, 0.1255, 0.0829, 0.1264)
jump.rate.plus=c(0.1254, 0.0646, 0.1258, 0.0833, 0.1265)
dali.rate.minus=c(0.3455, 0.6132, 0.5248, 0.7351, 0.5625)
dali.rate.plus=c(0.3465, 0.6181, 0.5281, 0.7390, 0.5610)
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali's peaks", cex=0.8)
lines(jump.rate.plus, col=2, type="o")
lines(dali.rate.minus, col=3, type="o")
lines(dali.rate.plus, col=4, type="o")
legend(1, 1, c("jump.rate.minus", "jump.rate.plus", "dali.rate.minus", "dali.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "rep5")
axis(1, at=seq(1,5), labels=axis.labels)


```


### Comparison with strong TAB-seq


```{r, echo=T}
jump.rate.minus=c(0.2610, 0.1230, 0.2865, 0.1672, 0.2730)
jump.rate.plus=c(0.2612, 0.1234, 0.2891, 0.1686, 0.2734)
TAB.rate.minus=c(0.0422, 0.0905, 0.0881, 0.1437, 0.0930)
TAB.rate.plus=c(0.0421, 0.0913, 0.0894, 0.1458, 0.0932)
plot(jump.rate.minus, ylim=c(0, 0.6), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks", cex.main=0.9)
lines(jump.rate.plus, col=2, type="o")
lines(TAB.rate.minus, col=3, type="o")
lines(TAB.rate.plus, col=4, type="o")
legend(1, 0.6, c("jump.rate.minus", "jump.rate.plus", "StrongTAB.rate.minus", "StrongTAB.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "rep5")
axis(1, at=seq(1,5), labels=axis.labels)
```


### enrichment-rep4 minus
```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(122190, 29091)
count[,2]=1474825-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="hmC_Seal", main="24ng rep4 minus strand", ylab="Number of peak windows")

```

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(246529, 17073)
count[,2]=1474825-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="strong TAB-seq", main="24ng rep4 minus strand", ylab="Number of peak windows")

```


###################################################################################################
## 12ng 5hmC replicate
There are 5 12ng 5hmC replicates. 

rep | plus | minus
-----|--------|------
rep 1| 1,538,440| 1,543,973
rep 2|2,817,848 |2,829,593
rep 3|1,805,014 |1,813,743
rep 4|1,962,631|1,976,519
rep 5| 1,365,146|1,375,074
Table: number of reads in each replicate

```{r, echo=T}
win.minus=c(1144297, 1008422, 1436959, 1197291)
win.plus=c(1312102, 1036507, 1441863, 1200396)
plot(win.minus, type="o", xaxt='n', xlab="replicates", ylab="Number of peak windows", col=1, main="Number of peak windows (20bp, FDR0.05)", cex=0.8)
lines(win.plus, col=2, type="o")
legend(1, 1420000, c("minus strand", "plus strand"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep1 ", "rep2", "rep3", "rep5")
axis(1, at=seq(1,4), labels=axis.labels)
```






### Comparison with Dali's peaks
```{r, echo=T}
jump.rate.minus=c(0.0876, 0.0665, 0.0734, 0.0933)
jump.rate.plus=c(0.0858, 0.0672, 0.0734, 0.0938)
dali.rate.minus=c(0.6385, 0.5558, 0.6989, 0.7158)
dali.rate.plus=c(0.7222, 0.5711, 0.7026, 0.7202)
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali's peaks", cex=0.8)
lines(jump.rate.plus, col=2, type="o")
lines(dali.rate.minus, col=3, type="o")
lines(dali.rate.plus, col=4, type="o")
legend(1, 0.5, c("jump.rate.minus", "jump.rate.plus", "dali.rate.minus", "dali.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep5")
axis(1, at=seq(1,4), labels=axis.labels)


```

### Comparison with strong TAB-seq 

```{r, echo=T}
jump.rate.minus=c(0.1182, 0.0822, 0.0932, 0.1250)
jump.rate.plus=c(0.1180, 0.0826, 0.0933, 0.1254)
TAB.rate.minus=c(0.0776, 0.0472, 0.0765, 0.0861)
TAB.rate.plus=c(0.0888, 0.0487, 0.0767,0.0865)
plot(jump.rate.minus, ylim=c(0, 0.4), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks", cex.main=0.9)
lines(jump.rate.plus, col=2, type="o")
lines(TAB.rate.minus, col=3, type="o")
lines(TAB.rate.plus, col=4, type="o")
legend(1, 0.4, c("jump.rate.minus", "jump.rate.plus", "StrongTAB.rate.minus", "StrongTAB.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep5")
axis(1, at=seq(1,4), labels=axis.labels)
```

### enrichment-rep 5 minus strand 

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(111734, 23171)
count[,2]=1197291-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="hmC_Seal", main="12ng rep5 minus strand", ylab="Number of peak windows")

```

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(149717, 31057)
count[,2]=1197291-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="strong TAB-seq", main="12ng rep5 minus strand", ylab="Number of peak windows")

```

#####################################################################################################
## 6ng 5hmC replicate

### reads distribution
```{r, echo=T}
reads=matrix(nrow=11, ncol=1)
rownames(reads)=c(seq(1,10), ">10"); reads[,1]=c(473607, 8448, 370, 60, 30, 13, 8, 7, 5, 6, 94)
barplot(reads[,1]/sum(reads[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of called peak windows at FDR0.05 (rep1 minus)", xlab="Number of reads")
```

There are 5 6ng 5hmC replicates. 

rep | plus | minus
-----|--------|------
rep 1| 506,562| 510,127
rep 2| 900,135|902,501
rep 3|1,049,521 |1,051,758
rep 4|785,990|792,942
rep 5| 705,891|713,330
Table: number of reads in each replicate


```{r, echo=T}

win.plus=c(482648, 838387, 966623, 736000, 665408)
win.minus=c(482398, 836889, 964130, 735655, 665737)
plot(win.plus, type="o", xaxt='n', xlab="replicates", ylab="# windows of 20bp", col=1, main="Number of peak windows (20bp, FDR0.05)", cex=0.5)
lines(win.minus, col=2, type="o")
legend(2.5, 700000, c("plus_FDR0.05", "minus_FDR0.05"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep 1", "rep2", "rep3", "rep4", "rep5")
axis(1, at=seq(1,5), labels=axis.labels)

```




### Comparison with Dali's peaks

```{r, echo=T}
jump.rate.minus=c(0.0934, 0.0771, 0.0764, 0.0917, 0.0922)
jump.rate.plus=c(0.0926, 0.0773, 0.0765, 0.0913, 0.0920)
dali.rate.minus=c(0.4538, 0.5586, 0.5996, 0.5753, 0.5469)
dali.rate.plus=c(0.4523, 0.5612, 0.6015, 0.5719, 0.5444)
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows of 20 bp (FDR0.05) with Dali's peaks", cex=0.8)
lines(jump.rate.plus, col=2, type="o")
lines(dali.rate.minus, col=3, type="o")
lines(dali.rate.plus, col=4, type="o")
legend(1, 1, c("jump.rate.minus", "jump.rate.plus", "dali.rate.minus", "dali.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "rep5")
axis(1, at=seq(1,5), labels=axis.labels)


```

### Comparison with strong TAB-seq


```{r, echo=T}
jump.rate.minus=c(0.1653, 0.1413, 0.1411, 0.1720, 0.1724)
jump.rate.plus=c(0.1657, 0.1410, 0.1409, 0.1721, 0.1723)
TAB.rate.minus=c(0.0492, 0.0715, 0.0818, 0.0771, 0.0704)
TAB.rate.plus=c(0.0492, 0.0715, 0.0820, 0.0769, 0.0699)
plot(jump.rate.minus, ylim=c(0, 0.6), type="o", xaxt='n', xlab="number of reads", ylab="overlapping rate", col=1, main="Overlapping rate of peak windows (20bp,FDR0.05) with strong TAB-seq peaks", cex.main=0.9)
lines(jump.rate.plus, col=2, type="o")
lines(TAB.rate.minus, col=3, type="o")
lines(TAB.rate.plus, col=4, type="o")
legend(1, 0.6, c("jump.rate.minus", "jump.rate.plus", "StrongTAB.rate.minus", "StrongTAB.rate.plus"), col=c(1,2,3,4), lty=c(1,1,1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels)
```

### enrichment-rep 5 minus strand 

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(61404, 12907)
count[,2]=665737-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="hmC_Seal", main="6ng rep5 minus strand", ylab="Number of peak windows")

```

```{r, echo=T}
count=matrix(nrow=2, ncol=2)
count[,1]=c(114794, 17049)
count[,2]=665737-count[,1]
barplot(count[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="strong TAB-seq", main="6ng rep5 minus strand", ylab="Number of peak windows")

```


#####################################################################################################

## 2point4ng 5hmC replicate

rep | plus | minus
-----|---------|-----
rep 1| 270,885 | 272,807
rep 2| 165,553| 167,277
rep 3|279,404| 281,109
rep 4| 213,739| 215,604
combine| 929,581 | 926,797
Table: number of reads in every replicate. 

```{r, echo=T}

win.plus2=c(257809, 152655, 264996, 202465, 505788) # FDR0.05 (~pvalue cutoff 0.01)
win.minus2=c(257918, 152853, 264218, 202051, 521062)
plot(win.plus2, type="o", ylim=c(150000, 530000), xaxt='n', xlab="replicates", ylab="# windows of 20bp", col=1, main="Number of peak windows", cex=0.5)
lines(win.minus2, col=2, type="o")
legend(1, 400000, c("plus_FDR0.05", "minus_FDR0.05"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep 1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels)

```


### Reads distribution 
```{r, echo=T}
reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), ">10"); reads.plus[,1]=c(253804, 3709, 159, 31, 16, 11, 1, 6, 7, 4, 61)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of windows with reads (rep1 plus)", xlab="Number of reads")
```
```{r, echo=T}
reads.plus=matrix(nrow=11, ncol=1)
rownames(reads.plus)=c(seq(1,10), ">10"); reads.plus[,1]=c(371791, 128383, 4260, 985, 115, 67, 27, 14, 11, 10, 125)
barplot(reads.plus[,1]/sum(reads.plus[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of windows with reads (combine plus)", xlab="Number of reads")
```


```{r, echo=T}
reads.minus=matrix(nrow=11, ncol=1)
rownames(reads.minus)=c(seq(1,10), ">10"); reads.minus[,1]=c(253875, 3720, 166, 31, 21, 8, 7, 3, 2, 3, 82)
barplot(reads.minus[,1]/sum(reads.minus[,1]), ylim=c(0,1), ylab="Freq", main="Frequency of called peak windows at FDR0.05  (rep1 minus)", xlab="Number of reads")

```


### Comparison with Dali's peaks 

```{r, echo=T}
jump.rate.minus=c(0.0854, 0.0711, 0.0830, 0.0809, 0.0829)
jump.rate.plus=c(0.0852, 0.0713, 0.0818, 0.0793, 0.0822)
dali.rate.minus=c(0.2711, 0.1481, 0.2712, 0.2137, 0.4154)
dali.rate.plus=c(0.2692, 0.1490, 0.2674, 0.2093, 0.4014)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="", ylab="Proportion of Jump-seq", col=1, main="20bp, FDR>0.05", cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)

plot(dali.rate.minus, ylim=c(0,1), type="o", xaxt='n', xlab="", ylab="Proportion of nano hmC_seal", col=3, main="20bp, FDR>0.05", cex.main=0.8, cex.lab=0.8)
lines(dali.rate.plus, col=4, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)
mtext("overlap between Jump-seq & nano hmC_seal", outer = TRUE, cex = 1)

```

```{r, echo=T}
combine.minus=c(0.0818, 0.0840, 0.1435, 0.1059, 0.1000, 0.0641, 0.0606, 0.0588, 0, 0, 0)
plot(seq(1,11), combine.minus, type="o", xaxt='n', xlab="Number of reads", ylim=c(0, 0.2), ylab="Proportion of Jump-seq (combine minus)")
axis.labels <-c(seq(1,10), ">10")
abline(h=0.0194, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)
```


* This is the enrichment of peak windows at different reads counts for combined 2.4ng replicates. 
*  This is not a good measure since the number of widnows with large number of reads usually is small, say 2 windows, the overlapping rate with nano hmc_seal is not so reliable, (with large variance). 
* red line is background rate, 1.94%. 

```{r, echo=T}
combine.plus=c(0.0814, 0.0828, 0.1254, 0.1289, 0.1478, 0.1045, 0.0741, 0, 0.0909, 0, 0.0080)
plot(seq(1,11), combine.plus, type="o", xaxt='n', xlab="Number of reads", ylim=c(0, 0.2), ylab="Proportion of Jump-seq (combine plus)")
axis.labels <-c(seq(1,10), ">10")
abline(h=0.0194, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)
```

### Comparison with strong TAB-seq

```{r, echo=T}
jump.rate.minus=c(0.1543, 0.1295, 0.1544, 0.1483, 0.1489)
jump.rate.plus=c(0.1564, 0.1292, 0.1547, 0.1491, 0.1496)
TAB.rate.minus=c(0.0246, 0.0122, 0.0253, 0.0186, 0.0477)
TAB.rate.plus=c(0.0249, 0.0122, 0.0253, 0.0187, 0.0462)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(jump.rate.minus, ylim=c(0, 1), type="o", xaxt='n', xlab="", ylab="Proportion of Jump-seq", col=1, main="20bp, FDR>0.05", cex.main=0.8, cex.lab=0.8)
lines(jump.rate.plus, col=2, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(1,2), lty=c(1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)


plot(TAB.rate.minus, ylim=c(0,1), type="o", xaxt='n', xlab="", ylab="Proportion of TAB-seq", col=3, main="20bp, FDR>0.05", cex.main=0.8, cex.lab=0.8)
lines(TAB.rate.plus, col=4, type="o")
legend(1, 0.7, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("rep1", "rep2", "rep3", "rep4", "combine")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8)
mtext("overlap between Jump-seq & TAB-seq", outer = TRUE, cex = 1)
```

```{r, echo=T}
combine.minus=c(0.1462, 0.1519, 0.2660, 0.2118, 0.2357, 0.0769, 0.2121, 0.1176, 0.0769, 0.3333, 0.2349)
plot(seq(1,11), combine.minus, type="o", xaxt='n', xlab="Number of reads", ylim=c(0, 0.4), ylab="Proportion of Jump-seq (combine minus)")
axis.labels <-c(seq(1,10), ">10")
abline(h=0.0258, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)
```

```{r, echo=T}
combine.plus=c(0.1466, 0.1532, 0.2643, 0.2609, 0.2174, 0.1642, 0.1481, 0.3571, 0.1818, 0.1000, 0.2960)
plot(seq(1,11), combine.plus, type="o", xaxt='n', xlab="Number of reads", ylim=c(0, 0.4), ylab="Proportion of Jump-seq (combine plus)")
axis.labels <-c(seq(1,10), ">10")
abline(h=0.0258, col=2)
axis(1, at=seq(1,11), labels=axis.labels, cex.axis=0.8)
```

### enrichment-2.4ng combine minus strand 

```{r, echo=T}
count1=matrix(nrow=2, ncol=2)
count1[,1]=c(43213, 10092)
count1[,2]=521062-count1[,1]
barplot(count1[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="hmC_Seal", main="2.4ng (combine)", ylab="Number of peak windows")

```

```{r, echo=T}
count2=matrix(nrow=2,ncol=2)
count2[,1]=c(98049, 13382)
count2[,2]=521062-count2[,1]
barplot(count2[,1], beside=F, col=c("red", "darkblue"), legend=c("Observed", "Expected"), xlab="strong TAB-seq", main="2.4ng (combine)", ylab="Number of peak windows")

```


### jump distribution-2.4ng 

```{r, echo=T, eval=T}
minus.2.4ng=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\CHe-lu-2.4ng_mixed_with_other_library.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_minus_with_1_coverage.ave.perbase", header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)-1,minus.2.4ng[,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
plot(seq(-10, 10, by=1)-1,minus.2.4ng[91:111,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
mtext("CHe-lu-2.4ng_mixed_with_other_library.minus", outer = TRUE, cex = 1)


```

```{r, echo=T}
plus.2.4ng=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\JumpSeq-project\\data\\CHe-lu-2.4ng_mixed_with_other_library.mm9.umi_encoded_adaptor_removed_no_mismatch.sorted.dedup_GSM882244_FDR_0.0484_plus_with_1_coverage.ave.perbase", header=F)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))
plot(seq(-100, 100, by=1)+1,plus.2.4ng[,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
plot(seq(-10, 10, by=1)+1,plus.2.4ng[91:111,1], type="l", ylab="coverage", xlab="Distance", main="")
abline(v=0, col=2)
mtext("CHe-lu-2.4ng_mixed_with_other_library.plus", outer = TRUE, cex = 1)
```

### enrichment at cis-regulatory regions 
download cis-regulatory peaks at [Ren lab][renlab]. They are sinlge base peaks and extended to 20 bp windows.  

[renlab]: http://chromosome.sdsc.edu/mouse/download.html

```{r, echo=T}
percent=matrix(nrow=2, ncol=5)
percent[,1]=c(0.0017, 0.0006)
percent[,2]=c(0.0013, 0.0006)

percent[,3]=c(0.0009, 0.0004)
percent[,4]=c(0.0006, 0.0004)

percent[,5]=c(0.0018, 0.0002)
colnames(percent)=c("ctcf", "enhancer", "h3k27ac", "h3k4me3", "polII")

barplot(percent, beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,0.004),
xlab="Proportion of Jump-seq windows", main="combine minus (FDR0.05)", ylab="", cex.names=1)

```



####################################################################################################
## Summary

### Overlap pattern at different genomic concentration 

* Jump-seq overlaps with Dali's peaks

Horizontal line is genome-wide overlap rate of 0.0203. 

```{r, echo=T}
jump.rate=c(0.0829, 0.0922, 0.0933, 0.1264, 0.1397)
dali.rate=c(0.4154, 0.5469, 0.7158, 0.5625, 0.9189)
plot(jump.rate, type="o", ylim=c(0,1), xaxt='n', xlab="samples", ylab="overlap rate", col=1, main="", cex=0.8)
lines(dali.rate, col=2, type="o")
abline(h=0.0203, col=3)
legend(1, 1, c("Jump-seq", "Dali's peak", "background rate"), col=c(1,2, 3), lty=c(1,1, 1))
axis.labels <-c("2.4ng combine ", "6ng", "12ng", "24ng", "48ng(combine)")
axis(1, at=seq(1,5), labels=axis.labels)

```



* Jump-seq overlaps with  strong TAB-seq

Horizontal line is the genome-wide overlap rate of 0.0258.  
```{r, echo=T}
jump.rate=c(0.1489, 0.1724, 0.1250, 0.2730, 0.2432)
TAB.rate=c(0.0477, 0.0704, 0.0861, 0.0930, 0.4693)
plot(jump.rate, type="o", ylim=c(0,0.5), xaxt='n', xlab="samples", ylab="overlap rate", col=1, main="", cex=0.8)
lines(TAB.rate, col=2, type="o")
abline(h=0.0260, col=3) # background rate
legend(1, 0.5, c("Jump-seq", "strong TAB-seq", "background rate"), col=c(1,2,3), lty=c(1,1, 1))
axis.labels <-c("2.4ng combine ", "6ng", "12ng", "24ng", "48ng(combine)")
axis(1, at=seq(1,5), labels=axis.labels)

```

### Enrichemnt analysis 

```{r, echo=T}
percent=matrix(nrow=2, ncol=10)
percent[,1]=c(43213, 10092)/521062
percent[,2]=c(98049, 13382)/521062

percent[,3]=c(61404, 12907)/665737
percent[,4]=c(114794, 17049)/665737

percent[,5]=c(111734, 23171)/1197291
percent[,6]=c(149717, 31057)/1197291

percent[,7]=c(122190, 28799)/1474825
percent[,8]=c(246529, 37851)/1474825

percent[,9]=c(529216, 73689)/3789453
percent[,10]=c(921479, 98580)/3789453
colnames(percent)=rep(c("2.4ng", "6ng", "12ng", "24ng", "48ng"), each=2)

par(mfrow=c(1,2))
barplot(percent[,c(1,3,5,7,9)], beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,0.3),
xlab="Figure 1A", main="", ylab="Proportion of Nano-seal peaks", cex.names=0.5)

barplot(percent[,c(2,4,6,8,10)], beside=T, col=c("red", "darkblue"), legend=c("Jump-seq peaks", "Random windows"), ylim=c(0,0.4),
xlab="Figure 1B", main="", ylab="Proportion of Tab-seq peaks", cex.names=0.5)
```
## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
