---
title: "PN analysis"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE}
knitr::read_chunk("chunks.R")
library(knitr)
library(RColorBrewer)
library(gplots)
set.seed(123)
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

## mouse genome (mm9)

download from UCSC table brower. choose ucsc gene and use default setting. 
The output bed file is like 

```{r, echo=T, eval=F}
        start    end     UCSC Transcript ID
chr1    3195984 3205713 uc007aet.1      0       -       3195984 3195984 0       2       1414,2194,      0,7535,
chr1    3204562 3661579 uc007aeu.1      0       -       3206102 3661429 0       3       2487,200,947,   0,207220,
456070,
chr1    3638391 3648985 uc007aev.1      0       -       3638391 3638391 0       2       2199,58,        0,10536,
....
```

Use the first 3 columns as coordinate for the genes in the fourth column. Get the gene id conversion file following [here][here]

[here]:http://genomewiki.ucsc.edu/index.php/Gene_id_conversion. 

```{r, echo=T, eval=F}
uc007afh.1      Lypla1
uc007afg.1      Lypla1
uc007afi.2      Tcea1
uc011wht.1      Tcea1
uc011whu.1      Tcea1
...
```
So each gene symbol may have multiple UCSC gene transcript and merge them together. 





## lu1-jumpbarcode1(5hmC)

Transposase barcode  | total unique reads | plus reads / minus reads
----|-------|----------
1 (E9.5, negative)| 55,944| 28,245/27,699
2 (PN1)|8,429|4,089/4,340
3 (PN2)| 32,606| 14,799/17,807
4 (PN3)| 41,445 | 19,319/22,126
5 (PN4)| 29,586| 14,089/15,497
6| 648 | 280/368
7 (PN1)| 30,982 | 15,316/15,666
8 (PN2)| 7,782 | 3,877/3,905
9 (PN3)| 32,754 | 16,329/16,425
10 (PN4)| 42,382 | 21,133/21,249
Table: reads in every transposase bacode sample 

* Paternal: 6, 7, 8, 9, 10; maternal: 1, 2, 3, 4, 5


### Maternal samples 


![Whole genome reads distribution of plus strand-TP2345 ](figure/lu1_jumpbarcode1/lu1_jumpbarcode1_tp2345_plus.png)

* The tracks from top to bottom: Ref_seq(mm9), PN1, PN2, PN3, PN4
* To use IGV, I created 21 bp window by extending 10 bp updtream and 10 bp downstream at every reads location. 


####  mouse imprinting genes 

150 imprinted genes are downloaded from this [site][printed].


[printed]:http://www.geneimprint.com/site/genes-by-species.Mus+musculus


```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.plus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2[match(as.character(imprinting.gene$Gene), reads.gene.tp2$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3[match(as.character(imprinting.gene$Gene), reads.gene.tp3$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4[match(as.character(imprinting.gene$Gene), reads.gene.tp4$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5[match(as.character(imprinting.gene$Gene), reads.gene.tp5$Gene),]
combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")
combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at plus strand) from maternal samples at different stages")
```

```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, (max.vau-1), c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* use Kmeans to cluster genes with number of clusters fixed at 3. 

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```




![Whole genome reads distribution of minus strand-TP2345 ](figure/lu1_jumpbarcode1/lu1_jumpbarcode1_tp2345_minus.png)

```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.minus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]

combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at minus strand) from maternal samples at different stages")
```

```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, (max.vau-1), c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```




### Paternal samples 

![Whole genome reads distribution of plus strand-TP78910 ](figure/lu1_jumpbarcode1/lu1_jumpbarcode1_tp78910_plus.png)

```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.plus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]

combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at plus strand) from paternal samples at different stages")
```

```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, (max.vau-1), c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```





![Whole genome reads distribution of minus strand-TP78910 ](figure/lu1_jumpbarcode1/lu1_jumpbarcode1_tp78910_minus.png)


```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.minus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]


combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at minus strand) from paternal samples at different stages")
```

```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, (max.vau-1), c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```



#################################################
#################################################

## lu1-jumpbarcode3/4

### MII whole embryo (transposon-2)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
3| 20,851 | 10,218/10,633
4|1,437|705/732




![Whole genome reads distribution ](figure/lu1_jumpbarcode34/lu1.wholeembryo.png)



### DBA 2 sperm (transposon-7)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
3| 62,366  | 30,614/31,752
4|3,982|1,920/2,062


### ICR sperm (transposon-8)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
3| 235 | 7,216/7,350
4|1,013|482/531







#############################################################
#############################################################


## lu2-jumpbarcode2(5mC)

Transposase barcode  | total unique reads | plus reads / minus reads
----|-------|----------
1(E9.5, negative)| 1,587| 752/835
2 (PN1)|16,647|8,298/8,349
3 (PN2)| 26,834| 13,076/13758
4 (PN3)| 12,157 | 6052/6105
5 (PN4)| 9,003| 4,433/4,570
6| 45,489 | 22,894/22,595
7 (PN1)| 6,752 | 3,467/3,285
8 (PN2)| 4,547 | 2,346/2,201
9 (PN3)| 2,087 | 1,082/1,005
10 (PN4)| 14,692 | 7,361/7,331
Table: reads in every transposase bacode sample 

#####################################################
### Maternal samples 

![Whole genome reads distribution of minus strand-TP2345 ](figure/lu2_jumpbarcode2/lu2_jumpbarcode2_tp2345_minus.png)




#### imprinting genes 





```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.minus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]

combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of 38 imprinting genes (at minus strand) from Maternal samples at different stages")
```

```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(3, 8, c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```





![Whole genome reads distribution of plus strand-TP2345 ](figure/lu2_jumpbarcode2/lu2_jumpbarcode2_tp2345_plus.png)

```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.plus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]


combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at plus strand) from Maternal samples at different stages")
```



```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(3, 8, c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```

######################################################

### Paternal samples 

![Whole genome reads distribution of minus strand-TP78910 ](figure/lu2_jumpbarcode2/lu2_jumpbarcode2_tp78910_minus.png)

```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.minus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]

combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads of imprinting genes (at minus strand) from Maternal samples at different stages")
```



```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, 5, c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```



![Whole genome reads distribution of plus strand-TP2345 ](figure/lu2_jumpbarcode2/lu2_jumpbarcode2_tp78910_plus.png)
```{r,  echo=F}
imprinting.gene=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\MouseGenome\\imprinting_genes.txt", header=T)
reads.gene.tp2.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5.minus=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.plus.cove.base.level.shiftonebase.txt", header=T)
imprinting.gene.reads.tp2.minus=reads.gene.tp2.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp2.minus$Gene),]
imprinting.gene.reads.tp3.minus=reads.gene.tp3.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp3.minus$Gene),]
imprinting.gene.reads.tp4.minus=reads.gene.tp4.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp4.minus$Gene),]
imprinting.gene.reads.tp5.minus=reads.gene.tp5.minus[match(as.character(imprinting.gene$Gene), reads.gene.tp5.minus$Gene),]

combine=cbind(imprinting.gene.reads.tp2.minus, imprinting.gene.reads.tp3.minus$No.reads, imprinting.gene.reads.tp4.minus$No.reads, imprinting.gene.reads.tp5.minus$No.reads)
combine=combine[,-3]
colnames(combine)[2:5]=c("PN1", "PN2", "PN3", "PN4")

combine.nona=combine[complete.cases(combine), ]
zero.index=intersect(intersect(intersect(which(combine.nona$PN1==0), which(combine.nona$PN2==0)), which(combine.nona$PN3==0)), which(combine.nona$PN4==0))
combine.fin=combine.nona[-c(zero.index),]
rownames(combine.fin)=NULL
```

```{r, echo=T}
kable(combine.fin, caption="Number of reads  of imprinting genes (at plus strand) from Maternal samples at different stages")
```



```{r, echo=F}
clust=kmeans(combine.fin[,2:5],3)
means=clust$centers
max.vau=max(means)
plot(means[1,], ylim=c(0,max.vau), type="o", col=1, xaxt='n', ylab="No.reads", xlab="")
lines(means[2,], col=2, type="o")
lines(means[3,], col=3, type="o")
legend(1, (max.vau-1), c("clust 1", "clust 2", "clust 3"), col=c(1,2,3), lty=c(1,1,1))
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels)
```

* genes in cluster1 

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==1)])
```

* genes in cluster 2

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==2)])
```

* genes in cluster 3

```{r, echo=F}
as.character(combine.fin$Gene[which(clust$cluster==3)])
```




########################################################
########################################################

## heatmap

### whole genome 

* every column is a normalized vector of reads counts of a gene, 

```{r, echo=F, warning=F,eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.plus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5hmC maternal (plus)", cexCol=0.7, labCol=F, na.rm=T)

```

![qqter](figure/)






```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.minus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5hmC maternal (minus)", cexCol=0.7, labCol=F, na.rm=T)

```



```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.plus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5hmC Paternal (plus)", cexCol=0.7, labCol=F, na.rm=T)

```


```{r, echo=F, warning=F,eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.minus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5hmC Paternal (minus)", cexCol=0.7, labCol=F, na.rm=T)

```



```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.plus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5mC maternal (plus)", cexCol=0.7, labCol=F, na.rm=T)

```



```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP02.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP03.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP04.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP05.bam.minus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5mC maternal (minus)", cexCol=0.7, labCol=F, na.rm=T)

```


```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.plus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.plus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5mC Paternal (plus)", cexCol=0.7, labCol=F, na.rm=T)

```



```{r, echo=F, warning=F, eval=F}
reads.gene.tp2=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP07.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp3=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP08.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp4=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP09.bam.minus.cove.base.level.shiftonebase.txt", header=T)
reads.gene.tp5=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\umi_encoded_adaptor_removed_no_mismatch_sorted_dedup_TP10.bam.minus.cove.base.level.shiftonebase.txt", header=T)


combine=cbind(reads.gene.tp2$No.reads, reads.gene.tp3$No.reads, reads.gene.tp4$No.reads, reads.gene.tp5$No.reads)
all.reads=as.matrix(t(combine))
rownames(all.reads)=c("PN1", "PN2", "PN3", "PN4")
colnames(all.reads)=reads.gene.tp2$Gene
heatcol <- brewer.pal(9, "Greens")
scale.reads=scale(all.reads)

heatmap.2(scale.reads, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main=" 5mC Paternal (minus)", cexCol=0.7, labCol=F, na.rm=T)

```

### important regions

* each column is a normalized vector of reads count. 

#### 5mC


```{r, echo=F, warning=FALSE}
mc=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\important.region.reads.summary.txt", header=T)
pa.mc=mc[c(14,16,18,20),]; ma.mc=mc[c(4,6,8,10),]
rownames(pa.mc)=c("PN1", "PN2", "PN3", "PN4")
colnames(pa.mc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")
rownames(ma.mc)=c("PN1", "PN2", "PN3", "PN4")
colnames(ma.mc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")

heatcol <- brewer.pal(9, "Greens")
scale.pa.mc=scale(as.matrix(pa.mc))
scale.ma.mc=scale(as.matrix(ma.mc))

heatmap.2(scale.pa.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5mC (plus strand)", cexCol=0.7)


heatmap.2(scale.ma.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5mC  (plus strand)", cexCol=0.7)


par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

plot(scale.ma.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:9)
  lines(scale.ma.mc[,i], col=i, type="o")
legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)

plot(scale.pa.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-paternal", cex.main=0.8, cex.lab=0.5)
for (i in 2:9)
  lines(scale.pa.mc[,i], col=i, type="o")
#legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("plus strand", outer = TRUE, cex = 1)

```




```{r, echo=F, warning=FALSE}
mc=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu2_jumpbarcode2\\important.region.reads.summary.txt", header=T)
pa.mc=mc[c(13,15,17,19),]; ma.mc=mc[c(3,5,7,9),]
rownames(pa.mc)=c("PN1", "PN2", "PN3", "PN4")
colnames(pa.mc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")
rownames(ma.mc)=c("PN1", "PN2", "PN3", "PN4")
colnames(ma.mc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")

heatcol <- brewer.pal(9, "Greens")
scale.pa.mc=scale(as.matrix(pa.mc))
scale.ma.mc=scale(as.matrix(ma.mc))

heatmap.2(scale.pa.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5mC (minus strand)", cexCol=0.7)


heatmap.2(scale.ma.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5mC  (minus strand)", cexCol=0.7)


par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

plot(scale.ma.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:9)
  lines(scale.ma.mc[,i], col=i, type="o")
legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)

plot(scale.pa.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-paternal", cex.main=0.8, cex.lab=0.5)
for (i in 2:9)
  lines(scale.pa.mc[,i], col=i, type="o")
#legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("minus strand", outer = TRUE, cex = 1)






```



#### 5hmC


```{r, echo=F, warning=FALSE}
hmc=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\important.region.reads.summary.txt", header=T)
pa.hmc=hmc[c(14,16,18,20),]; ma.hmc=hmc[c(4,6,8,10),]
rownames(pa.hmc)=c("PN1", "PN2", "PN3", "PN4")
colnames(pa.hmc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")
rownames(ma.hmc)=c("PN1", "PN2", "PN3", "PN4")
colnames(ma.hmc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")

heatcol <- brewer.pal(9, "Greens")
scale.pa.hmc=scale(as.matrix(pa.hmc))
scale.ma.hmc=scale(as.matrix(ma.hmc))

heatmap.2(scale.pa.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC (plus strand)", cexCol=0.8)


heatmap.2(scale.ma.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC  (plus strand)", cexCol=0.8)


par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

plot(scale.ma.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:9)
  lines(scale.ma.hmc[,i], col=i, type="o")
legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)

plot(scale.pa.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-paternal", cex.main=0.8, cex.lab=0.5)
for (i in 2:9)
  lines(scale.pa.hmc[,i], col=i, type="o")
#legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("plus strand", outer = TRUE, cex = 1)




```




```{r, echo=F, warning=FALSE}
hmc=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\important.region.reads.summary.txt", header=T)
pa.hmc=hmc[c(13,15,17,19),]; ma.hmc=hmc[c(3,5,7,9),]
rownames(pa.hmc)=c("PN1", "PN2", "PN3", "PN4")
colnames(pa.hmc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")
rownames(ma.hmc)=c("PN1", "PN2", "PN3", "PN4")
colnames(ma.hmc)=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "imprinting", "promoter", "enhancer")

heatcol <- brewer.pal(9, "Greens")
scale.pa.hmc=scale(as.matrix(pa.hmc))
scale.ma.hmc=scale(as.matrix(ma.hmc))

heatmap.2(scale.pa.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC (minus strand)", cexCol=0.8)


heatmap.2(scale.ma.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC  (minus strand)", cexCol=0.8)

par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

plot(scale.ma.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:9)
  lines(scale.ma.hmc[,i], col=i, type="o")
legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)

plot(scale.pa.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-paternal", cex.main=0.8, cex.lab=0.5)
for (i in 2:9)
  lines(scale.pa.hmc[,i], col=i, type="o")
#legend(1, 3, c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "trans", "imprinting", "promoter", "enhancer"), col=seq(1,9), lty=rep(1,9), ncol=3)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("minus strand", outer = TRUE, cex = 1)




```


### reads correlation

```{r, echo=F}
#hmc=c(32606, 41445, 7782, 32754)
#diff.hmc=c((41445-32606), (32754-7782))
#mc=c(26834, 12157, 4547, 2087)
#diff.mc=c((12157-26834), (2087-4547))
#cor(hmc, mc)
```

## Paternal site changes at PN4 (5mC vs 5hmC)

* 5hmC plus strand has 20,767 sites and 105 (0.5056%) are overlapping (20 bp windows) with 5mC plus strand; 5mC plus strand has 6575 sites and 73 (1.1103%) are overlapping with 5hmC plus strand.  


window size | 5hmC (plus, 20,767) | 5mC (plus, 6,575) | 5hmC (minus, 20, 933) | 5mC (minus)
------------|-------------|--------------|------------|------------
20 bp |105 (0.5056%) | 73 (1.1103%)| 272 | 141
100 bp| 367 | 202 | 446 | 183
1000 bp| 497 | 309 | 593 | 283


window size | 5hmC (plus, 20,767) | 5mC (PN3:plus, 939) | 5hmC (minus, 20,933) | 5mC ( PN3:  minus, 939)
------------|-------------|--------------|------------|------------
20 bp |0 |0 | 53 | 124 
100 bp| 
1000 bp| 
Table:  overlap between 5hmC (PN4) and 5mC (PN3)


### PN1 (5mC) vs PN4 (5hmC)

```{r, echo=F}
pn1.pn4.overlap=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\pn1.pn4.overlap", header=F)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

hmc.plus=pn1.pn4.overlap[c(1,3,5,7),3]; hmc.minus=pn1.pn4.overlap[c(2,4,6,8),6]
plot(hmc.minus, ylim=c(0,0.2), type="o", xaxt='n', xlab="", ylab="Proportion of 5hmC at PN4", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(hmc.plus, col=4, type="o")
legend(1, 0.15, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)


mc.plus=pn1.pn4.overlap[c(1,3,5,7),6]; mc.minus=pn1.pn4.overlap[c(2,4,6,8),6]
plot(mc.minus, ylim=c(0,0.2), type="o", xaxt='n', xlab="", ylab="Proportion of 5mC at PN1", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(mc.plus, col=4, type="o")
legend(1, 0.15, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("overlap between 5mC (PN1) & 5hmC (PN4)", outer = TRUE, cex = 1)

```





### PN2 (5mC) vs PN4 (5hmC)

```{r, echo=F}
pn2.pn4.overlap=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\pn2.pn4.overlap", header=F)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

hmc.plus=pn2.pn4.overlap[c(1,3,5,7),3]; hmc.minus=pn2.pn4.overlap[c(2,4,6,8),6]
plot(hmc.minus, ylim=c(0,0.2), type="o", xaxt='n', xlab="", ylab="Proportion of 5hmC at PN4", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(hmc.plus, col=4, type="o")
legend(1, 0.15, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)


mc.plus=pn2.pn4.overlap[c(1,3,5,7),6]; mc.minus=pn2.pn4.overlap[c(2,4,6,8),6]
plot(mc.minus, ylim=c(0,0.2), type="o", xaxt='n', xlab="", ylab="Proportion of 5mC at PN2", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(mc.plus, col=4, type="o")
legend(1, 0.15, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("overlap between 5mC (PN2) & 5hmC (PN4)", outer = TRUE, cex = 1)

```



### PN3 (5mC) vs PN4 (5hmC)

```{r, echo=F}
pn3.pn4.overlap=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\PN\\lu1_jumpbarcode1\\pn3.pn4.overlap", header=F)
par(mfrow=c(1,2), oma = c(0, 0, 2, 0))

hmc.plus=pn3.pn4.overlap[c(1,3,5,7),3]; hmc.minus=pn3.pn4.overlap[c(2,4,6,8),6]
plot(hmc.minus, ylim=c(0,0.4), type="o", xaxt='n', xlab="", ylab="Proportion of 5hmC at PN4", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(hmc.plus, col=4, type="o")
legend(1, 0.3, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)


mc.plus=pn3.pn4.overlap[c(1,3,5,7),6]; mc.minus=pn3.pn4.overlap[c(2,4,6,8),6]
plot(mc.minus, ylim=c(0,0.4), type="o", xaxt='n', xlab="", ylab="Proportion of 5mC at PN3", col=3, main="", cex.main=0.8, cex.lab=0.8)
lines(mc.plus, col=4, type="o")
legend(1, 0.3, c("minus", "plus"), col=c(3,4), lty=c(1,1))
axis.labels <-c("20bp", "100bp", "1kb", "5kb")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8)
mtext("overlap between 5mC (PN3) & 5hmC (PN4)", outer = TRUE, cex = 1)

```


## PN2 vs PN3

### Maternal 

window size | PN2 (plus, 10,534) | PN3 (plus, 5,101) | PN2 (minus, 10,935) | PN3 (minus, 5,208)
------------|-------------|--------------|------------|------------
20 bp | 0 | 0| 2,128 | 1,256
100 bp| 1,829 | 1,133 | 2,210| 1,268
1000 bp| 1,808 | 1,133 |2,201 | 1,272
Table: 5mC: Overlapping windows between PN2 and PN3. 


window size | PN2 (plus, 13,733) | PN3 (plus, 18,114) | PN2 (minus, 14, 787) | PN3 (minus, 19,467)
------------|-------------|--------------|------------|------------
20 bp | 0 | 0|2,535 | 2,341
100 bp|1,682 | 1,655| 2, 586 | 2,391
1000 bp|1,788 | 1,777 | 2,712 | 2,543
Table: 5hmC: Overlapping windows between PN2 and PN3.  



### Paternal


window size | PN2 (plus, 2,139) | PN3 (plus, 946) | PN2 (minus, 2,011) | PN3 (minus, 938)
------------|-------------|--------------|------------|------------
20 bp |52 | 89| 36| 37
100 bp| 57 | 96 | 74 | 57
1000 bp|90 |105| 131 | 69
Table: 5mC: Overlapping windows between PN2 and PN3. 


window size | PN2 (plus, 3,814) | PN3 (plus, 16,042) | PN2 (minus, 3,860) | PN3 (minus, 16,142)
------------|------------------|------------|------------|-------------|---------
20 bp |0 | 0| 157| 271
100 bp| 136 | 315 | 211 | 463
1000 bp| 181| 402| 273 | 542
Table: 5hmC: Overlapping windows between PN2 and PN3.  




## lu2-jumpbarcode5/6


### lu2-MII whole embryo (transposon-2)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
5| 801 | 397/404
6|4,927|2,450/2,477


### lu2-DBA 2 sperm (transposon-7)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
5| 286  | 145/141
6|1,499|733/2,766


### lu2- ICR sperm (transposon-8)

Jump barcode  | total unique reads | plus reads / minus reads
----|-------|----------
5| 235 | 110/125
6|1,436|720/716



## lu1-jumpbarcode3

![Whole genome reads distribution of TP4](figure/lu1_jumpbarcode3/lu1_jumpbarcode3tp4.png)





<!-- Insert the session information into the document -->
```{r session-info}
```
