---
title: "Sample with spike in"
author: "Shengtong  Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
knitr::read_chunk("chunks.R")
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
set.seed(123)
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

All analysiz have been adjusted with spike in sample. 

## Rep1-20180113 


* In this batch, sperm fails. 

* PN1 female, PN2 female and PN5 are replaced with deep sequencing data on 2018/06/06

```{r, echo=F}
#region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\important.region.reads", header=F) # use original annotation
region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\important.region.by.xiaolong.reads", header=F) # use Xiaolong's annotation

samples=c("hmc-PN5", "hmc-1cell", "hmc-2cellearly", "hmc-2celllate", "hmc-MII", "hmc-sperm", "hmc-ES", "mc-MII", "mc-sperm", "mc-ES", "mc-600pgmES", "hmc-PN1female", "hmc-PN1male", "hmc-PN2female", "hmc-PN2male", "hmc-PN3female", "hmc-PN3male", "hmc-PN4female", "hmc-PN4male")
#regions=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "promoter", "enhancer")
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.sample=length(samples)
num.regions=length(regions)

region.reads.correct=matrix(nrow=num.sample, ncol=num.regions) # add two strands 
for (i in 1:num.sample)
  region.reads.correct[i,]=as.numeric(region.reads[((i-1)*2+1),2:(num.regions+1)]+region.reads[i*2,2:(num.regions+1)])
#region.reads.correct=region.reads.correct[,-1]
spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\spike.in.reads", header=F)

supple.region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\important.region.by.xiaolong.reads", header=F, fill=T)[-3,]
supple.spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\spike.in.reads", header=F, fill=T)[-3,]

region.reads.correct[12,]=as.numeric(supple.region.reads[9,2:ncol(supple.region.reads)]) # for PN1-female, replace the old with the supplement
region.reads.correct[14,]=as.numeric(supple.region.reads[10,2:ncol(supple.region.reads)]) # for PN2-female, replace the old with the supplement
region.reads.correct[1,]=as.numeric(supple.region.reads[1,2:ncol(supple.region.reads)]) # for PN5, replace the old with the supplement

spikein.reads[12,2]=supple.spikein.reads[9,2] # replace with new deep sequencing data
spikein.reads[14,2]=supple.spikein.reads[10,2]
spikein.reads[1,2]=supple.spikein.reads[1,2]


reads.summary=cbind(region.reads.correct, as.numeric(spikein.reads[,2]))
rownames(reads.summary)=samples
colnames(reads.summary)=c(regions, "spikein")
last.col.index=num.regions+1
scale.factor=reads.summary[,last.col.index]/min(reads.summary[,last.col.index])
scale.reads=matrix(nrow=num.sample, ncol=num.regions)
for (i in 1:num.sample)
  scale.reads[i,]=as.numeric(reads.summary[i,1:num.regions]*scale.factor[i])
rownames(scale.reads)=rownames(reads.summary)
colnames(scale.reads)=colnames(reads.summary)[1:num.regions]
#write.csv(scale.reads, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\scale.reads.xiaolong.annotation.csv")
```

### Paternal 

```{r, echo=F, warning=F}
all.stage=c("Sperm", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly", "2celllate", "ES")
sub.stage=c( "PN1","PN2","PN3","PN4")
#regions=c("exon", "intron", "3UTR", "5UTR", "CpGIsland", "transposable", "promoter", "enhancer")
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron","Intergenic", "Promoter")
scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")

hmc_male=rbind(new.scale.reads[20,], scale.reads[13,], scale.reads[15,], scale.reads[17,], scale.reads[19,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
row.names(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
heatcol <- brewer.pal(9, "Greens")
hmc_male2=hmc_male[2:5,]
scale.male.hmc=scale(as.matrix(hmc_male))
scale.male.hmc2=scale(as.matrix(hmc_male2))
colnames(scale.male.hmc2)=regions
colnames(scale.male.hmc)=regions
rownames(scale.male.hmc)=all.stage
rownames(scale.male.hmc2)=sub.stage

heatmap.2(scale.male.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC", cexCol=0.8)
heatmap.2(scale.male.hmc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.male.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.male.hmc[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-all.stage
axis(1, at=seq(1,length(all.stage)), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.male.hmc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.male.hmc2[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-sub.stage
axis(1, at=seq(1,length(sub.stage)), labels=axis.labels, cex.axis=0.8, las=2)
```


```{r, echo=F, eval=F}
library(gridGraphics)
library(grid)

grab_grob <- function(){
  grid.echo()
  grid.grab()
}

arr <-list()
arr[[1]]=scale.male.hmc
arr[[2]]=scale.male.hmc2

library(gplots)
gl <- lapply(1:2, function(i){
  heatmap.2(arr[[i]], Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC", cexCol=0.8)
  grab_grob()
}
)

grid.newpage()
library(gridExtra)
grid.arrange(grobs=gl, ncol=2, clip=TRUE)

```


### Maternal 


```{r, echo=F, warning=F}
hmc_female=rbind(scale.reads[5,], scale.reads[12,], scale.reads[14,], scale.reads[16,], scale.reads[18,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )

rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]
heatcol <- brewer.pal(9, "Greens")
scale.female.hmc=scale(as.matrix(hmc_female))
hmc_female2=hmc_female[2:5,]
scale.female.hmc=scale(as.matrix(hmc_female))
scale.female.hmc2=scale(as.matrix(hmc_female2))
colnames(scale.female.hmc2)=regions
colnames(scale.female.hmc)=regions
rownames(scale.female.hmc)=c("MII", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly", "2celllate", "ES")
rownames(scale.female.hmc2)=c( "PN1","PN2","PN3","PN4")
heatmap.2(scale.female.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC", cexCol=0.8)
heatmap.2(scale.female.hmc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.female.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.female.hmc[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("MII", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly", "2celllate", "ES")
axis(1, at=seq(1,10), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.female.hmc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.female.hmc2[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8, las=2)

```


###


```{r, echo=F}
gene.body.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\gene.body.reads", header=F)
gene.exon.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\gene.exon.reads", header=F)
gene.promoter.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\gene.promoter.reads", header=F)
spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\spike.in.reads", header=F)
```

### Combining maternal and paternal 

```{r, echo=F}
scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")

hmc_female=rbind(scale.reads[5,], scale.reads[12,], scale.reads[14,], scale.reads[16,], scale.reads[18,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
hmc_male=rbind(new.scale.reads[20,], scale.reads[13,], scale.reads[15,], scale.reads[17,], scale.reads[19,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
hmc_combine=rbind(hmc_female[1:5,], hmc_male)
scale_hmc_combine=scale(as.matrix(hmc_combine[,-1])) # remove the first column of labels 
min.limit=min(scale_hmc_combine)-0.5; max.limit=max(scale_hmc_combine)+0.5
par(mfrow=c(1,3))
plot(scale_hmc_combine[1:5,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=1)
for (i in 2:num.regions)
  lines(scale_hmc_combine[1:5,i], col=i, type="o")
legend(1, max.limit, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.65)
axis.labels <-c("MII", "PN1","PN2","PN3","PN4")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale_hmc_combine[6:10,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale_hmc_combine[6:10,i], col=i, type="o")
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("Sperm", "PN1","PN2","PN3","PN4")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale_hmc_combine[11:15,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main="", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale_hmc_combine[11:15,i], col=i, type="o")
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN5","1cell","2cellearly", "2celllate", "ES")
axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

```


## Rep 2-20180228


* 1 cell is replaced with deep sequencing data on 2018/06/06


```{r, echo=F, eval=F}
#region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\important.region.reads", header=F)[,-1]
region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\important.region.by.xiaolong.reads", header=F)[,-1]
spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\spike.in.reads", header=F)

supple.region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\important.region.by.xiaolong.reads", header=F, fill=T)[-3,]
supple.spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\spike.in.reads", header=F, fill=T)[-3,]
region.reads[2,]=supple.region.reads[2,2:ncol(supple.region.reads)]
spikein.reads[2,2]=supple.spikein.reads[2,2]

reads.summary=as.matrix(cbind(region.reads, as.numeric(spikein.reads[,2])))
rownames(reads.summary)=c("hmc-PN5", "hmc-1cell", "hmc-2cellearly", "hmc-2celllate", "hmc-MII", "hmc-sperm", "hmc-sperm-repeat", "hmc-ES",  "hmc-PN1female", "hmc-PN1male", "hmc-PN2female", "hmc-PN2male", "hmc-PN3female", "hmc-PN3male", "hmc-PN4female", "hmc-PN4male")
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)
colnames(reads.summary)=c(regions, "spikein")
scale.factor=reads.summary[,ncol(reads.summary)]/min(reads.summary[,ncol(reads.summary)])
scale.reads=matrix(nrow=nrow(region.reads), ncol=num.regions)
for (i in 1:nrow(region.reads))
  scale.reads[i,]=reads.summary[i,1:num.regions]*scale.factor[i]
rownames(scale.reads)=rownames(reads.summary)
colnames(scale.reads)=colnames(reads.summary)[1:num.regions]
#write.csv(scale.reads, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\scale.reads.xiaolong.annotation.csv")
```

### Paternal 

```{r, echo=F, warning=F}
all.stages=c("Sperm", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly", "2celllate", "ES")
num.stages=length(all.stages)
second.rep.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")
hmc_male=rbind(new.scale.reads[20,], second.rep.scale.reads[10,], second.rep.scale.reads[12,], second.rep.scale.reads[14,], second.rep.scale.reads[16,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
heatcol <- brewer.pal(9, "Greens")
hmc_male2=hmc_male[2:5,]
scale.male.hmc=scale(as.matrix(hmc_male))
scale.male.hmc2=scale(as.matrix(hmc_male2))
colnames(scale.male.hmc2)=regions
colnames(scale.male.hmc)=regions
rownames(scale.male.hmc)=all.stages
rownames(scale.male.hmc2)=c( "PN1","PN2","PN3","PN4")
heatmap.2(scale.male.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC", cexCol=0.8)
heatmap.2(scale.male.hmc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5hmC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.male.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.male.hmc[,i], col=i, type="o")
legend(1, 4, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-all.stage
axis(1, at=seq(1,num.stages), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.male.hmc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.male.hmc2[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8, las=2)
```


### Maternal 

```{r, echo=F, warning=F}
all.stages=c("MII", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly", "2celllate", "ES")
num.stages=length(all.stages)
regions=regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)
second.rep.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\scale.reads.xiaolong.annotation.csv")
hmc_female=rbind(second.rep.scale.reads[5,], second.rep.scale.reads[9,], second.rep.scale.reads[11,], second.rep.scale.reads[13,], second.rep.scale.reads[15,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]
heatcol <- brewer.pal(9, "Greens")
hmc_female2=hmc_female[2:5,]
scale.female.hmc=scale(as.matrix(hmc_female))
scale.female.hmc2=scale(as.matrix(hmc_female2))
colnames(scale.female.hmc2)=regions
colnames(scale.female.hmc)=regions
rownames(scale.female.hmc)=all.stages
rownames(scale.female.hmc2)=c( "PN1","PN2","PN3","PN4")
heatmap.2(scale.female.hmc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC", cexCol=0.8)
heatmap.2(scale.female.hmc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5hmC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.female.hmc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.female.hmc[,i], col=i, type="o")
legend(1, 4, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-all.stages
axis(1, at=seq(1,num.stages), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.female.hmc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.female.hmc2[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8, las=2)
```


## 5hmC  two replicates 

```{r, echo=F}
male.first.stage=c("Sperm", "PN1","PN2","PN3","PN4")
female.first.stage=c("MII", "PN1","PN2","PN3","PN4")
common.second.stage=c("PN5","1cell","2cellearly", "2celllate", "ES")
################# replicate 1 ##############
scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180113\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")
hmc_female=rbind(scale.reads[5,], scale.reads[12,], scale.reads[14,], scale.reads[16,], scale.reads[18,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]

hmc_male=rbind(new.scale.reads[20,], scale.reads[13,], scale.reads[15,], scale.reads[17,], scale.reads[19,], scale.reads[1,], scale.reads[2:4,], scale.reads[7,] )
rownames(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
first_hmc_combine=rbind(hmc_female[1:5,], hmc_male)

scale_first_hmc_combine=scale(as.matrix(first_hmc_combine)) # remove the first column of labels 

############## replicate 2 ####################
second.rep.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180228\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")
hmc_male=rbind(new.scale.reads[20,], second.rep.scale.reads[10,], second.rep.scale.reads[12,], second.rep.scale.reads[14,], second.rep.scale.reads[16,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_male)=hmc_male[,1]
hmc_male=hmc_male[,-1]
hmc_female=rbind(second.rep.scale.reads[5,], second.rep.scale.reads[9,], second.rep.scale.reads[11,], second.rep.scale.reads[13,], second.rep.scale.reads[15,], second.rep.scale.reads[1,], second.rep.scale.reads[2:4,], second.rep.scale.reads[8,] )
rownames(hmc_female)=hmc_female[,1]
hmc_female=hmc_female[,-1]

second_hmc_combine=rbind(hmc_female[1:5,], hmc_male)
scale_second_hmc_combine=scale(as.matrix(second_hmc_combine))



scale_hmc_combine=scale_first_hmc_combine
min.limit=min(scale_hmc_combine)-0.5; max.limit=max(scale_hmc_combine)+0.5
######### plot 1*3 
#par(mfrow=c(1,3))
#plot(scale_hmc_combine[1:5,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5hmC-Maternal", cex.main=0.8, cex.lab=1)
#for (i in 2:num.regions)
#  lines(scale_hmc_combine[1:5,i], col=i, type="o")
#legend(1, max.limit, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.65)
#axis.labels <-c("MII", "PN1","PN2","PN3","PN4")
#axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

#plot(scale_hmc_combine[6:10,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main=" 5hmC-Paternal", cex.main=0.8, cex.lab=0.6)
#for (i in 2:num.regions)
#  lines(scale_hmc_combine[6:10,i], col=i, type="o")
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
#axis.labels <-c("Sperm", "PN1","PN2","PN3","PN4")
#axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

#plot(scale_hmc_combine[11:15,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="", col=1, main="", cex.main=0.8, cex.lab=0.6)
#for (i in 2:num.regions)
#  lines(scale_hmc_combine[11:15,i], col=i, type="o")
#legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
#axis.labels <-c("PN5","1cell","2cellearly", "2celllate", "ES")
#axis(1, at=seq(1,5), labels=axis.labels, cex.axis=0.8, las=2)

#######
male.index=c(seq(1,5), seq(11,15))
female.index=c(seq(6,10), seq(11,15))
plot(scale_hmc_combine[male.index,1], ylim=c(min.limit,max.limit), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main="5hmC at different stages", cex.main=0.8, cex.lab=1, lty=1, pch=16, lwd=2)
for (i in 2:num.regions)
  lines(scale_hmc_combine[male.index,i], col=i, type="o", lty=1, pch=16, lwd=2)
legend(1, max.limit, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.65)
for (i in 1:num.regions)
  lines(scale_hmc_combine[female.index,i], col=i, type="o", lty=3, pch=16, lwd=2)
#legend(1, max.limit, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.65)

axis.labels <-c("MII/Sperm", "PN1","PN2","PN3","PN4", common.second.stage)
axis(1, at=seq(1,length(male.index)), labels=axis.labels, cex.axis=0.8, las=2)

```


### Comparison with deep sequencing data  YL-0413


![Comparison with deep sequencing in IGV](figure/igv_comparison.png)


```{r, echo=F}
overlap.rate=matrix(nrow=3, ncol=2)
overlap.rate[1,]=c(0.7893, 0.9501)
overlap.rate[2,]=c(0.7776, 0.9727)
overlap.rate[3,]=c(0.8174, 0.9130)
rownames(overlap.rate)=c("YL-17 vs lu3-pn1male", "YL-20 vs lu16-sperm", "YL-21 vs lu17-ES" )
barplot(t(overlap.rate),beside=T,  col=c("red", "blue"), ylim=c(0,1.2), legend=c("deep sequencing", "20180228-5hmC"), main="Proportion of overlapping 100 bp windows", args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=1 )

```


## 5mC-20180227


* MII is replaced with deep sequencing data on 2018/06/06

```{r, echo=F, eval=F}
samples=c("mc-PN5","mc-1cell", "mc-2cellearly", "mc-MII", "mc-sperm", "mc-sperm-repeat", "mc-ES",  "mc-PN1female", "mc-PN1male", "mc-PN2female", "mc-PN2male", "mc-PN3female", "mc-PN3male", "mc-PN4female", "mc-PN4male", "mc-2celllate")
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)
region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180227\\important.region.by.xiaolong.reads", header=F)[,-1]
supple.region.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\important.region.by.xiaolong.reads", header=F, fill=T)[-3,]
region.reads[4,]=supple.region.reads[3,2:ncol(supple.region.reads)] # for mc-MII, replace the old with the supplement
spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180227\\spike.in.reads", header=F)
supple.spikein.reads=read.table("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180606\\spike.in.reads", header=F, fill=T)[-3,]
spikein.reads[4,2]=supple.spikein.reads[3,2]
region.reads=rbind(region.reads, supple.region.reads[8,2:ncol(supple.region.reads)]) # for mc-2cell late, add it to original sample 
spikein.reads[16,2]=supple.spikein.reads[8,2]


reads.summary=as.matrix(cbind(region.reads, as.numeric(spikein.reads[,2])))
rownames(reads.summary)=samples
colnames(reads.summary)=c(regions, "spikein")
scale.factor=reads.summary[,ncol(reads.summary)]/min(reads.summary[,ncol(reads.summary)])
scale.reads=matrix(nrow=nrow(region.reads), ncol=num.regions)
for (i in 1:nrow(region.reads))
  scale.reads[i,]=reads.summary[i,1:num.regions]*scale.factor[i]
rownames(scale.reads)=rownames(reads.summary)
colnames(scale.reads)=colnames(reads.summary)[1:num.regions]
#write.csv(scale.reads, file="D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180227\\scale.reads.xiaolong.annotation.csv")
```

### Paternal 

```{r, echo=F, warning=F}
all.stages=c("Sperm", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly",  "2celllate", "ES")
num.stages=length(all.stages)
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)
scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180227\\scale.reads.xiaolong.annotation.csv")
new.scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180420\\scale.reads.xiaolong.annotation.csv")
#mc_male=rbind(scale.reads[5,], scale.reads[9,], scale.reads[11,], scale.reads[13,], scale.reads[15,], scale.reads[1:3,], scale.reads[7,] )
mc_male=rbind(new.scale.reads[8,], scale.reads[9,], scale.reads[11,], scale.reads[13,], scale.reads[15,], scale.reads[1:3,], scale.reads[16,], scale.reads[7,] )
rownames(mc_male)=mc_male[,1]
mc_male=mc_male[,-1]
heatcol <- brewer.pal(9, "Greens")
mc_male2=mc_male[2:5,]
scale.male.mc=scale(as.matrix(mc_male))
scale.male.mc2=scale(as.matrix(mc_male2))
colnames(scale.male.mc2)=regions
colnames(scale.male.mc)=regions
rownames(scale.male.mc)=all.stages
rownames(scale.male.mc2)=c( "PN1","PN2","PN3","PN4")
heatmap.2(scale.male.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5mC", cexCol=0.8)
heatmap.2(scale.male.mc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Paternal 5mC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.male.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-Paternal", cex.main=0.8, cex.lab=0.6, lwd=2)
for (i in 2:num.regions)
  lines(scale.male.mc[,i], col=i, type="o", lwd=2)
legend(1, 4, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-all.stages
axis(1, at=seq(1,num.stages), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.male.mc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-Paternal", cex.main=0.8, cex.lab=0.6)
for (i in 2:num.regions)
  lines(scale.male.mc2[,i], col=i, type="o")
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=3, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8, las=2)
```


### Maternal 

```{r, echo=F, warning=F}
all.stages=c("MII", "PN1","PN2","PN3","PN4","PN5","1cell","2cellearly",  "2celllate", "ES")
num.stages=length(all.stages)
regions=c("3UTR", "5UTR", "CDS", "TTS", "Intron", "Intergenic","Promoter")
num.regions=length(regions)

scale.reads=read.csv("D:\\ResearchWork\\StatisticalGenetics\\JumpSeq\\20180227\\scale.reads.xiaolong.annotation.csv")
mc_female=rbind(scale.reads[4,], scale.reads[8,], scale.reads[10,], scale.reads[12,], scale.reads[14,], scale.reads[1:3,], scale.reads[16,], scale.reads[7,] )
rownames(mc_female)=mc_female[,1]
mc_female=mc_female[,-1]
heatcol <- brewer.pal(9, "Greens")
mc_female2=mc_female[2:5,]
scale.female.mc=scale(as.matrix(mc_female))
scale.female.mc2=scale(as.matrix(mc_female2))
colnames(scale.female.mc2)=regions
colnames(scale.female.mc)=regions
rownames(scale.female.mc)=all.stages
rownames(scale.female.mc2)=c( "PN1","PN2","PN3","PN4")
heatmap.2(scale.female.mc, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5mC", cexCol=0.8)
heatmap.2(scale.female.mc2, Rowv =F, Colv = F, scale="none", key=TRUE,col=heatcol, symbreaks=T,trace="none", density.info="none",keysize=2, key.xlab=NA, main="Maternal 5mC", cexCol=0.8)

par(mfrow=c(1,2))
plot(scale.female.mc[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-Maternal", cex.main=0.8, cex.lab=0.6, lwd=2)
for (i in 2:num.regions)
  lines(scale.female.mc[,i], col=i, type="o", lwd=2)
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.45)
axis.labels <-all.stages
axis(1, at=seq(1,num.stages), labels=axis.labels, cex.axis=0.8, las=2)

plot(scale.female.mc2[,1], ylim=c(-2,3), type="o", xaxt='n', xlab="", ylab="normalized reads count", col=1, main=" 5mC-Maternal", cex.main=0.8, cex.lab=0.6, lwd=2)
for (i in 2:num.regions)
  lines(scale.female.mc2[,i], col=i, type="o", lwd=2)
legend(1, 3, regions, col=seq(1,num.regions), lty=rep(1,num.regions), ncol=2, cex=0.45)
axis.labels <-c("PN1", "PN2", "PN3", "PN4")
axis(1, at=seq(1,4), labels=axis.labels, cex.axis=0.8, las=2)
```



## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
