---
title: "Single Cell"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

## Quality Control (QC)
### reads distribution across all 960 seal single cells 
most reads are not mappable to reference genome, for instance 

```{r, eval=F}
cat CHe-lu-2_S2_L006_R2_001_Nbarcode_TTGGGGCTAGATGTGTATAAGAGACAG.adaptor_removed_no_mismatch_sorted_dedup_D23.bam.flagstat
416 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
0 + 0 mapped (0.00%:-nan%)
416 + 0 paired in sequencing
208 + 0 read1
208 + 0 read2
0 + 0 properly paired (0.00%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (0.00%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```



### bulk seal 5hmC data 

```{r, echo=T, eval=FALSE}
cat CHe-lu-2_S2_L006_R2_001_Nbarcode_CTTCTAGCAGATGTGTATAAGAGACAG.adaptor_removed_no_mismatch_sorted_dedup_A01.bam.flagstat
9912976 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
1358618 + 0 mapped (13.71%:-nan%)
9912976 + 0 paired in sequencing
4956488 + 0 read1
4956488 + 0 read2
1358618 + 0 properly paired (13.71%:-nan%)
1358618 + 0 with itself and mate mapped
0 + 0 singletons (0.00%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)


```

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
